{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "74c04205-1d11-4b06-8f15-29c4e6fdf0f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51ae2473-354f-4a51-b767-84d600536ece",
   "metadata": {},
   "source": [
    "# ONT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "da2337b0-a4f9-4dd2-857e-0075b3576f29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_df = pd.read_csv('hprc_metadata_sample_files_ONT.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "840d6350-7ebd-41d4-82d8-0bf18a77f8e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if ont_df['notes'].isnull().all() == True:\n",
    "    ont_df = ont_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b737011d-c7db-4a50-8a72-0798a64f612c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: Integrate with HPRC_metadata/merge_metadata.py combine_readstats.keep_columns\n",
    "ont_meta_columns = ['filename',\n",
    "        'sample_ID',\n",
    "        'path',\n",
    "        'filetype',\n",
    "        'library_ID',\n",
    "        'library_strategy',\n",
    "        'library_source',\n",
    "        'library_selection',\n",
    "        'library_layout',\n",
    "        'platform',\n",
    "        'instrument_model',\n",
    "        'design_description',\n",
    "        'data_type',\n",
    "        'shear_method',\n",
    "        'size_selection',\n",
    "        'seq_kit',\n",
    "        'basecaller',\n",
    "        'basecaller_version',\n",
    "        'basecaller_model',\n",
    "        'generator_facility',\n",
    "        'generator_contact']\n",
    "\n",
    "ont_readstat_columns = ['sample_ID', \n",
    "                        'filename',\n",
    "                        'read_N50',\n",
    "                        'Gb',\n",
    "                        'coverage',\n",
    "                        '100kb+',\n",
    "                        '200kb+',\n",
    "                        '300kb+',\n",
    "                        '400kb+',\n",
    "                        '500kb+',\n",
    "                        '1Mb+',\n",
    "                        'whales']\n",
    "\n",
    "ont_df_nan_columns = ont_df.columns[ont_df.isna().any()].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "60c2ac04-0e46-4d3b-90c4-5f02a5c2f517",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14\n"
     ]
    }
   ],
   "source": [
    "ont_sample_missing_1_submitter_df = ont_df[ont_df.isnull().any(axis=1)]\n",
    "\n",
    "# Number of ONT samples that have at least one missing submitter file missing\n",
    "print(len(set(ont_df[ont_df['sample_ID'].isin(ont_sample_missing_1_submitter_df['sample_ID'].tolist())]['sample_ID'].tolist())))\n",
    "\n",
    "# Export ONT sample files that have at least one missing submitter file missing\n",
    "ont_df[ont_df['sample_ID'].isin(ont_sample_missing_1_submitter_df['sample_ID'].tolist())].reset_index(drop=True).to_csv('hprc_metadata_sample_files_missing_1_submitter_ONT.tsv', sep='\\t')\n",
    "\n",
    "for nan_column in ont_df_nan_columns:\n",
    "    ont_df[nan_column] = ont_df[nan_column].ffill().tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "39efbb45-d23e-4205-8248-c8cd35a1e589",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_meta_df = ont_df[ont_meta_columns].drop_duplicates(subset=['sample_ID'])\n",
    "ont_sample_readstats_df = ont_df[ont_readstat_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "647365b5-1cb9-49ef-ae19-9fc744499f14",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check no missing sample_ID for the sample meta and readstat subset\n",
    "assert ont_df[~ont_df['sample_ID'].isin(ont_sample_meta_df['sample_ID'].tolist())].shape[0] == ont_df[~ont_df['sample_ID'].isin(ont_sample_readstats_df['sample_ID'].tolist())].shape[0] == 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7d632452-8890-4f6d-8db9-4a0b7a5a3eb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_readstats_df = ont_sample_readstats_df.groupby('sample_ID').agg({'read_N50': 'mean',\n",
    "                                                                          'coverage':'sum',\n",
    "                                                                          '100kb+': 'sum',\n",
    "                                                                          '200kb+': 'sum',\n",
    "                                                                          '300kb+': 'sum',\n",
    "                                                                          '400kb+': 'sum',\n",
    "                                                                          '500kb+': 'sum',\n",
    "                                                                          '1Mb+': 'sum',\n",
    "                                                                          'whales': 'sum'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d4a230a1-5044-483e-beff-036478edd6a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df = pd.merge(ont_sample_meta_df, ont_sample_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0328faca-30c7-4fcb-b488-672aae9ed04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(set(ont_sample_df['sample_ID'].tolist())) == ont_sample_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ef949b29-e2dd-484c-acf2-20a7d8497b3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df.to_csv('hprc_metadata_sample_aggregate_ONT.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c195e0ba-6b3b-49ae-816d-80c1a6c5dd1b",
   "metadata": {},
   "source": [
    "# HiFi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "212775d8-d606-4918-8e71-eb7e294fbbf8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>accession</th>\n",
       "      <th>study</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>...</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>ccs_algorithm</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>m54329U_200124_193652.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1596347</td>\n",
       "      <td>27122049640</td>\n",
       "      <td>27.12</td>\n",
       "      <td>47</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>m54329U_200127_180554.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1755465</td>\n",
       "      <td>29883779569</td>\n",
       "      <td>29.88</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>m54329U_200129_001928.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1779732</td>\n",
       "      <td>30200500419</td>\n",
       "      <td>30.20</td>\n",
       "      <td>45</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>m54329U_200130_064539.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1636421</td>\n",
       "      <td>27919782706</td>\n",
       "      <td>27.92</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>m54329U_200201_051510.ccs.bam</td>\n",
       "      <td>HG01123</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01123/ra...</td>\n",
       "      <td>SRR13684290</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861232</td>\n",
       "      <td>2072143</td>\n",
       "      <td>34554070679</td>\n",
       "      <td>34.55</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 17kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        filename sample_ID  \\\n",
       "0  m54329U_200124_193652.ccs.bam   HG01891   \n",
       "1  m54329U_200127_180554.ccs.bam   HG01891   \n",
       "2  m54329U_200129_001928.ccs.bam   HG01891   \n",
       "3  m54329U_200130_064539.ccs.bam   HG01891   \n",
       "4  m54329U_200201_051510.ccs.bam   HG01123   \n",
       "\n",
       "                                                path    accession      study  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "1  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "2  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "3  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "4  s3://human-pangenomics/working/HPRC/HG01123/ra...  SRR13684290  SRP305758   \n",
       "\n",
       "  biosample_accession  total_reads     total_bp  total_Gbp  min  ...  \\\n",
       "0        SAMN17861236      1596347  27122049640      27.12   47  ...   \n",
       "1        SAMN17861236      1755465  29883779569      29.88   46  ...   \n",
       "2        SAMN17861236      1779732  30200500419      30.20   45  ...   \n",
       "3        SAMN17861236      1636421  27919782706      27.92   46  ...   \n",
       "4        SAMN17861232      2072143  34554070679      34.55   46  ...   \n",
       "\n",
       "                          design_description        data_type  shear_method  \\\n",
       "0  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "1  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "2  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "3  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "4  HiFi sequencing of 17kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "\n",
       "   size_selection                        ccs_algorithm  polymerase_version  \\\n",
       "0         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "1         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "2         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "3         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "4         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "\n",
       "   seq_plate_chemistry_version        generator_facility generator_contact  \\\n",
       "0                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "1                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "2                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "3                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "4                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "\n",
       "  notes  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_df = pd.read_csv('hprc_metadata_sample_files_HiFi.tsv', sep='\\t')\n",
    "hifi_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "23991c33-88eb-48cc-8375-3c4b4a3e9fc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if hifi_df['notes'].isnull().all() == True:\n",
    "    hifi_df = hifi_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1b35b613-a4aa-4042-8db2-7345dcdab51f",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_hifi_1_submitter_columns = ['sample_ID', 'filename', 'library_ID', 'library_strategy', 'library_source', 'library_selection', 'library_layout', 'platform', 'instrument_model', \n",
    "                    'design_description', 'data_type', 'shear_method', 'size_selection', 'ccs_algorithm', 'polymerase_version', 'seq_plate_chemistry_version', \n",
    "                    'generator_facility', 'generator_contact', 'notes']\n",
    "\n",
    "required_hifi_5_readstats_columns = ['sample_ID','filename', 'total_reads', 'total_bp', 'total_Gbp', 'min', 'max', 'mean', 'quartile_25', 'quartile_50', 'quartile_75', 'N25', 'N50', 'N75']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8f54dd29-ba81-4462-b686-93c60ac63e1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "require_hifi_1_submitter_dict = {sample: {key: None for key in required_hifi_1_submitter_columns} for sample in hifi_df['sample_ID'].tolist()} \n",
    "for sample in list(set(hifi_df['sample_ID'].tolist())):\n",
    "    # print(sample, required_hifi_1_submitter_columns)\n",
    "    \n",
    "    sample_array = hifi_df[hifi_df['sample_ID'].isin([sample])][required_hifi_1_submitter_columns].values\n",
    "    \n",
    "    # print(sample)\n",
    "    # print(sample_array)\n",
    "    # print(len(required_hifi_1_submitter_columns))\n",
    "    \n",
    "    assert len(required_hifi_1_submitter_columns) == sample_array.shape[1]\n",
    "    \n",
    "    for column_index in range(sample_array.shape[1]):\n",
    "        # print(sample_array[0, column_index], sample_array[:, column_index])\n",
    "        if np.all(sample_array[:, column_index] == sample_array[0, column_index]):\n",
    "            # print(sample_array[0, column_index], \n",
    "            #       required_hifi_1_submitter_columns[column_index], \n",
    "            #       sample_array[:, column_index].tolist()[0])\n",
    "            # print('\\n')\n",
    "            require_hifi_1_submitter_dict[sample][required_hifi_1_submitter_columns[column_index]] = sample_array[0, column_index]\n",
    "        else:\n",
    "            # print(sample_array[0, column_index])\n",
    "            require_hifi_1_submitter_dict[sample][required_hifi_1_submitter_columns[column_index]] = sample_array[:, column_index]\n",
    "            # print(required_hifi_1_submitter_columns[column_index],\n",
    "                  # sample_array[:, column_index])\n",
    "            # print('\\n')\n",
    "            # pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "98ae3ffb-c207-4ad9-a382-ac3734a597a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# contains_none = any(value is None or (isinstance(value, list) and None in value) for value in require_hifi_1_submitter_dict.values())\n",
    "# contains_nan = any(value == 'nan' or (isinstance(value, list) and 'nan' in value) for value in require_hifi_1_submitter_dict.values())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "48985796-7253-4a33-9d43-dafe00d6db6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "numpy.ndarray"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(require_hifi_1_submitter_dict['HG01891']['filename'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5ea6de06-9a28-4e1b-9d06-7fa8642f37bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'sample_ID': 'HG02717',\n",
       " 'filename': array(['m64043_200403_163826.ccs.bam', 'm64043_200405_180950.ccs.bam',\n",
       "        'm64043_200407_002219.ccs.bam', 'm64043_200408_064651.ccs.bam'],\n",
       "       dtype=object),\n",
       " 'library_ID': 'HG02717_lib1',\n",
       " 'library_strategy': 'WGS',\n",
       " 'library_source': 'GENOMIC',\n",
       " 'library_selection': 'size fractionation',\n",
       " 'library_layout': 'single',\n",
       " 'platform': 'PACBIO_SMRT',\n",
       " 'instrument_model': 'PacBio Sequel II',\n",
       " 'design_description': 'HiFi sequencing of 20kb fractionated gDNA',\n",
       " 'data_type': 'unaligned reads',\n",
       " 'shear_method': 'Megaruptor 1',\n",
       " 'size_selection': 'SageELF',\n",
       " 'ccs_algorithm': '8.0.0.80529',\n",
       " 'polymerase_version': array([nan, nan, nan, nan], dtype=object),\n",
       " 'seq_plate_chemistry_version': array([nan, nan, nan, nan], dtype=object),\n",
       " 'generator_facility': 'Washington University',\n",
       " 'generator_contact': 'tgraves@wustl.edu',\n",
       " 'notes': array([nan, nan, nan, nan], dtype=object)}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "require_hifi_1_submitter_dict['HG02717']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "900a19f4-4844-42fc-82a7-6e53e537836c",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "samples_missing_submmiter_1_hifi = []\n",
    "for sample_id, meta_dict in require_hifi_1_submitter_dict.items():\n",
    "    # print(sample_id, meta_dict)\n",
    "    for key, value in meta_dict.items():\n",
    "        if isinstance(value, np.ndarray):\n",
    "            if any(item == 'nan' or pd.isna(item) for item in value):\n",
    "                if key == 'notes':\n",
    "                    pass\n",
    "                else:\n",
    "                    samples_missing_submmiter_1_hifi.append(sample_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "35b48e9c-f397-4a6d-9e17-8e6a7f0bb0fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_1_submitter_df = pd.DataFrame.from_dict(require_hifi_1_submitter_dict, orient='index')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "05f5bd27-1fe8-42bb-b188-6fffe7c3d8f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_1_submitter_df[hifi_1_submitter_df['sample_ID'].isin(samples_missing_submmiter_1_hifi)].to_csv('hprc_metadata_sample_files_missing_1_submitter_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "051a5a1f-6913-468c-bdfb-8376cdd36231",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Retired\n",
    "# hifi_sample_meta_df = hifi_df[required_hifi_1_submitter_columns].drop_duplicates(subset=['sample_ID']).reset_index(drop=True)\n",
    "# hifi_sample_missing_1_submitter_list = []\n",
    "# for sample in hifi_sample_meta_df['sample_ID'].tolist():\n",
    "#     if sum(hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin([sample])].isna().any()) == 0:\n",
    "#         pass\n",
    "#     else:\n",
    "#         hifi_sample_missing_1_submitter_df = hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin([sample])]\n",
    "#         if hifi_sample_missing_1_submitter_df.columns[hifi_sample_missing_1_submitter_df.isna().any()].tolist() == ['notes']:\n",
    "#             pass\n",
    "#         else:\n",
    "#             # print(sample, hifi_sample_missing_1_submitter_df.columns[hifi_sample_missing_1_submitter_df.isna().any()].tolist())\n",
    "#             hifi_sample_missing_1_submitter_list.append(sample)\n",
    "# print(len(hifi_sample_missing_1_submitter_list))\n",
    "\n",
    "# hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin(hifi_sample_missing_1_submitter_list)].reset_index().to_csv('hprc_metadata_sample_files_missing_1_submitter_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7c2f06ce-35fd-4189-bfd2-0de55acf4de9",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_readstats_df = hifi_df[required_hifi_5_readstats_columns]\n",
    "hifi_sample_readstats_df = hifi_sample_readstats_df.groupby('sample_ID').agg({'total_reads': 'sum',\n",
    "                                                   'total_bp': 'sum',\n",
    "                                                   'total_Gbp': 'sum',\n",
    "                                                   'min': 'mean',\n",
    "                                                   'max': 'mean',\n",
    "                                                   'quartile_25': 'mean',\n",
    "                                                   'quartile_50': 'mean',\n",
    "                                                   'quartile_75': 'mean',\n",
    "                                                   'N25': 'mean',\n",
    "                                                   'N50': 'mean', # assembly term shortest contig of the total assembly length (different for ONT)\n",
    "                                                   'N75': 'mean'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fa772927-2134-43ee-81fb-6301dec93e2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df = pd.merge(hifi_1_submitter_df, hifi_sample_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "672deb01-56a7-4a4f-9af8-938224e3ff67",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check that sample's with missing 1_submitter metadata are present\n",
    "assert hifi_sample_df[hifi_sample_df['sample_ID'].isin(samples_missing_submmiter_1_hifi)].shape[0] == len(set(samples_missing_submmiter_1_hifi))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "59a94d0f-69f3-4bd2-9a1a-5386dc0957fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df.to_csv('hprc_metadata_sample_aggregate_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0e135ef1-15dd-4a71-a466-d13c7ef1828d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>filename</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>...</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>quartile_50</th>\n",
       "      <th>quartile_75</th>\n",
       "      <th>N25</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01891</td>\n",
       "      <td>[m54329U_200124_193652.ccs.bam, m54329U_200127...</td>\n",
       "      <td>HG01891.HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>115126112334</td>\n",
       "      <td>115.12</td>\n",
       "      <td>46.0</td>\n",
       "      <td>44384.75</td>\n",
       "      <td>15781.75</td>\n",
       "      <td>16865.00</td>\n",
       "      <td>18096.000000</td>\n",
       "      <td>15930.00</td>\n",
       "      <td>17039.5</td>\n",
       "      <td>18287.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01123</td>\n",
       "      <td>[m54329U_200201_051510.ccs.bam, m54329U_200203...</td>\n",
       "      <td>HG01123_HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 17kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>119417336179</td>\n",
       "      <td>119.40</td>\n",
       "      <td>46.0</td>\n",
       "      <td>43082.25</td>\n",
       "      <td>14998.75</td>\n",
       "      <td>16470.25</td>\n",
       "      <td>18144.000000</td>\n",
       "      <td>15302.25</td>\n",
       "      <td>16815.5</td>\n",
       "      <td>18543.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02559</td>\n",
       "      <td>[m54329U_200211_192235.ccs.bam, m54329U_200217...</td>\n",
       "      <td>[HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>128540762559</td>\n",
       "      <td>128.54</td>\n",
       "      <td>46.2</td>\n",
       "      <td>47600.40</td>\n",
       "      <td>18524.80</td>\n",
       "      <td>19951.20</td>\n",
       "      <td>21644.200000</td>\n",
       "      <td>18764.80</td>\n",
       "      <td>20247.0</td>\n",
       "      <td>22029.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02486</td>\n",
       "      <td>[m54329U_200215_021808.ccs.bam, m54329U_200313...</td>\n",
       "      <td>[HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 22kb fractionated gDNA, Hi...</td>\n",
       "      <td>...</td>\n",
       "      <td>123904875583</td>\n",
       "      <td>123.92</td>\n",
       "      <td>46.5</td>\n",
       "      <td>45554.50</td>\n",
       "      <td>17439.50</td>\n",
       "      <td>18733.00</td>\n",
       "      <td>20268.833333</td>\n",
       "      <td>17658.50</td>\n",
       "      <td>19000.0</td>\n",
       "      <td>20602.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01361</td>\n",
       "      <td>[m54329U_200306_185930.ccs.bam, m54329U_200308...</td>\n",
       "      <td>[HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 20kb fractionated gDNA, Hi...</td>\n",
       "      <td>...</td>\n",
       "      <td>129995062425</td>\n",
       "      <td>129.99</td>\n",
       "      <td>46.2</td>\n",
       "      <td>48381.20</td>\n",
       "      <td>18101.60</td>\n",
       "      <td>19655.20</td>\n",
       "      <td>21501.200000</td>\n",
       "      <td>18408.20</td>\n",
       "      <td>20016.2</td>\n",
       "      <td>21955.600000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_ID                                           filename  \\\n",
       "0   HG01891  [m54329U_200124_193652.ccs.bam, m54329U_200127...   \n",
       "1   HG01123  [m54329U_200201_051510.ccs.bam, m54329U_200203...   \n",
       "2   HG02559  [m54329U_200211_192235.ccs.bam, m54329U_200217...   \n",
       "3   HG02486  [m54329U_200215_021808.ccs.bam, m54329U_200313...   \n",
       "4   HG01361  [m54329U_200306_185930.ccs.bam, m54329U_200308...   \n",
       "\n",
       "                                          library_ID library_strategy  \\\n",
       "0                                  HG01891.HiFiEx_f2              WGS   \n",
       "1                                  HG01123_HiFiEx_f2              WGS   \n",
       "2  [HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...              WGS   \n",
       "3  [HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...              WGS   \n",
       "4  [HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...              WGS   \n",
       "\n",
       "  library_source   library_selection library_layout     platform  \\\n",
       "0        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "1        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "2        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "3        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "4        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "\n",
       "   instrument_model                                 design_description  ...  \\\n",
       "0  PacBio Sequel II          HiFi sequencing of 18kb fractionated gDNA  ...   \n",
       "1  PacBio Sequel II          HiFi sequencing of 17kb fractionated gDNA  ...   \n",
       "2  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA  ...   \n",
       "3  PacBio Sequel II  [HiFi sequencing of 22kb fractionated gDNA, Hi...  ...   \n",
       "4  PacBio Sequel II  [HiFi sequencing of 20kb fractionated gDNA, Hi...  ...   \n",
       "\n",
       "       total_bp total_Gbp   min       max quartile_25 quartile_50  \\\n",
       "0  115126112334    115.12  46.0  44384.75    15781.75    16865.00   \n",
       "1  119417336179    119.40  46.0  43082.25    14998.75    16470.25   \n",
       "2  128540762559    128.54  46.2  47600.40    18524.80    19951.20   \n",
       "3  123904875583    123.92  46.5  45554.50    17439.50    18733.00   \n",
       "4  129995062425    129.99  46.2  48381.20    18101.60    19655.20   \n",
       "\n",
       "    quartile_75       N25      N50           N75  \n",
       "0  18096.000000  15930.00  17039.5  18287.750000  \n",
       "1  18144.000000  15302.25  16815.5  18543.000000  \n",
       "2  21644.200000  18764.80  20247.0  22029.400000  \n",
       "3  20268.833333  17658.50  19000.0  20602.333333  \n",
       "4  21501.200000  18408.20  20016.2  21955.600000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_sample_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82da8e4e-c8c6-4a3c-a918-50089bc06aad",
   "metadata": {},
   "source": [
    "## Check: Methylation Header in BAM\n",
    "\n",
    "Example commands \n",
    "```bash\n",
    "samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_211105_213234-bc1001.5mc.hifi_reads.bam \\\n",
    "\t| head -n 1 | grep \"MM:Z:C+m\"\n",
    " ```\n",
    "\n",
    "\n",
    "A couple keys here:\n",
    "\n",
    "* Identifying full s3 paths to the files you are trying to check, you sometimes have to built that path back after an aws s3 ls\n",
    "\n",
    "* Leverage samtools ability to view a file directly in s3.  I had to downgrade to samtools 1.18 for it to work for me.  \n",
    "\n",
    "* Use the -H flag on samtools view to limit view to just the header, or it will print the whole file!\n",
    "\n",
    "* Figure out what in the header indicates if methylation tags are present in the file, knowing that primrose is not the current methylation caller anymore, i don't think. you may have to check for the actual tags in the bam, per this: https://www.genomeark.org/documentation/hifi-reads-bam-tags.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a5454da7-0818-49a8-9fc2-d11d5a5a1c63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>filename</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>...</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>quartile_50</th>\n",
       "      <th>quartile_75</th>\n",
       "      <th>N25</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01891</td>\n",
       "      <td>[m54329U_200124_193652.ccs.bam, m54329U_200127...</td>\n",
       "      <td>HG01891.HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>115126112334</td>\n",
       "      <td>115.12</td>\n",
       "      <td>46.0</td>\n",
       "      <td>44384.75</td>\n",
       "      <td>15781.75</td>\n",
       "      <td>16865.00</td>\n",
       "      <td>18096.000000</td>\n",
       "      <td>15930.00</td>\n",
       "      <td>17039.5</td>\n",
       "      <td>18287.750000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01123</td>\n",
       "      <td>[m54329U_200201_051510.ccs.bam, m54329U_200203...</td>\n",
       "      <td>HG01123_HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 17kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>119417336179</td>\n",
       "      <td>119.40</td>\n",
       "      <td>46.0</td>\n",
       "      <td>43082.25</td>\n",
       "      <td>14998.75</td>\n",
       "      <td>16470.25</td>\n",
       "      <td>18144.000000</td>\n",
       "      <td>15302.25</td>\n",
       "      <td>16815.5</td>\n",
       "      <td>18543.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02559</td>\n",
       "      <td>[m54329U_200211_192235.ccs.bam, m54329U_200217...</td>\n",
       "      <td>[HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>...</td>\n",
       "      <td>128540762559</td>\n",
       "      <td>128.54</td>\n",
       "      <td>46.2</td>\n",
       "      <td>47600.40</td>\n",
       "      <td>18524.80</td>\n",
       "      <td>19951.20</td>\n",
       "      <td>21644.200000</td>\n",
       "      <td>18764.80</td>\n",
       "      <td>20247.0</td>\n",
       "      <td>22029.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02486</td>\n",
       "      <td>[m54329U_200215_021808.ccs.bam, m54329U_200313...</td>\n",
       "      <td>[HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 22kb fractionated gDNA, Hi...</td>\n",
       "      <td>...</td>\n",
       "      <td>123904875583</td>\n",
       "      <td>123.92</td>\n",
       "      <td>46.5</td>\n",
       "      <td>45554.50</td>\n",
       "      <td>17439.50</td>\n",
       "      <td>18733.00</td>\n",
       "      <td>20268.833333</td>\n",
       "      <td>17658.50</td>\n",
       "      <td>19000.0</td>\n",
       "      <td>20602.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01361</td>\n",
       "      <td>[m54329U_200306_185930.ccs.bam, m54329U_200308...</td>\n",
       "      <td>[HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 20kb fractionated gDNA, Hi...</td>\n",
       "      <td>...</td>\n",
       "      <td>129995062425</td>\n",
       "      <td>129.99</td>\n",
       "      <td>46.2</td>\n",
       "      <td>48381.20</td>\n",
       "      <td>18101.60</td>\n",
       "      <td>19655.20</td>\n",
       "      <td>21501.200000</td>\n",
       "      <td>18408.20</td>\n",
       "      <td>20016.2</td>\n",
       "      <td>21955.600000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_ID                                           filename  \\\n",
       "0   HG01891  [m54329U_200124_193652.ccs.bam, m54329U_200127...   \n",
       "1   HG01123  [m54329U_200201_051510.ccs.bam, m54329U_200203...   \n",
       "2   HG02559  [m54329U_200211_192235.ccs.bam, m54329U_200217...   \n",
       "3   HG02486  [m54329U_200215_021808.ccs.bam, m54329U_200313...   \n",
       "4   HG01361  [m54329U_200306_185930.ccs.bam, m54329U_200308...   \n",
       "\n",
       "                                          library_ID library_strategy  \\\n",
       "0                                  HG01891.HiFiEx_f2              WGS   \n",
       "1                                  HG01123_HiFiEx_f2              WGS   \n",
       "2  [HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...              WGS   \n",
       "3  [HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...              WGS   \n",
       "4  [HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...              WGS   \n",
       "\n",
       "  library_source   library_selection library_layout     platform  \\\n",
       "0        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "1        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "2        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "3        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "4        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "\n",
       "   instrument_model                                 design_description  ...  \\\n",
       "0  PacBio Sequel II          HiFi sequencing of 18kb fractionated gDNA  ...   \n",
       "1  PacBio Sequel II          HiFi sequencing of 17kb fractionated gDNA  ...   \n",
       "2  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA  ...   \n",
       "3  PacBio Sequel II  [HiFi sequencing of 22kb fractionated gDNA, Hi...  ...   \n",
       "4  PacBio Sequel II  [HiFi sequencing of 20kb fractionated gDNA, Hi...  ...   \n",
       "\n",
       "       total_bp total_Gbp   min       max quartile_25 quartile_50  \\\n",
       "0  115126112334    115.12  46.0  44384.75    15781.75    16865.00   \n",
       "1  119417336179    119.40  46.0  43082.25    14998.75    16470.25   \n",
       "2  128540762559    128.54  46.2  47600.40    18524.80    19951.20   \n",
       "3  123904875583    123.92  46.5  45554.50    17439.50    18733.00   \n",
       "4  129995062425    129.99  46.2  48381.20    18101.60    19655.20   \n",
       "\n",
       "    quartile_75       N25      N50           N75  \n",
       "0  18096.000000  15930.00  17039.5  18287.750000  \n",
       "1  18144.000000  15302.25  16815.5  18543.000000  \n",
       "2  21644.200000  18764.80  20247.0  22029.400000  \n",
       "3  20268.833333  17658.50  19000.0  20602.333333  \n",
       "4  21501.200000  18408.20  20016.2  21955.600000  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "135ac065-af89-40c9-8d8b-44372a1cef83",
   "metadata": {},
   "outputs": [],
   "source": [
    "s3_df = pd.read_csv('s3-bucket.tsv', sep='\\t', index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "577e47bd-2c49-4207-a380-f3eba082e06e",
   "metadata": {},
   "outputs": [],
   "source": [
    "s3_dict = {data[0]:data[1] for data in s3_df.values}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "64327f67-9ca4-427b-a073-cf1d1fcf65ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HG01106'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(set(hifi_sample_df['sample_ID'].tolist()))[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "68bfc3da-49fd-463f-9de7-a2e59007d793",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "122"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(hifi_sample_df['sample_ID'].tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "e1d1ffdc-9bb4-4b33-a366-ecd02ee5fa8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>filename</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>ccs_algorithm</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01891</td>\n",
       "      <td>[m54329U_200124_193652.ccs.bam, m54329U_200127...</td>\n",
       "      <td>HG01891.HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG01123</td>\n",
       "      <td>[m54329U_200201_051510.ccs.bam, m54329U_200203...</td>\n",
       "      <td>HG01123_HiFiEx_f2</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 17kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02559</td>\n",
       "      <td>[m54329U_200211_192235.ccs.bam, m54329U_200217...</td>\n",
       "      <td>[HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, nan, nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG02486</td>\n",
       "      <td>[m54329U_200215_021808.ccs.bam, m54329U_200313...</td>\n",
       "      <td>[HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 22kb fractionated gDNA, Hi...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, second library prep, second library prep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG01361</td>\n",
       "      <td>[m54329U_200306_185930.ccs.bam, m54329U_200308...</td>\n",
       "      <td>[HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>[HiFi sequencing of 20kb fractionated gDNA, Hi...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, nan, nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>HG01934</td>\n",
       "      <td>[m64136_210723_184622.hifi_reads.bam, m64136_2...</td>\n",
       "      <td>HG01934_lib1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>10.1.0.115913</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>bc1020 barcoded adapter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>HG00408</td>\n",
       "      <td>[m64136_211111_194404-bc1002.5mc.hifi_reads.ba...</td>\n",
       "      <td>HG00408_lib1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>6.0.0</td>\n",
       "      <td>P2.2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>HG02129</td>\n",
       "      <td>[m64136_220422_181133-bc1001.5mc.hifi_reads.ba...</td>\n",
       "      <td>HG02129_lib1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>6.0.0</td>\n",
       "      <td>P2.2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>NA20805</td>\n",
       "      <td>[m64136_220726_203708-bc1020.5mc.hifi_reads.ba...</td>\n",
       "      <td>NA20805_lib1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>6.0.0</td>\n",
       "      <td>P2.2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>NA20752</td>\n",
       "      <td>[m64457e_220827_152226-bc2052.5mc.hifi_reads.b...</td>\n",
       "      <td>PG20752.HFSS</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 23kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>6.3.0</td>\n",
       "      <td>P3.2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_ID                                           filename  \\\n",
       "0     HG01891  [m54329U_200124_193652.ccs.bam, m54329U_200127...   \n",
       "1     HG01123  [m54329U_200201_051510.ccs.bam, m54329U_200203...   \n",
       "2     HG02559  [m54329U_200211_192235.ccs.bam, m54329U_200217...   \n",
       "3     HG02486  [m54329U_200215_021808.ccs.bam, m54329U_200313...   \n",
       "4     HG01361  [m54329U_200306_185930.ccs.bam, m54329U_200308...   \n",
       "..        ...                                                ...   \n",
       "117   HG01934  [m64136_210723_184622.hifi_reads.bam, m64136_2...   \n",
       "118   HG00408  [m64136_211111_194404-bc1002.5mc.hifi_reads.ba...   \n",
       "119   HG02129  [m64136_220422_181133-bc1001.5mc.hifi_reads.ba...   \n",
       "120   NA20805  [m64136_220726_203708-bc1020.5mc.hifi_reads.ba...   \n",
       "121   NA20752  [m64457e_220827_152226-bc2052.5mc.hifi_reads.b...   \n",
       "\n",
       "                                            library_ID library_strategy  \\\n",
       "0                                    HG01891.HiFiEx_f2              WGS   \n",
       "1                                    HG01123_HiFiEx_f2              WGS   \n",
       "2    [HG02559.HiFiEx_b6, HG02559.HiFiEx_f2, HG02559...              WGS   \n",
       "3    [HG02486.HiFiEx_f2, HG02486.HiFiEx2_f2, HG0248...              WGS   \n",
       "4    [HG01361.HiFiEx_f2, HG01361.HiFiEx_f2, HG01361...              WGS   \n",
       "..                                                 ...              ...   \n",
       "117                                       HG01934_lib1              WGS   \n",
       "118                                       HG00408_lib1              WGS   \n",
       "119                                       HG02129_lib1              WGS   \n",
       "120                                       NA20805_lib1              WGS   \n",
       "121                                       PG20752.HFSS              WGS   \n",
       "\n",
       "    library_source   library_selection library_layout     platform  \\\n",
       "0          GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "1          GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "2          GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "3          GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "4          GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "..             ...                 ...            ...          ...   \n",
       "117        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "118        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "119        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "120        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "121        GENOMIC              RANDOM         single  PACBIO_SMRT   \n",
       "\n",
       "     instrument_model                                 design_description  \\\n",
       "0    PacBio Sequel II          HiFi sequencing of 18kb fractionated gDNA   \n",
       "1    PacBio Sequel II          HiFi sequencing of 17kb fractionated gDNA   \n",
       "2    PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA   \n",
       "3    PacBio Sequel II  [HiFi sequencing of 22kb fractionated gDNA, Hi...   \n",
       "4    PacBio Sequel II  [HiFi sequencing of 20kb fractionated gDNA, Hi...   \n",
       "..                ...                                                ...   \n",
       "117  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA   \n",
       "118  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA   \n",
       "119  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA   \n",
       "120  PacBio Sequel II          HiFi sequencing of 20kb fractionated gDNA   \n",
       "121  PacBio Sequel II          HiFi sequencing of 23kb fractionated gDNA   \n",
       "\n",
       "           data_type  shear_method size_selection  \\\n",
       "0    unaligned reads        g-TUBE        SageELF   \n",
       "1    unaligned reads        g-TUBE        SageELF   \n",
       "2    unaligned reads        g-TUBE        SageELF   \n",
       "3    unaligned reads        g-TUBE        SageELF   \n",
       "4    unaligned reads        g-TUBE        SageELF   \n",
       "..               ...           ...            ...   \n",
       "117  unaligned reads  Megaruptor 1        SageELF   \n",
       "118  unaligned reads  Megaruptor 1        SageELF   \n",
       "119  unaligned reads  Megaruptor 1        SageELF   \n",
       "120  unaligned reads  Megaruptor 1        SageELF   \n",
       "121  unaligned reads  Megaruptor 3       PippinHT   \n",
       "\n",
       "                           ccs_algorithm polymerase_version  \\\n",
       "0    ccs 4.0.0 (commit SL-release-8.0.0)               P2.0   \n",
       "1    ccs 4.0.0 (commit SL-release-8.0.0)               P2.0   \n",
       "2    ccs 4.0.0 (commit SL-release-8.0.0)               P2.0   \n",
       "3    ccs 4.0.0 (commit SL-release-8.0.0)               P2.0   \n",
       "4    ccs 4.0.0 (commit SL-release-8.0.0)               P2.0   \n",
       "..                                   ...                ...   \n",
       "117                        10.1.0.115913                 P2   \n",
       "118                                6.0.0               P2.2   \n",
       "119                                6.0.0               P2.2   \n",
       "120                                6.0.0               P2.2   \n",
       "121                                6.3.0               P3.2   \n",
       "\n",
       "    seq_plate_chemistry_version        generator_facility  generator_contact  \\\n",
       "0                          C2.0  University of Washington    kmiyamot@uw.edu   \n",
       "1                          C2.0  University of Washington    kmiyamot@uw.edu   \n",
       "2                          C2.0  University of Washington    kmiyamot@uw.edu   \n",
       "3                          C2.0  University of Washington    kmiyamot@uw.edu   \n",
       "4                          C2.0  University of Washington    kmiyamot@uw.edu   \n",
       "..                          ...                       ...                ...   \n",
       "117                          C2     Washington University  tgraves@wustl.edu   \n",
       "118                          C2     Washington University  tgraves@wustl.edu   \n",
       "119                          C2     Washington University  tgraves@wustl.edu   \n",
       "120                          C2     Washington University  tgraves@wustl.edu   \n",
       "121                          C2  University of Washington    kmiyamot@uw.edu   \n",
       "\n",
       "                                                 notes  \n",
       "0                                 [nan, nan, nan, nan]  \n",
       "1                                 [nan, nan, nan, nan]  \n",
       "2                            [nan, nan, nan, nan, nan]  \n",
       "3    [nan, second library prep, second library prep...  \n",
       "4                            [nan, nan, nan, nan, nan]  \n",
       "..                                                 ...  \n",
       "117                            bc1020 barcoded adapter  \n",
       "118                                    [nan, nan, nan]  \n",
       "119                                    [nan, nan, nan]  \n",
       "120                                    [nan, nan, nan]  \n",
       "121                                    [nan, nan, nan]  \n",
       "\n",
       "[122 rows x 19 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_sample_df[required_hifi_1_submitter_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "9e24397f-09e6-4243-a252-8f4d740b6d80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'m54329U_211105_213234-bc1001.5mc.hifi_reads.bam'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_sample_df[hifi_sample_df['sample_ID'].isin(['HG03704'])]['filename'].tolist()[0][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "6e9e87ae-3763-472b-a3c6-e7e804ec0965",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_211105_213234-bc1001.5mc.hifi_reads.bam'"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s3_dict[hifi_sample_df[hifi_sample_df['sample_ID'].isin(['HG03704'])]['filename'].tolist()[0][0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7f4d4377-328c-4083-a162-b3b08dbedb63",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HG01891\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01891/raw_data/PacBio_HiFi/m54329U_200124_193652.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01891/raw_data/PacBio_HiFi/m54329U_200127_180554.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01891/raw_data/PacBio_HiFi/m54329U_200129_001928.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01891/raw_data/PacBio_HiFi/m54329U_200130_064539.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01123\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01123/raw_data/PacBio_HiFi/m54329U_200201_051510.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01123/raw_data/PacBio_HiFi/m54329U_200203_181235.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01123/raw_data/PacBio_HiFi/m54329U_200205_002609.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01123/raw_data/PacBio_HiFi/m54329U_200206_215926.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02559\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02559/raw_data/PacBio_HiFi/m54329U_200211_192235.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02559/raw_data/PacBio_HiFi/m54329U_200217_055345.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02559/raw_data/PacBio_HiFi/m54329U_200219_181523.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02559/raw_data/PacBio_HiFi/m64076_200215_021919.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02559/raw_data/PacBio_HiFi/m64076_200217_055339.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02486\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m54329U_200215_021808.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m54329U_200313_161638.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m54329U_200315_215927.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m54329U_200317_041347.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m54329U_200320_213710.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02486/raw_data/PacBio_HiFi/m64076_200211_192227.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01361\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01361/raw_data/PacBio_HiFi_Swap_Fixed/m54329U_200306_185930.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01361/raw_data/PacBio_HiFi_Swap_Fixed/m54329U_200308_194417.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01361/raw_data/PacBio_HiFi_Swap_Fixed/m54329U_200310_015838.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01361/raw_data/PacBio_HiFi_Swap_Fixed/m54329U_200311_082605.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01361/raw_data/PacBio_HiFi_Swap_Fixed/m64076_200221_234426.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02572\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02572/raw_data/PacBio_HiFi/m54329U_200319_002813.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02572/raw_data/PacBio_HiFi/m64076_200313_161705.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02572/raw_data/PacBio_HiFi/m64076_200317_041201.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02572/raw_data/PacBio_HiFi/m64076_200318_103811.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03516\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03516/raw_data/PacBio_HiFi/m54329U_200610_234222.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03516/raw_data/PacBio_HiFi/m54329U_200612_200443.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03516/raw_data/PacBio_HiFi/m54329U_200614_021746.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03516/raw_data/PacBio_HiFi/m54329U_200615_084313.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG04204\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04204/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1003_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04204/raw_data/PacBio_HiFi/m54329U_210514_222517-bc1003_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04204/raw_data/PacBio_HiFi/m54329U_210517_210109-bc1003_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04204/raw_data/PacBio_HiFi/m54329U_210519_065152-bc1003_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04204/raw_data/PacBio_HiFi/m54329U_210520_213459-bc1003_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG04228\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04228/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1008_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04228/raw_data/PacBio_HiFi/m54329U_210725_100455.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04228/raw_data/PacBio_HiFi/m64076_210617_050849.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04228/raw_data/PacBio_HiFi/m64076_210618_160639.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04228/raw_data/PacBio_HiFi/m64076_210620_030428.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02135\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1009_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m54329U_210615_233707-bc1009_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m54329U_210715_074913.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m64076_210529_081414.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m64076_210602_215048.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02135/raw_data/PacBio_HiFi/m64076_210719_222908.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02074\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02074/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1010_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02074/raw_data/PacBio_HiFi/m54329U_210626_074001.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02074/raw_data/PacBio_HiFi/m54329U_210630_194237.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02074/raw_data/PacBio_HiFi/m64076_210630_101034.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02071\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02071/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1011_BAK8A_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02071/raw_data/PacBio_HiFi/m64076_210604_225751.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02071/raw_data/PacBio_HiFi/m64076_210607_215041.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02071/raw_data/PacBio_HiFi/m64076_210609_084637.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03942\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03942/raw_data/PacBio_HiFi/m54329U_210507_224951-bc1017_BAK8B_OA.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03942/raw_data/PacBio_HiFi/m54329U_210613_071322.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03942/raw_data/PacBio_HiFi/m64076_210606_095247.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03942/raw_data/PacBio_HiFi/m64076_210612_093800.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03942/raw_data/PacBio_HiFi/m64076_210615_181342.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03927\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03927/raw_data/PacBio_HiFi/m54329U_210522_083046.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03927/raw_data/PacBio_HiFi/m64076_210514_222349.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03927/raw_data/PacBio_HiFi/m64076_210517_210115.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03927/raw_data/PacBio_HiFi/m64076_210519_075634.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00642\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00642/raw_data/PacBio_HiFi/m54329U_210524_174516.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00642/raw_data/PacBio_HiFi/m64076_210516_091844.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00642/raw_data/PacBio_HiFi/m64076_210520_213355.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00642/raw_data/PacBio_HiFi/m64076_210522_082905.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02647\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02647/raw_data/PacBio_HiFi/m54329U_210604_200442.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02647/raw_data/PacBio_HiFi/m64076_210527_231639.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02647/raw_data/PacBio_HiFi/m64076_210530_192034.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02647/raw_data/PacBio_HiFi/m64076_210601_061523.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03688\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03688/raw_data/PacBio_HiFi/m54329U_210608_221930.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03688/raw_data/PacBio_HiFi/m54329U_210611_221457.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03688/raw_data/PacBio_HiFi/m54329U_210618_202343.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03688/raw_data/PacBio_HiFi/m64076_210610_224207.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG04199\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04199/raw_data/PacBio_HiFi/m54329U_210614_155303.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04199/raw_data/PacBio_HiFi/m54329U_210620_071842.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04199/raw_data/PacBio_HiFi/m54329U_210621_233920.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04199/raw_data/PacBio_HiFi/m64076_210613_183806.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01346\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01346/raw_data/PacBio_HiFi/m54329U_210624_221223.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01346/raw_data/PacBio_HiFi/m54329U_210629_024631.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01346/raw_data/PacBio_HiFi/m64076_210526_105450.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01346/raw_data/PacBio_HiFi/m64076_210621_234241.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02132\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02132/raw_data/PacBio_HiFi/m54329U_210627_170630.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02132/raw_data/PacBio_HiFi/m54329U_210704_040110.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02132/raw_data/PacBio_HiFi/m54329U_210705_145805.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02132/raw_data/PacBio_HiFi/m64076_210525_000003.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02300\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02300/raw_data/PacBio_HiFi/m54329U_210702_182525.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02300/raw_data/PacBio_HiFi/m54329U_210710_123004.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02300/raw_data/PacBio_HiFi/m54329U_210711_215946.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02683\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02683/raw_data/PacBio_HiFi/m54329U_210707_171722.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02683/raw_data/PacBio_HiFi/m54329U_210709_030038.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02683/raw_data/PacBio_HiFi/m64076_210703_103644.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01884\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01884/raw_data/PacBio_HiFi/m54329U_210713_205453.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01884/raw_data/PacBio_HiFi/m54329U_210716_214728.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01884/raw_data/PacBio_HiFi/m54329U_210718_084331.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02293\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02293/raw_data/PacBio_HiFi/m54329U_210719_222842.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02293/raw_data/PacBio_HiFi/m54329U_210721_092510.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02293/raw_data/PacBio_HiFi/m64076_210716_220011.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03704\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_211105_213234-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_211111_222027-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_220203_054640-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_220203_054640-bc1018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m54329U_220207_224702-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03704/raw_data/PacBio_HiFi/m64076_220216_013707-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02735\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02735/raw_data/PacBio_HiFi/m54329U_211109_002533-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02735/raw_data/PacBio_HiFi/m54329U_211204_121716-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02735/raw_data/PacBio_HiFi/m54329U_211209_115915-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02735/raw_data/PacBio_HiFi/m64076_211209_102926-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01074\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01074/raw_data/PacBio_HiFi/m54329U_211110_112322-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01074/raw_data/PacBio_HiFi/m54329U_211208_005551-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01074/raw_data/PacBio_HiFi/m54329U_211211_014820-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01074/raw_data/PacBio_HiFi/m54329U_211212_111219-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03041\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03041/raw_data/PacBio_HiFi/m54329U_211214_012740-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03041/raw_data/PacBio_HiFi/m54329U_211215_225017-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03041/raw_data/PacBio_HiFi/m54329U_211217_080751-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00639\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00639/raw_data/PacBio_HiFi/m54329U_211222_104516-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00639/raw_data/PacBio_HiFi/m54329U_211230_014258-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00639/raw_data/PacBio_HiFi/m54329U_211231_123903-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00639/raw_data/PacBio_HiFi/m64076_220216_013707-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01081\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01081/raw_data/PacBio_HiFi/m54329U_211223_214216-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01081/raw_data/PacBio_HiFi/m54329U_220101_220852-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01081/raw_data/PacBio_HiFi/m54329U_220103_072943-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03017\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03017/raw_data/PacBio_HiFi/m54329U_211225_070909-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03017/raw_data/PacBio_HiFi/m54329U_220109_103416-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03017/raw_data/PacBio_HiFi/m54329U_220117_123334-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03017/raw_data/PacBio_HiFi/m64076_220216_013707-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03017/raw_data/PacBio_HiFi/m64076_221001_041132-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03239\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03239/raw_data/PacBio_HiFi/m54329U_220105_011819-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03239/raw_data/PacBio_HiFi/m64076_220106_231243-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03239/raw_data/PacBio_HiFi/m64076_220108_100904-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00558\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00558/raw_data/PacBio_HiFi/m54329U_220107_233847-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00558/raw_data/PacBio_HiFi/m54329U_220114_143904-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00558/raw_data/PacBio_HiFi/m54329U_220116_013607-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00558/raw_data/PacBio_HiFi/m64076_220216_013707-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03139\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03139/raw_data/PacBio_HiFi/m54329U_220519_225150-bc2019.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03139/raw_data/PacBio_HiFi/m54329U_220523_155616-bc2019.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03139/raw_data/PacBio_HiFi/m54329U_220525_111437-bc2019.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA18522\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18522/raw_data/PacBio_HiFi/m54329U_220604_002013-bc2021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18522/raw_data/PacBio_HiFi/m54329U_220605_194957-bc2021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18522/raw_data/PacBio_HiFi/m54329U_220607_194753-bc2021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02922\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m54329U_220816_182601-bc2045.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m54329U_220903_190900-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m54329U_220905_144015-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m54329U_220907_100858-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m64457e_220902_165459-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m64457e_220904_135104-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02922/raw_data/PacBio_HiFi/m64457e_220906_104717-bc2053.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA18747\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18747/raw_data/PacBio_HiFi/m54329U_220823_191353-bc2011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18747/raw_data/PacBio_HiFi/m64076_220826_143529-bc2049.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18747/raw_data/PacBio_HiFi/m64076_220828_113336-bc2049.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18747/raw_data/PacBio_HiFi/m64076_221001_041132-bc2011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00099\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00099/raw_data/PacBio_HiFi/m54329U_220825_174247-bc2012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00099/raw_data/PacBio_HiFi/m54329U_220827_143814-bc2050.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00099/raw_data/PacBio_HiFi/m54329U_220829_095708-bc2050.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00280\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00280/raw_data/PacBio_HiFi/m54329U_220901_221341-bc2051.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00280/raw_data/PacBio_HiFi/m64076_220831_191646-bc2051.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00280/raw_data/PacBio_HiFi/m64457e_220825_195635-bc2051.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA18983\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m54329U_220903_190900-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m54329U_220905_144015-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m54329U_220907_100858-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m64457e_220902_165459-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m64457e_220904_135104-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18983/raw_data/PacBio_HiFi/m64457e_220906_104717-bc2055.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA19043\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA19043/raw_data/PacBio_HiFi/m54329U_220916_162216-bc2061.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA19043/raw_data/PacBio_HiFi/m54329U_220918_114340-bc2061.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA19043/raw_data/PacBio_HiFi/m54329U_220920_065703-bc2061.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02717\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02717/raw_data/PacBio_HiFi/m64043_200403_163826.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02717/raw_data/PacBio_HiFi/m64043_200405_180950.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02717/raw_data/PacBio_HiFi/m64043_200407_002219.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02717/raw_data/PacBio_HiFi/m64043_200408_064651.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02886\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02886/raw_data/PacBio_HiFi/m64043_200410_214826.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02886/raw_data/PacBio_HiFi/m64043_200412_040054.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02886/raw_data/PacBio_HiFi/m64043_200413_102554.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02886/raw_data/PacBio_HiFi/m64043_200414_165107.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02630\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02630/raw_data/PacBio_HiFi/m64043_200501_162248.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02630/raw_data/PacBio_HiFi/m64043_200502_223511.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02630/raw_data/PacBio_HiFi/m64043_200504_050026.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02630/raw_data/PacBio_HiFi/m64043_200505_112554.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03453\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03453/raw_data/PacBio_HiFi/m64043_200508_172634.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03453/raw_data/PacBio_HiFi/m64043_200509_233929.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03453/raw_data/PacBio_HiFi/m64043_200511_060458.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03453/raw_data/PacBio_HiFi/m64043_200512_123035.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03579\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03579/raw_data/PacBio_HiFi/m64043_200515_165406.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03579/raw_data/PacBio_HiFi/m64043_200516_230634.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03579/raw_data/PacBio_HiFi/m64043_200518_053124.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03579/raw_data/PacBio_HiFi/m64043_200519_115622.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03540\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03540/raw_data/PacBio_HiFi/m64043_200521_171703.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03540/raw_data/PacBio_HiFi/m64043_200522_232930.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03540/raw_data/PacBio_HiFi/m64043_200524_055430.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03540/raw_data/PacBio_HiFi/m64043_200525_121851.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02622\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02622/raw_data/PacBio_HiFi/m64043_200530_164723.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02622/raw_data/PacBio_HiFi/m64043_200601_191521.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02622/raw_data/PacBio_HiFi/m64043_200603_012738.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02622/raw_data/PacBio_HiFi/m64043_200604_075218.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03471\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03471/raw_data/PacBio_HiFi/m64043_200606_175618.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03471/raw_data/PacBio_HiFi/m64043_200608_185843.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03471/raw_data/PacBio_HiFi/m64043_200610_011156.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03471/raw_data/PacBio_HiFi/m64043_200611_073649.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01928\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01928/raw_data/PacBio_HiFi/m64043_200612_200936.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01928/raw_data/PacBio_HiFi/m64043_200614_191756.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01928/raw_data/PacBio_HiFi/m64043_200616_013031.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01928/raw_data/PacBio_HiFi/m64043_200617_075530.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01175\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01175/raw_data/PacBio_HiFi/m64043_200618_201934.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01175/raw_data/PacBio_HiFi/m64043_200620_173220.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01175/raw_data/PacBio_HiFi/m64043_200621_234442.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01175/raw_data/PacBio_HiFi/m64043_200623_060946.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01106\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01106/raw_data/PacBio_HiFi/m64043_200625_174853.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01106/raw_data/PacBio_HiFi/m64043_200627_000137.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01106/raw_data/PacBio_HiFi/m64043_200628_062711.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01106/raw_data/PacBio_HiFi/m64043_200629_125238.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00735\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00735/raw_data/PacBio_HiFi/m64043_200702_173033.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00735/raw_data/PacBio_HiFi/m64043_200703_234328.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00735/raw_data/PacBio_HiFi/m64043_200705_060840.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00735/raw_data/PacBio_HiFi/m64043_200706_123423.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00438\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00438/raw_data/PacBio_HiFi/m64043_200710_174426.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00438/raw_data/PacBio_HiFi/m64043_200711_235708.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00438/raw_data/PacBio_HiFi/m64043_200713_062240.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00438/raw_data/PacBio_HiFi/m64043_200714_124814.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00673\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00673/raw_data/PacBio_HiFi/m64043_200716_182902.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00673/raw_data/PacBio_HiFi/m64043_200718_004213.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00673/raw_data/PacBio_HiFi/m64043_200719_070806.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00673/raw_data/PacBio_HiFi/m64043_200720_133355.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00423\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00423/raw_data/PacBio_HiFi/m64043_210205_190424.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00423/raw_data/PacBio_HiFi/m64043_210207_011920.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00423/raw_data/PacBio_HiFi/m64043_210209_184051.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00423/raw_data/PacBio_HiFi/m64043_210211_005516.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02523\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02523/raw_data/PacBio_HiFi/m64043_210506_183430.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02523/raw_data/PacBio_HiFi/m64043_210508_033135.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02523/raw_data/PacBio_HiFi/m64043_210509_110105.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03710\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03710/raw_data/PacBio_HiFi/m64043_210514_184131.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03710/raw_data/PacBio_HiFi/m64043_210516_020821.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03710/raw_data/PacBio_HiFi/m64043_210517_175123.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02698\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02698/raw_data/PacBio_HiFi/m64043_210519_013026.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02698/raw_data/PacBio_HiFi/m64136_210514_184433.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02698/raw_data/PacBio_HiFi/m64136_210517_184202.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01255\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01255/raw_data/PacBio_HiFi/m64043_210520_180149.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01255/raw_data/PacBio_HiFi/m64136_210516_022225.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01255/raw_data/PacBio_HiFi/m64136_210519_033858.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03654\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03654/raw_data/PacBio_HiFi/m64043_210522_023949.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03654/raw_data/PacBio_HiFi/m64043_210523_101851.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03654/raw_data/PacBio_HiFi/m64043_210524_174752.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03654/raw_data/PacBio_HiFi/m64043_210528_171552.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00738\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00738/raw_data/PacBio_HiFi/m64043_210530_003337.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00738/raw_data/PacBio_HiFi/m64043_210531_080529.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00738/raw_data/PacBio_HiFi/m64043_210601_160048.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00738/raw_data/PacBio_HiFi/m64136_210602_151340.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03669\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03669/raw_data/PacBio_HiFi/m64043_210604_175624.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03669/raw_data/PacBio_HiFi/m64043_210606_013256.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03669/raw_data/PacBio_HiFi/m64043_210607_090252.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03669/raw_data/PacBio_HiFi/m64043_210610_203329.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02602\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02602/raw_data/PacBio_HiFi/m64043_210612_053030.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02602/raw_data/PacBio_HiFi/m64043_210613_133026.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02602/raw_data/PacBio_HiFi/m64043_210614_223013.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02602/raw_data/PacBio_HiFi/m64043_210618_184604.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01993\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01993/raw_data/PacBio_HiFi/m64043_210620_041711.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01993/raw_data/PacBio_HiFi/m64043_210622_163331.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01993/raw_data/PacBio_HiFi/m64043_210624_032817.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01496\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01496/raw_data/PacBio_HiFi/m64043_210625_184443.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01496/raw_data/PacBio_HiFi/m64043_210627_054642.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01496/raw_data/PacBio_HiFi/m64043_210628_151651.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00544\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00544/raw_data/PacBio_HiFi/m64043_210702_183246.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00544/raw_data/PacBio_HiFi/m64136_210520_180433.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00544/raw_data/PacBio_HiFi/m64136_210522_014758.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00544/raw_data/PacBio_HiFi/m64136_210523_100337.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00544/raw_data/PacBio_HiFi/m64136_210525_165246.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01099\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01099/raw_data/PacBio_HiFi/m64043_210704_052805.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01099/raw_data/PacBio_HiFi/m64043_210705_162656.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01099/raw_data/PacBio_HiFi/m64043_210707_032602.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02004\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02004/raw_data/PacBio_HiFi/m64043_210709_205614.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02004/raw_data/PacBio_HiFi/m64136_210702_183526.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02004/raw_data/PacBio_HiFi/m64136_210704_053041.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02004/raw_data/PacBio_HiFi/m64136_210705_162724.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01943\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01943/raw_data/PacBio_HiFi/m64043_210716_230222.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01943/raw_data/PacBio_HiFi/m64043_210718_100009.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01943/raw_data/PacBio_HiFi/m64043_210719_210019.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02809\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02809/raw_data/PacBio_HiFi/m64043_210723_184340.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02809/raw_data/PacBio_HiFi/m64043_210725_053930.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02809/raw_data/PacBio_HiFi/m64043_210726_163819.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02809/raw_data/PacBio_HiFi/m64043_210728_163353.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02015\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02015/raw_data/PacBio_HiFi/m64043_210901_164337-bc1022.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02015/raw_data/PacBio_HiFi/m64043_210903_163032-bc1022.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02015/raw_data/PacBio_HiFi/m64043_210905_032603-bc1022.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01975\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01975/raw_data/PacBio_HiFi/m64043_211113_003158-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01975/raw_data/PacBio_HiFi/m64043_211114_112934-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01975/raw_data/PacBio_HiFi/m64043_211115_222749-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01975/raw_data/PacBio_HiFi/m64043_211222_184851-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "MGISTL_PAN027_HG06807\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG06807/raw_data/PacBio_HiFi/m64043_211118_185250-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG06807/raw_data/PacBio_HiFi/m64043_211120_054831-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG06807/raw_data/PacBio_HiFi/m64043_211121_164646-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG06807/raw_data/PacBio_HiFi/m64043_211222_184851-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01192\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01192/raw_data/PacBio_HiFi/m64043_211124_192057.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01192/raw_data/PacBio_HiFi/m64043_211126_061537.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01192/raw_data/PacBio_HiFi/m64043_211127_171224.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00597\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00597/raw_data/PacBio_HiFi/m64043_211210_180342-bc1021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00597/raw_data/PacBio_HiFi/m64043_211212_045901-bc1021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00597/raw_data/PacBio_HiFi/m64043_211213_155654-bc1021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00597/raw_data/PacBio_HiFi/m64043_211222_184851-bc1021.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02258\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02258/raw_data/PacBio_HiFi/m64043_211217_180919-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02258/raw_data/PacBio_HiFi/m64043_211219_050421-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02258/raw_data/PacBio_HiFi/m64043_211220_160211-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02258/raw_data/PacBio_HiFi/m64043_220114_181052-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01261\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01261/raw_data/PacBio_HiFi/m64043_211222_184851-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01261/raw_data/PacBio_HiFi/m64136_211105_181124-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01261/raw_data/PacBio_HiFi/m64136_211107_050725-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01261/raw_data/PacBio_HiFi/m64136_211108_160613-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03834\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03834/raw_data/PacBio_HiFi/m64043_220107_191226-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03834/raw_data/PacBio_HiFi/m64043_220109_060731-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03834/raw_data/PacBio_HiFi/m64043_220110_170603-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02056\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02056/raw_data/PacBio_HiFi/m64043_220114_181052-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02056/raw_data/PacBio_HiFi/m64136_211118_185339-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02056/raw_data/PacBio_HiFi/m64136_211120_043838-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02056/raw_data/PacBio_HiFi/m64136_211121_153454-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02273\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02273/raw_data/PacBio_HiFi/m64043_220429_171542-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02273/raw_data/PacBio_HiFi/m64043_220501_041051-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02273/raw_data/PacBio_HiFi/m64043_220503_153752-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03225\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03225/raw_data/PacBio_HiFi/m64043_220516_150157-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03225/raw_data/PacBio_HiFi/m64043_220518_120044-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03225/raw_data/PacBio_HiFi/m64043_220726_203720-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03225/raw_data/PacBio_HiFi/m64136_220518_004520-bc1008.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02976\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02976/raw_data/PacBio_HiFi/m64043_220617_195135-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02976/raw_data/PacBio_HiFi/m64043_220619_164635-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02976/raw_data/PacBio_HiFi/m64043_220621_134351-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02976/raw_data/PacBio_HiFi/m64043_220726_203720-bc1009.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA18971\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18971/raw_data/PacBio_HiFi/m64043_220701_181144-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18971/raw_data/PacBio_HiFi/m64043_220703_150718-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18971/raw_data/PacBio_HiFi/m64043_220705_120954-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18971/raw_data/PacBio_HiFi/m64043_220728_173215-bc1011.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA18612\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18612/raw_data/PacBio_HiFi/m64043_220708_205637-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18612/raw_data/PacBio_HiFi/m64043_220710_175250-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA18612/raw_data/PacBio_HiFi/m64043_220712_145025-bc1015.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02965\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02965/raw_data/PacBio_HiFi/m64043_220715_182700-bc1017.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02965/raw_data/PacBio_HiFi/m64043_220717_152237-bc1017.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02965/raw_data/PacBio_HiFi/m64043_220719_122051-bc1017.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03195\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03195/raw_data/PacBio_HiFi/m64043_220726_203720-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03195/raw_data/PacBio_HiFi/m64136_220617_195203-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03195/raw_data/PacBio_HiFi/m64136_220619_164654-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03195/raw_data/PacBio_HiFi/m64136_220621_134558-bc1010.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02155\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02155/raw_data/PacBio_HiFi/m64043_220728_173215-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02155/raw_data/PacBio_HiFi/m64136_220701_181202-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02155/raw_data/PacBio_HiFi/m64136_220703_150713-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02155/raw_data/PacBio_HiFi/m64136_220705_120453-bc1012.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00323\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00323/raw_data/PacBio_HiFi/m64043_220728_173215-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00323/raw_data/PacBio_HiFi/m64136_220708_205611-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00323/raw_data/PacBio_HiFi/m64136_220710_175208-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00323/raw_data/PacBio_HiFi/m64136_220712_144943-bc1016.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00140\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00140/raw_data/PacBio_HiFi/m64043_220728_173215-bc1018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00140/raw_data/PacBio_HiFi/m64136_220715_182717-bc1018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00140/raw_data/PacBio_HiFi/m64136_220717_152248-bc1018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00140/raw_data/PacBio_HiFi/m64136_220719_122056-bc1018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02257\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02257/raw_data/PacBio_HiFi/m64076_200125_231256.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02257/raw_data/PacBio_HiFi/m64076_200127_180545.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02257/raw_data/PacBio_HiFi/m64076_200129_001835.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02257/raw_data/PacBio_HiFi/m64076_200130_064345.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01358\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01358/raw_data/PacBio_HiFi/m64076_200201_051547.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01358/raw_data/PacBio_HiFi/m64076_200203_181219.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01358/raw_data/PacBio_HiFi/m64076_200206_215943.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01358/raw_data/PacBio_HiFi/m64076_200208_041234.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01258\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01258/raw_data/PacBio_HiFi_Swap_Fixed/m64076_200306_185917.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01258/raw_data/PacBio_HiFi_Swap_Fixed/m64076_200308_194406.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01258/raw_data/PacBio_HiFi_Swap_Fixed/m64076_200310_015720.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01258/raw_data/PacBio_HiFi_Swap_Fixed/m64076_200311_082315.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG002\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG002/raw_data/PacBio_HiFi/UW_supplement/m64076_201013_225902.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC_PLUS/HG002/raw_data/PacBio_HiFi/UW_supplement/m64076_201016_191536.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG04115\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04115/raw_data/PacBio_HiFi/m64076_210623_104349.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04115/raw_data/PacBio_HiFi/m64076_210625_220013.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04115/raw_data/PacBio_HiFi/m64076_210627_085544.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG04184\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04184/raw_data/PacBio_HiFi/m64076_210701_234028.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04184/raw_data/PacBio_HiFi/m64076_210708_084125.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04184/raw_data/PacBio_HiFi/m64076_210709_232529.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG04184/raw_data/PacBio_HiFi/m64076_210712_194121.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02738\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02738/raw_data/PacBio_HiFi/m64076_210706_214544.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02738/raw_data/PacBio_HiFi/m64076_210711_102225.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02738/raw_data/PacBio_HiFi/m64076_210714_200553.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02738/raw_data/PacBio_HiFi/m64076_210718_085619.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02451\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02451/raw_data/PacBio_HiFi/m64076_211109_232223-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02451/raw_data/PacBio_HiFi/m64076_211208_005550-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02451/raw_data/PacBio_HiFi/m64076_211211_014755-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02451/raw_data/PacBio_HiFi/m64076_211212_112351-bc1003.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03130\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03130/raw_data/PacBio_HiFi/m64076_220513_215716-bc2018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03130/raw_data/PacBio_HiFi/m64076_220516_221911-bc2018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03130/raw_data/PacBio_HiFi/m64076_220518_191414-bc2018.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03209\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03209/raw_data/PacBio_HiFi/m64076_220520_234906-bc2020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03209/raw_data/PacBio_HiFi/m64076_220524_161808-bc2020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03209/raw_data/PacBio_HiFi/m64076_220526_115049-bc2020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03209/raw_data/PacBio_HiFi/m64076_221001_041132-bc2020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02165\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02165/raw_data/PacBio_HiFi/m64076_220902_143723-bc2054.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02165/raw_data/PacBio_HiFi/m64076_220904_113359-bc2054.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02165/raw_data/PacBio_HiFi/m64076_220906_083112-bc2054.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02040\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02040/raw_data/PacBio_HiFi/m64076_220908_204946-bc2056.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02040/raw_data/PacBio_HiFi/m64076_220910_174812-bc2056.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02040/raw_data/PacBio_HiFi/m64076_220912_140850-bc2056.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA20905\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20905/raw_data/PacBio_HiFi/m64076_220914_214248-bc2060.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20905/raw_data/PacBio_HiFi/m64076_220916_183810-bc2060.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20905/raw_data/PacBio_HiFi/m64076_220918_143117-bc2060.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01978\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01978/raw_data/PacBio_HiFi/m64136_200530_164818.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01978/raw_data/PacBio_HiFi/m64136_200602_153012.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01978/raw_data/PacBio_HiFi/m64136_200603_214308.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01978/raw_data/PacBio_HiFi/m64136_200605_040848.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01952\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01952/raw_data/PacBio_HiFi/m64136_200612_201033.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01952/raw_data/PacBio_HiFi/m64136_200614_192134.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01952/raw_data/PacBio_HiFi/m64136_200616_013426.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01952/raw_data/PacBio_HiFi/m64136_200617_080019.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02148\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02148/raw_data/PacBio_HiFi/m64136_200618_202033.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02148/raw_data/PacBio_HiFi/m64136_200620_173618.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02148/raw_data/PacBio_HiFi/m64136_200621_234916.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02148/raw_data/PacBio_HiFi/m64136_200623_061529.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00741\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00741/raw_data/PacBio_HiFi/m64136_200625_174949.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00741/raw_data/PacBio_HiFi/m64136_200627_000247.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00741/raw_data/PacBio_HiFi/m64136_200628_062837.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00741/raw_data/PacBio_HiFi/m64136_200629_125431.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01071\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01071/raw_data/PacBio_HiFi/m64136_200702_173125.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01071/raw_data/PacBio_HiFi/m64136_200703_234438.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01071/raw_data/PacBio_HiFi/m64136_200705_061033.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01071/raw_data/PacBio_HiFi/m64136_200706_123635.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00621\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00621/raw_data/PacBio_HiFi/m64136_200710_174522.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00621/raw_data/PacBio_HiFi/m64136_200711_235843.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00621/raw_data/PacBio_HiFi/m64136_200713_062514.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00621/raw_data/PacBio_HiFi/m64136_200714_125149.ccs.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02027\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02027/raw_data/PacBio_HiFi/m64136_210205_190622.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02027/raw_data/PacBio_HiFi/m64136_210207_012053.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02027/raw_data/PacBio_HiFi/m64136_210209_184241.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02027/raw_data/PacBio_HiFi/m64136_210211_010151.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02083\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02083/raw_data/PacBio_HiFi/m64136_210317_200525.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02083/raw_data/PacBio_HiFi/m64136_210319_191015.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02083/raw_data/PacBio_HiFi/m64136_210321_022417.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02083/raw_data/PacBio_HiFi/m64136_210323_162524.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02615\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02615/raw_data/PacBio_HiFi/m64136_210528_171842.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02615/raw_data/PacBio_HiFi/m64136_210530_004827.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02615/raw_data/PacBio_HiFi/m64136_210531_083342.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01433\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01433/raw_data/PacBio_HiFi/m64136_210604_175856.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01433/raw_data/PacBio_HiFi/m64136_210606_025622.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01433/raw_data/PacBio_HiFi/m64136_210607_115559.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01433/raw_data/PacBio_HiFi/m64136_210610_203618.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG03831\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03831/raw_data/PacBio_HiFi/m64136_210612_053350.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03831/raw_data/PacBio_HiFi/m64136_210613_130744.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03831/raw_data/PacBio_HiFi/m64136_210614_204240.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG03831/raw_data/PacBio_HiFi/m64136_210618_184849.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00609\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00609/raw_data/PacBio_HiFi/m64136_210620_054327.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00609/raw_data/PacBio_HiFi/m64136_210622_163607.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00609/raw_data/PacBio_HiFi/m64136_210624_033152.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01981\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01981/raw_data/PacBio_HiFi/m64136_210625_184731.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01981/raw_data/PacBio_HiFi/m64136_210627_054246.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01981/raw_data/PacBio_HiFi/m64136_210628_163953.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02280\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02280/raw_data/PacBio_HiFi/m64136_210716_230343.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02280/raw_data/PacBio_HiFi/m64136_210718_091611.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02280/raw_data/PacBio_HiFi/m64136_210719_201505.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG01934\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01934/raw_data/PacBio_HiFi/m64136_210723_184622.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01934/raw_data/PacBio_HiFi/m64136_210725_054129.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01934/raw_data/PacBio_HiFi/m64136_210726_163800.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG01934/raw_data/PacBio_HiFi/m64136_210728_163638.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG00408\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00408/raw_data/PacBio_HiFi/m64136_211111_194404-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00408/raw_data/PacBio_HiFi/m64136_211113_063830-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG00408/raw_data/PacBio_HiFi/m64136_211114_164241-bc1002.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "HG02129\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02129/raw_data/PacBio_HiFi/m64136_220422_181133-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02129/raw_data/PacBio_HiFi/m64136_220424_050632-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/HG02129/raw_data/PacBio_HiFi/m64136_220426_155627-bc1001.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA20805\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20805/raw_data/PacBio_HiFi/m64136_220726_203708-bc1020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20805/raw_data/PacBio_HiFi/m64136_220728_160902-bc1020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20805/raw_data/PacBio_HiFi/m64136_220730_113921-bc1020.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "NA20752\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20752/raw_data/PacBio_HiFi/m64457e_220827_152226-bc2052.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20752/raw_data/PacBio_HiFi/m64457e_220829_104034-bc2052.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "/private/home/iviolich/tools/samtools-1.18/samtools view s3://human-pangenomics/working/HPRC/NA20752/raw_data/PacBio_HiFi/m64457e_220831_195843-bc2052.5mc.hifi_reads.bam | head -n 10 | tr '\\t' '\\n' | less -S | grep 'MM:Z:C+m' | wc -l\n",
      "\n",
      "\n",
      "465\n"
     ]
    }
   ],
   "source": [
    "bam_list = []\n",
    "for sample_data in hifi_sample_df[['sample_ID', 'filename']].values:\n",
    "    print(sample_data[0])\n",
    "    bam_list.append(len(sample_data[1]))\n",
    "    for file in sample_data[1]:\n",
    "        print(\n",
    "\n",
    "        # -H $i 2>/dev/null | grep -q primrose\n",
    "        # sam_cmd_1 = '/private/home/iviolich/tools/samtools-1.18/samtools view ' + s3_dict[file] + ' | head -n 1 | grep \"MM:Z:C+m\"'\n",
    "        \n",
    "        sam_cmd_1 = '/private/home/iviolich/tools/samtools-1.18/samtools view -H ' + s3_dict[file] + \" | head -n 10 | grep 'MM:Z:C+m' | wc -l\"  # update toe boolean\n",
    "        sam_cmd_2 = '/private/home/iviolich/tools/samtools-1.18/samtools view -H ' + s3_dict[file] + \" | grep 'primrose' | wc -l\"\n",
    "        # boolean\n",
    "        # check 50\n",
    "\n",
    "        # print(sam_cmd_1)\n",
    "        print(sam_cmd_2)\n",
    "    print('\\n')\n",
    "\n",
    "print(sum(bam_list))\n",
    "\n",
    "# samtools view some_file.bam | head -n 1 | tr '\\t' '\\n' | less -S"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5a4b645-6bc4-4094-881d-d83112a51c5b",
   "metadata": {},
   "source": [
    "# Deep Consensus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "3ea8e4ee-9519-4415-a349-4d8586ccea2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>accession</th>\n",
       "      <th>study</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>...</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2687625</td>\n",
       "      <td>56102007798</td>\n",
       "      <td>56.10</td>\n",
       "      <td>107</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00099.m54329U_220827_143814.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2643186</td>\n",
       "      <td>52918724515</td>\n",
       "      <td>52.92</td>\n",
       "      <td>345</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099.m54329U_220829_095708.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2866917</td>\n",
       "      <td>57539149577</td>\n",
       "      <td>57.54</td>\n",
       "      <td>138</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00140/ra...</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33621941</td>\n",
       "      <td>572695</td>\n",
       "      <td>11345594738</td>\n",
       "      <td>11.35</td>\n",
       "      <td>117</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00140.m64136_220715_182717.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00140/ra...</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33621941</td>\n",
       "      <td>1674999</td>\n",
       "      <td>36918329655</td>\n",
       "      <td>36.92</td>\n",
       "      <td>400</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename sample_ID  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz   HG00099   \n",
       "1  HG00099.m54329U_220827_143814.dc.q20.fastq.gz   HG00099   \n",
       "2  HG00099.m54329U_220829_095708.dc.q20.fastq.gz   HG00099   \n",
       "3   HG00140.m64043_220728_173215.dc.q20.fastq.gz   HG00140   \n",
       "4   HG00140.m64136_220715_182717.dc.q20.fastq.gz   HG00140   \n",
       "\n",
       "                                                path    accession      study  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "1  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "2  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "3  s3://human-pangenomics/working/HPRC/HG00140/ra...  SRR26545346  SRP305758   \n",
       "4  s3://human-pangenomics/working/HPRC/HG00140/ra...  SRR26545346  SRP305758   \n",
       "\n",
       "  biosample_accession  total_reads     total_bp  total_Gbp  min  ...  \\\n",
       "0        SAMN33758778      2687625  56102007798      56.10  107  ...   \n",
       "1        SAMN33758778      2643186  52918724515      52.92  345  ...   \n",
       "2        SAMN33758778      2866917  57539149577      57.54  138  ...   \n",
       "3        SAMN33621941       572695  11345594738      11.35  117  ...   \n",
       "4        SAMN33621941      1674999  36918329655      36.92  400  ...   \n",
       "\n",
       "                                  design_description        data_type  \\\n",
       "0  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "1  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "2  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "3  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "4  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "\n",
       "   shear_method  size_selection  DeepConsensus_version  polymerase_version  \\\n",
       "0  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "1  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "2  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "3  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "4  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "\n",
       "   seq_plate_chemistry_version        generator_facility generator_contact  \\\n",
       "0                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "1                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "2                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "3                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "4                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "\n",
       "  notes  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dc_df = pd.read_csv('hprc_metadata_sample_files_DEEPCONSENSUS.tsv', sep='\\t')\n",
    "dc_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "82d21b80-2594-4ee5-8309-6d17420847c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if dc_df['notes'].isnull().all() == True:\n",
    "    dc_df = dc_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "e506f5fd-e106-4919-a6bd-7b020099e79f",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_deepconsensus_submitter_columns = ['filename', 'sample_ID', 'library_ID', 'library_strategy', 'library_source', \n",
    "                                            'library_selection', 'library_layout', 'platform', 'instrument_model', 'design_description', \n",
    "                                            'data_type', 'shear_method', 'size_selection', \n",
    "                                            'DeepConsensus_version', 'polymerase_version', \n",
    "                                            'seq_plate_chemistry_version', 'generator_facility', \n",
    "                                            'generator_contact']\n",
    "\n",
    "required_deepconsensus_readstats_columns = ['sample_ID', 'filename', 'total_reads', 'total_bp', 'total_Gbp', 'min', 'max', 'mean', 'quartile_25', 'quartile_50', 'quartile_75', 'N25', 'N50', 'N75']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c3b9c3d4-512f-45ca-8551-349e721ae16b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_meta_df = dc_df[required_deepconsensus_submitter_columns].drop_duplicates(subset=['sample_ID']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "7073f70c-fe9d-4963-b1b7-f3cdda60d466",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dc_sample_meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "b4f0fc95-5912-4de5-b2a3-2d8fc2f761df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "dc_sample_missing_1_submitter_list = []\n",
    "for sample in dc_sample_meta_df['sample_ID'].tolist():\n",
    "    if sum(dc_sample_meta_df[dc_sample_meta_df['sample_ID'].isin([sample])].isna().any()) == 0:\n",
    "        pass\n",
    "    else:\n",
    "        dc_sample_missing_1_submitter_df = dc_sample_meta_df[dc_sample_meta_df['sample_ID'].isin([sample])]\n",
    "        if dc_sample_missing_1_submitter_df.columns[dc_sample_missing_1_submitter_df.isna().any()].tolist() == ['notes']:\n",
    "            pass\n",
    "        else:\n",
    "            # print(sample, hifi_sample_missing_1_submitter_df.columns[hifi_sample_missing_1_submitter_df.isna().any()].tolist())\n",
    "            dc_sample_missing_1_submitter_list.append(sample)\n",
    "print(len(dc_sample_missing_1_submitter_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "6a37b0f3-bf96-478e-8727-821bac82cfe6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check all dc sample 5_readstats data is present\n",
    "assert sum(dc_df[required_deepconsensus_readstats_columns].isna().any()) == 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "6fda5605-7dce-4722-a8a2-f51db7670e35",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_readstats_df = dc_df[required_deepconsensus_readstats_columns]\n",
    "\n",
    "dc_sample_readstats_df = dc_sample_readstats_df.groupby('sample_ID').agg({'total_reads': 'sum',\n",
    "                                               'total_bp': 'sum',\n",
    "                                               'total_Gbp': 'sum',\n",
    "                                               'min': 'mean',\n",
    "                                               'max': 'mean',\n",
    "                                               'quartile_25': 'mean',\n",
    "                                               'quartile_50': 'mean',\n",
    "                                               'quartile_75': 'mean',\n",
    "                                               'N25': 'mean',\n",
    "                                               'N50': 'mean',\n",
    "                                               'N75': 'mean'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "4cdde921-3c6e-47b6-9b03-1944974f03fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_df = pd.merge(dc_sample_meta_df, dc_sample_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "319f745f-bf29-4a70-a94d-54a372a847d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_df.to_csv('hprc_metadata_sample_aggregate_DEEPCONSENSUS.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "a4548183-d841-4766-8bb1-ff1bb1fea6c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>...</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>quartile_50</th>\n",
       "      <th>quartile_75</th>\n",
       "      <th>N25</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PG00099_1.HFSS_dc</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>...</td>\n",
       "      <td>166559881890</td>\n",
       "      <td>166.56</td>\n",
       "      <td>196.666667</td>\n",
       "      <td>62248.666667</td>\n",
       "      <td>17449.333333</td>\n",
       "      <td>19636.333333</td>\n",
       "      <td>22651.666667</td>\n",
       "      <td>17950.333333</td>\n",
       "      <td>20409.333333</td>\n",
       "      <td>23637.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>HG00140_lib1_dc</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>...</td>\n",
       "      <td>126611334877</td>\n",
       "      <td>126.62</td>\n",
       "      <td>272.500000</td>\n",
       "      <td>67774.500000</td>\n",
       "      <td>18393.250000</td>\n",
       "      <td>20448.500000</td>\n",
       "      <td>23623.250000</td>\n",
       "      <td>18873.500000</td>\n",
       "      <td>21263.000000</td>\n",
       "      <td>24996.250000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00280.m54329U_220901_221341.dc.q20.fastq.gz</td>\n",
       "      <td>HG00280</td>\n",
       "      <td>PG00280.HFSS_dc</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>...</td>\n",
       "      <td>151564344052</td>\n",
       "      <td>151.56</td>\n",
       "      <td>95.000000</td>\n",
       "      <td>64738.666667</td>\n",
       "      <td>17189.666667</td>\n",
       "      <td>19452.666667</td>\n",
       "      <td>22546.000000</td>\n",
       "      <td>17712.333333</td>\n",
       "      <td>20266.666667</td>\n",
       "      <td>23568.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00323.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>HG00323</td>\n",
       "      <td>HG00323_lib1_dc</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>...</td>\n",
       "      <td>120027730827</td>\n",
       "      <td>120.03</td>\n",
       "      <td>368.750000</td>\n",
       "      <td>60432.750000</td>\n",
       "      <td>16941.500000</td>\n",
       "      <td>19105.500000</td>\n",
       "      <td>21983.250000</td>\n",
       "      <td>17417.000000</td>\n",
       "      <td>19893.750000</td>\n",
       "      <td>22834.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00408.m64136_211111_194404.dc.q20.fastq.gz</td>\n",
       "      <td>HG00408</td>\n",
       "      <td>HG00408_lib1_dc</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>...</td>\n",
       "      <td>134450022475</td>\n",
       "      <td>134.45</td>\n",
       "      <td>396.666667</td>\n",
       "      <td>63712.666667</td>\n",
       "      <td>16502.333333</td>\n",
       "      <td>17981.000000</td>\n",
       "      <td>21025.666667</td>\n",
       "      <td>16795.666667</td>\n",
       "      <td>18577.333333</td>\n",
       "      <td>22431.333333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename sample_ID         library_ID  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz   HG00099  PG00099_1.HFSS_dc   \n",
       "1   HG00140.m64043_220728_173215.dc.q20.fastq.gz   HG00140    HG00140_lib1_dc   \n",
       "2  HG00280.m54329U_220901_221341.dc.q20.fastq.gz   HG00280    PG00280.HFSS_dc   \n",
       "3   HG00323.m64043_220728_173215.dc.q20.fastq.gz   HG00323    HG00323_lib1_dc   \n",
       "4   HG00408.m64136_211111_194404.dc.q20.fastq.gz   HG00408    HG00408_lib1_dc   \n",
       "\n",
       "  library_strategy library_source   library_selection library_layout  \\\n",
       "0              WGS        GENOMIC  size fractionation         single   \n",
       "1              WGS        GENOMIC  size fractionation         single   \n",
       "2              WGS        GENOMIC  size fractionation         single   \n",
       "3              WGS        GENOMIC  size fractionation         single   \n",
       "4              WGS        GENOMIC  size fractionation         single   \n",
       "\n",
       "      platform  instrument_model  \\\n",
       "0  PACBIO_SMRT  PacBio Sequel II   \n",
       "1  PACBIO_SMRT  PacBio Sequel II   \n",
       "2  PACBIO_SMRT  PacBio Sequel II   \n",
       "3  PACBIO_SMRT  PacBio Sequel II   \n",
       "4  PACBIO_SMRT  PacBio Sequel II   \n",
       "\n",
       "                                  design_description  ...      total_bp  \\\n",
       "0  HiFi sequencing of 20kb fractionated gDNA reba...  ...  166559881890   \n",
       "1  HiFi sequencing of 20kb fractionated gDNA reba...  ...  126611334877   \n",
       "2  HiFi sequencing of 20kb fractionated gDNA reba...  ...  151564344052   \n",
       "3  HiFi sequencing of 20kb fractionated gDNA reba...  ...  120027730827   \n",
       "4  HiFi sequencing of 20kb fractionated gDNA reba...  ...  134450022475   \n",
       "\n",
       "  total_Gbp         min           max   quartile_25   quartile_50  \\\n",
       "0    166.56  196.666667  62248.666667  17449.333333  19636.333333   \n",
       "1    126.62  272.500000  67774.500000  18393.250000  20448.500000   \n",
       "2    151.56   95.000000  64738.666667  17189.666667  19452.666667   \n",
       "3    120.03  368.750000  60432.750000  16941.500000  19105.500000   \n",
       "4    134.45  396.666667  63712.666667  16502.333333  17981.000000   \n",
       "\n",
       "    quartile_75           N25           N50           N75  \n",
       "0  22651.666667  17950.333333  20409.333333  23637.666667  \n",
       "1  23623.250000  18873.500000  21263.000000  24996.250000  \n",
       "2  22546.000000  17712.333333  20266.666667  23568.000000  \n",
       "3  21983.250000  17417.000000  19893.750000  22834.500000  \n",
       "4  21025.666667  16795.666667  18577.333333  22431.333333  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dc_sample_df.head() # finished table with no non-unique\n",
    "# filename, library_ID"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26491e20-a204-4a0f-bf90-40231f6187e3",
   "metadata": {},
   "source": [
    "# Sample Aggregate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "50d26c92-2ec6-470c-b60a-742ac8dc65d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_samples = set(dc_sample_df['sample_ID'].tolist()).union(hifi_sample_df['sample_ID'].tolist(), ont_sample_df['sample_ID'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "c5a6fc28-704e-4f08-8a7f-9847fc0370ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>filetype</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>...</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>read_N50</th>\n",
       "      <th>coverage</th>\n",
       "      <th>100kb+</th>\n",
       "      <th>200kb+</th>\n",
       "      <th>300kb+</th>\n",
       "      <th>400kb+</th>\n",
       "      <th>500kb+</th>\n",
       "      <th>1Mb+</th>\n",
       "      <th>whales</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>02_08_22_R941_HG00558_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG00558</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00558/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG00558_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>98073.000000</td>\n",
       "      <td>54.43</td>\n",
       "      <td>26.69</td>\n",
       "      <td>7.30</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>02_08_22_R941_HG01252_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG01252</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01252/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG01252_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>79697.666667</td>\n",
       "      <td>47.53</td>\n",
       "      <td>17.45</td>\n",
       "      <td>3.02</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>02_08_22_R941_HG02258_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG02258</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG02258/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG02258_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>98838.500000</td>\n",
       "      <td>49.73</td>\n",
       "      <td>25.01</td>\n",
       "      <td>6.62</td>\n",
       "      <td>1.26</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG03834</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG03834/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG03834_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>107647.333333</td>\n",
       "      <td>48.80</td>\n",
       "      <td>26.33</td>\n",
       "      <td>8.97</td>\n",
       "      <td>3.23</td>\n",
       "      <td>1.35</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.03</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>02_15_22_R941_HG00658_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG00658</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00658/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_15_22_R941_HG00658_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>90313.333333</td>\n",
       "      <td>68.25</td>\n",
       "      <td>30.80</td>\n",
       "      <td>8.37</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            filename sample_ID  \\\n",
       "0  02_08_22_R941_HG00558_1_Guppy_6.5.7_450bps_mod...   HG00558   \n",
       "1  02_08_22_R941_HG01252_1_Guppy_6.5.7_450bps_mod...   HG01252   \n",
       "2  02_08_22_R941_HG02258_1_Guppy_6.5.7_450bps_mod...   HG02258   \n",
       "3  02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mod...   HG03834   \n",
       "4  02_15_22_R941_HG00658_1_Guppy_6.5.7_450bps_mod...   HG00658   \n",
       "\n",
       "                                                path filetype  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG00558/ra...      bam   \n",
       "1  s3://human-pangenomics/working/HPRC/HG01252/ra...      bam   \n",
       "2  s3://human-pangenomics/working/HPRC/HG02258/ra...      bam   \n",
       "3  s3://human-pangenomics/working/HPRC/HG03834/ra...      bam   \n",
       "4  s3://human-pangenomics/working/HPRC/HG00658/ra...      bam   \n",
       "\n",
       "                library_ID library_strategy library_source library_selection  \\\n",
       "0  02_08_22_R941_HG00558_1              WGS        GENOMIC            RANDOM   \n",
       "1  02_08_22_R941_HG01252_1              WGS        GENOMIC            RANDOM   \n",
       "2  02_08_22_R941_HG02258_1              WGS        GENOMIC            RANDOM   \n",
       "3  02_08_22_R941_HG03834_1              WGS        GENOMIC            RANDOM   \n",
       "4  02_15_22_R941_HG00658_1              WGS        GENOMIC            RANDOM   \n",
       "\n",
       "  library_layout         platform  ...  generator_contact       read_N50  \\\n",
       "0         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   98073.000000   \n",
       "1         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   79697.666667   \n",
       "2         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   98838.500000   \n",
       "3         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu  107647.333333   \n",
       "4         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   90313.333333   \n",
       "\n",
       "  coverage 100kb+ 200kb+ 300kb+ 400kb+ 500kb+  1Mb+ whales  \n",
       "0    54.43  26.69   7.30   1.60   0.45   0.18  0.00     35  \n",
       "1    47.53  17.45   3.02   0.66   0.18   0.07  0.00      7  \n",
       "2    49.73  25.01   6.62   1.26   0.26   0.08  0.00      5  \n",
       "3    48.80  26.33   8.97   3.23   1.35   0.64  0.03     99  \n",
       "4    68.25  30.80   8.37   1.85   0.49   0.18  0.00     16  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84027db9-8584-496d-ac1d-ab23844da48d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df.groupby(['Age', 'Gender']).agg(lambda x: x.tolist()).reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "addd0b0a-0220-421c-b709-41a0aac5aa12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>filename</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GM18522</td>\n",
       "      <td>[05_17_22_R941_GM18522_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18522/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GM18570</td>\n",
       "      <td>[08_10_22_R941_GM18570_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18570/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GM18612</td>\n",
       "      <td>[06_28_22_R941_GM18612_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18612/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GM18747</td>\n",
       "      <td>[06_28_22_R941_GM18747_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18747/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GM18971</td>\n",
       "      <td>[06_28_22_R941_GM18971_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18971/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>HG04184</td>\n",
       "      <td>[08_25_21_R941_HG04184_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04184/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>HG04187</td>\n",
       "      <td>[08_10_21_R941_HG04187_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04187/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>HG04199</td>\n",
       "      <td>[08_10_21_R941_HG04199_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04199/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>HG04204</td>\n",
       "      <td>[08_10_21_R941_HG04204_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04204/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>HG04228</td>\n",
       "      <td>[08_10_21_R941_HG04228_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04228/r...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>124 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_ID                                           filename  \\\n",
       "0     GM18522  [05_17_22_R941_GM18522_1_Guppy_6.5.7_450bps_mo...   \n",
       "1     GM18570  [08_10_22_R941_GM18570_1_Guppy_6.5.7_450bps_mo...   \n",
       "2     GM18612  [06_28_22_R941_GM18612_1_Guppy_6.5.7_450bps_mo...   \n",
       "3     GM18747  [06_28_22_R941_GM18747_1_Guppy_6.5.7_450bps_mo...   \n",
       "4     GM18971  [06_28_22_R941_GM18971_1_Guppy_6.5.7_450bps_mo...   \n",
       "..        ...                                                ...   \n",
       "119   HG04184  [08_25_21_R941_HG04184_1_Guppy_6.5.7_450bps_mo...   \n",
       "120   HG04187  [08_10_21_R941_HG04187_1_Guppy_6.5.7_450bps_mo...   \n",
       "121   HG04199  [08_10_21_R941_HG04199_1_Guppy_6.5.7_450bps_mo...   \n",
       "122   HG04204  [08_10_21_R941_HG04204_1_Guppy_6.5.7_450bps_mo...   \n",
       "123   HG04228  [08_10_21_R941_HG04228_1_Guppy_6.5.7_450bps_mo...   \n",
       "\n",
       "                                                  path  \n",
       "0    [s3://human-pangenomics/working/HPRC/NA18522/r...  \n",
       "1    [s3://human-pangenomics/working/HPRC/NA18570/r...  \n",
       "2    [s3://human-pangenomics/working/HPRC/NA18612/r...  \n",
       "3    [s3://human-pangenomics/working/HPRC/NA18747/r...  \n",
       "4    [s3://human-pangenomics/working/HPRC/NA18971/r...  \n",
       "..                                                 ...  \n",
       "119  [s3://human-pangenomics/working/HPRC/HG04184/r...  \n",
       "120  [s3://human-pangenomics/working/HPRC/HG04187/r...  \n",
       "121  [s3://human-pangenomics/working/HPRC/HG04199/r...  \n",
       "122  [s3://human-pangenomics/working/HPRC/HG04204/r...  \n",
       "123  [s3://human-pangenomics/working/HPRC/HG04228/r...  \n",
       "\n",
       "[124 rows x 3 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_sample_df[['sample_ID','filename','path']].groupby(['sample_ID']).agg(lambda x: x.tolist()).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f615cdf-fae2-4907-84ba-7345f2715996",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
