{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create a table with SRA IDs, readstats QC data, and submitter metadata\n",
    "\n",
    "**Below are the steps taken in this notebook:**\n",
    "1. Installs, import Statements & Global Variable Definitions\n",
    "2. Read in tables as dataframes\n",
    "3. Merge dataframes\n",
    "4. Write merged dataframe to data table and file\n",
    "\n",
    "**Important Note**  \n",
    "This may not be the best way to aggregate the data, because SRA submissions are done after QC, which means the workspace needs to be kept and run with newly uploaded info. It might be easier to just output the Readstats table and merge outside Terra/Anvil\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "%pip install gcsfs\n",
    "## capture CANNOT have comments above it\n",
    "## For reading CSVs stored in Google Cloud (without downloading them first)\n",
    "## May need to restart kernel after install "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "%pip install --upgrade --no-cache-dir --force-reinstall terra-pandas\n",
    "%pip install --upgrade --no-cache-dir  --force-reinstall git+https://github.com/DataBiosphere/terra-notebook-utils\n",
    "## For reading/writing data tables into pandas data frames\n",
    "## May need to restart kernel after install "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Statements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from firecloud import fiss\n",
    "import pandas as pd      \n",
    "import os                 \n",
    "import subprocess       \n",
    "import re                 \n",
    "import io\n",
    "import gcsfs\n",
    "\n",
    "from typing import Any, Callable, List, Optional\n",
    "from terra_notebook_utils import table, WORKSPACE_NAME, WORKSPACE_GOOGLE_PROJECT\n",
    "from terra_pandas import dataframe_to_table, table_to_dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global Variable Declarations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing project: human-pangenome-ucsc\n",
      "Workspace: HPRC_WRANGLING_DEEPCONSENSUS_v1pt2\n",
      "Workspace storage bucket: gs://fc-285575a2-c2d5-4f17-a6a1-d1811d4afab4/\n"
     ]
    }
   ],
   "source": [
    "# AnVIL_HPRC WorkspaceBucket\n",
    "anvil_hprc_bucket       = \"gs://fc-285575a2-c2d5-4f17-a6a1-d1811d4afab4/\"\n",
    "\n",
    "# table filenames (expected in the workspace bucket, see below)\n",
    "submitter_metadata = 'DEEPCONSENSUS_v1pt2_submitter_metadata.txt'\n",
    "sra_metadata = 'metadata-13754908-processed-ok.tsv'\n",
    "\n",
    "# submission id\n",
    "submission_id = 'DEEPCONSENSUS_v1pt2'\n",
    "\n",
    "# Get the Google billing project name and workspace name for current workspace\n",
    "PROJECT = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE =os.path.basename(os.path.dirname(os.getcwd()))\n",
    "bucket = os.environ['WORKSPACE_BUCKET'] + \"/\"\n",
    "\n",
    "# Verify that we've captured the environment variables\n",
    "print(\"Billing project: \" + PROJECT)\n",
    "print(\"Workspace: \" + WORKSPACE)\n",
    "print(\"Workspace storage bucket: \" + bucket)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read In tables as dataframes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Readstats table\n",
    "Find the individual readstats output files in the table and concatenate their contents.\n",
    "Add the hifi filename and sample name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>1</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>mean</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>quartile_50</th>\n",
       "      <th>quartile_75</th>\n",
       "      <th>N25</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "      <th>filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2169288</td>\n",
       "      <td>40034895477</td>\n",
       "      <td>40.03</td>\n",
       "      <td>43</td>\n",
       "      <td>49368</td>\n",
       "      <td>18455</td>\n",
       "      <td>16373</td>\n",
       "      <td>18217</td>\n",
       "      <td>20127</td>\n",
       "      <td>16798</td>\n",
       "      <td>18643</td>\n",
       "      <td>20614</td>\n",
       "      <td>HG00423.m64043_210205_190424.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2102877</td>\n",
       "      <td>38839139125</td>\n",
       "      <td>38.84</td>\n",
       "      <td>25</td>\n",
       "      <td>50437</td>\n",
       "      <td>18469</td>\n",
       "      <td>16369</td>\n",
       "      <td>18215</td>\n",
       "      <td>20151</td>\n",
       "      <td>16793</td>\n",
       "      <td>18648</td>\n",
       "      <td>20653</td>\n",
       "      <td>HG00423.m64043_210207_011920.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1917582</td>\n",
       "      <td>39047683785</td>\n",
       "      <td>39.05</td>\n",
       "      <td>47</td>\n",
       "      <td>50778</td>\n",
       "      <td>20362</td>\n",
       "      <td>17437</td>\n",
       "      <td>19156</td>\n",
       "      <td>22527</td>\n",
       "      <td>17779</td>\n",
       "      <td>19886</td>\n",
       "      <td>23774</td>\n",
       "      <td>HG00609.m64136_210622_163607.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1835281</td>\n",
       "      <td>34913876902</td>\n",
       "      <td>34.91</td>\n",
       "      <td>44</td>\n",
       "      <td>49808</td>\n",
       "      <td>19023</td>\n",
       "      <td>16246</td>\n",
       "      <td>18055</td>\n",
       "      <td>21265</td>\n",
       "      <td>16586</td>\n",
       "      <td>18981</td>\n",
       "      <td>22124</td>\n",
       "      <td>HG03654.m64043_210524_174752.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1435262</td>\n",
       "      <td>27331360113</td>\n",
       "      <td>27.33</td>\n",
       "      <td>722</td>\n",
       "      <td>50298</td>\n",
       "      <td>19042</td>\n",
       "      <td>16278</td>\n",
       "      <td>18102</td>\n",
       "      <td>21275</td>\n",
       "      <td>16617</td>\n",
       "      <td>19019</td>\n",
       "      <td>22123</td>\n",
       "      <td>HG03654.m64043_210528_171552.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "1 total_reads     total_bp total_Gbp  min    max   mean quartile_25  \\\n",
       "2     2169288  40034895477     40.03   43  49368  18455       16373   \n",
       "2     2102877  38839139125     38.84   25  50437  18469       16369   \n",
       "2     1917582  39047683785     39.05   47  50778  20362       17437   \n",
       "2     1835281  34913876902     34.91   44  49808  19023       16246   \n",
       "2     1435262  27331360113     27.33  722  50298  19042       16278   \n",
       "\n",
       "1 quartile_50 quartile_75    N25    N50    N75  \\\n",
       "2       18217       20127  16798  18643  20614   \n",
       "2       18215       20151  16793  18648  20653   \n",
       "2       19156       22527  17779  19886  23774   \n",
       "2       18055       21265  16586  18981  22124   \n",
       "2       18102       21275  16617  19019  22123   \n",
       "\n",
       "1                                      filename  \n",
       "2  HG00423.m64043_210205_190424.dc.q20.fastq.gz  \n",
       "2  HG00423.m64043_210207_011920.dc.q20.fastq.gz  \n",
       "2  HG00609.m64136_210622_163607.dc.q20.fastq.gz  \n",
       "2  HG03654.m64043_210524_174752.dc.q20.fastq.gz  \n",
       "2  HG03654.m64043_210528_171552.dc.q20.fastq.gz  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# readstats\n",
    "readstats_df = table_to_dataframe(\"readstats\", workspace=WORKSPACE, workspace_namespace=PROJECT)\n",
    "qc_list = []\n",
    "\n",
    "for index, row in readstats_df.iterrows():\n",
    "\n",
    "        sample_readstats_fp = row['ReadStatsReport']\n",
    "        sample_readstats_fn = os.path.basename(sample_readstats_fp)\n",
    "        df = pd.read_csv(sample_readstats_fp, header=None, sep='\\t')\n",
    "        df = df[df[0]=='sample.fastq']\n",
    "        # add sample and file name\n",
    "        #df = df.append({0:'undef', 1:'sample', 2:row['sample']}, ignore_index=True)\n",
    "        df = df.append({0:'undef', 1:'filename', 2:os.path.basename(row['hifi'])}, ignore_index=True)\n",
    "        # remove all but the variables (total_bp, quartile_25 etc)\n",
    "        df2 = df.drop(df.columns[[0, 1]], axis=1)\n",
    "        # make rownames ('total_bp', 'quartile_25' etc)\n",
    "        df2.index = df.iloc[:,1]\n",
    "        # remove the now meaningless 'sample.fastq' filename\n",
    "        df2 = df2.drop(index=['file'])\n",
    "        qc_list.append(df2)\n",
    "\n",
    "# merge\n",
    "readstats_df = pd.concat(qc_list, axis=1).transpose()\n",
    "readstats_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Submitter metadata table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00423.m64043_210205_190424.dc.q20.fastq.gz</td>\n",
       "      <td>HG00423</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00423.m64043_210207_011920.dc.q20.fastq.gz</td>\n",
       "      <td>HG00423</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00423.m64043_210209_184051.dc.q20.fastq.gz</td>\n",
       "      <td>HG00423</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00423.m64043_210211_005516.dc.q20.fastq.gz</td>\n",
       "      <td>HG00423</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00544.m64043_210702_183246.dc.q20.fastq.gz</td>\n",
       "      <td>HG00544</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       filename sample_ID  instrument_model  \\\n",
       "0  HG00423.m64043_210205_190424.dc.q20.fastq.gz   HG00423  PacBio Sequel II   \n",
       "1  HG00423.m64043_210207_011920.dc.q20.fastq.gz   HG00423  PacBio Sequel II   \n",
       "2  HG00423.m64043_210209_184051.dc.q20.fastq.gz   HG00423  PacBio Sequel II   \n",
       "3  HG00423.m64043_210211_005516.dc.q20.fastq.gz   HG00423  PacBio Sequel II   \n",
       "4  HG00544.m64043_210702_183246.dc.q20.fastq.gz   HG00544  PacBio Sequel II   \n",
       "\n",
       "   shear_method size_selection  DeepConsensus_version polymerase_version  \\\n",
       "0  Megaruptor 1        SageELF                    1.2                 P2   \n",
       "1  Megaruptor 1        SageELF                    1.2                 P2   \n",
       "2  Megaruptor 1        SageELF                    1.2                 P2   \n",
       "3  Megaruptor 1        SageELF                    1.2                 P2   \n",
       "4  Megaruptor 1        SageELF                    1.2                 P2   \n",
       "\n",
       "  seq_plate_chemistry_version     generator_facility  generator_contact  \n",
       "0                          C2  Washington University  tgraves@wustl.edu  \n",
       "1                          C2  Washington University  tgraves@wustl.edu  \n",
       "2                          C2  Washington University  tgraves@wustl.edu  \n",
       "3                          C2  Washington University  tgraves@wustl.edu  \n",
       "4                          C2  Washington University  tgraves@wustl.edu  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_fp = os.path.join(bucket, submitter_metadata)\n",
    "metadata_df = pd.read_csv(meta_fp, sep='\\t')\n",
    "metadata_df = metadata_df[['filename', 'sample_ID', 'instrument_model', 'shear_method', 'size_selection', 'DeepConsensus_version', \n",
    "                           'polymerase_version', 'seq_plate_chemistry_version', 'generator_facility', 'generator_contact']]\n",
    "metadata_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SRA table\n",
    "Must download after finishing submission.\n",
    "The downloaded table needs to be split in one row per file (instead of filename, filename2 <...> filenameN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>accession</th>\n",
       "      <th>library_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00423.m64043_210205_190424.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618944</td>\n",
       "      <td>HG00423_lib1_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00544.m64136_210520_180433.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618943</td>\n",
       "      <td>HG00544_lib1_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00544.m64043_210702_183246.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618932</td>\n",
       "      <td>HG00544_lib2_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00609.m64136_210622_163607.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618921</td>\n",
       "      <td>HG00609_lib1_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00642.m54329U_210524_174516.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618910</td>\n",
       "      <td>HG00642.HFSS_dc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename    accession       library_ID\n",
       "0   HG00423.m64043_210205_190424.dc.q20.fastq.gz  SRR25618944  HG00423_lib1_dc\n",
       "1   HG00544.m64136_210520_180433.dc.q20.fastq.gz  SRR25618943  HG00544_lib1_dc\n",
       "2   HG00544.m64043_210702_183246.dc.q20.fastq.gz  SRR25618932  HG00544_lib2_dc\n",
       "3   HG00609.m64136_210622_163607.dc.q20.fastq.gz  SRR25618921  HG00609_lib1_dc\n",
       "4  HG00642.m54329U_210524_174516.dc.q20.fastq.gz  SRR25618910  HG00642.HFSS_dc"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Read in SRA file and split by filename\n",
    "\n",
    "sra_file = os.path.join(bucket + sra_metadata)\n",
    "sample_df = pd.read_csv(sra_file, sep='\\t')\n",
    "\n",
    "fnames = [x for x in sample_df.columns if x.startswith('filename')]\n",
    "sra_df = pd.melt(sample_df, id_vars =['library_ID','accession'], value_vars = fnames, value_name='file')\n",
    "sra_df= sra_df.dropna()\n",
    "sra_df = sra_df[['file', 'accession', 'library_ID']]\n",
    "sra_df.columns = ['filename', 'accession', 'library_ID']\n",
    "sra_df.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge all dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(132, 3) (132, 13) (132, 10)\n"
     ]
    }
   ],
   "source": [
    "print(sra_df.shape, readstats_df.shape, metadata_df.shape)\n",
    "# two samples of HG02572 were not uploaded to SRA with the rest of this batch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(readstats_df['filename']) - set(sra_df['filename'])\n",
    "#m54329U_201103_231616.ccs.bam (HG002 sample) is not present in the submitter metadata or on SRA, but it is in the submissions "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>mean</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>...</th>\n",
       "      <th>N75</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00423.m64043_210205_190424.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618944</td>\n",
       "      <td>HG00423_lib1_dc</td>\n",
       "      <td>2169288</td>\n",
       "      <td>40034895477</td>\n",
       "      <td>40.03</td>\n",
       "      <td>43</td>\n",
       "      <td>49368</td>\n",
       "      <td>18455</td>\n",
       "      <td>16373</td>\n",
       "      <td>...</td>\n",
       "      <td>20614</td>\n",
       "      <td>HG00423</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00544.m64136_210520_180433.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618943</td>\n",
       "      <td>HG00544_lib1_dc</td>\n",
       "      <td>1622000</td>\n",
       "      <td>31807341724</td>\n",
       "      <td>31.81</td>\n",
       "      <td>48</td>\n",
       "      <td>50293</td>\n",
       "      <td>19609</td>\n",
       "      <td>16831</td>\n",
       "      <td>...</td>\n",
       "      <td>22870</td>\n",
       "      <td>HG00544</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00544.m64043_210702_183246.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618932</td>\n",
       "      <td>HG00544_lib2_dc</td>\n",
       "      <td>2258284</td>\n",
       "      <td>42595473911</td>\n",
       "      <td>42.60</td>\n",
       "      <td>473</td>\n",
       "      <td>48786</td>\n",
       "      <td>18861</td>\n",
       "      <td>17346</td>\n",
       "      <td>...</td>\n",
       "      <td>20289</td>\n",
       "      <td>HG00544</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00609.m64136_210622_163607.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618921</td>\n",
       "      <td>HG00609_lib1_dc</td>\n",
       "      <td>1917582</td>\n",
       "      <td>39047683785</td>\n",
       "      <td>39.05</td>\n",
       "      <td>47</td>\n",
       "      <td>50778</td>\n",
       "      <td>20362</td>\n",
       "      <td>17437</td>\n",
       "      <td>...</td>\n",
       "      <td>23774</td>\n",
       "      <td>HG00609</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00642.m54329U_210524_174516.dc.q20.fastq.gz</td>\n",
       "      <td>SRR25618910</td>\n",
       "      <td>HG00642.HFSS_dc</td>\n",
       "      <td>2098346</td>\n",
       "      <td>41901198683</td>\n",
       "      <td>41.90</td>\n",
       "      <td>3</td>\n",
       "      <td>50177</td>\n",
       "      <td>19968</td>\n",
       "      <td>16616</td>\n",
       "      <td>...</td>\n",
       "      <td>23857</td>\n",
       "      <td>HG00642</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>BluePippin</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename    accession  \\\n",
       "0   HG00423.m64043_210205_190424.dc.q20.fastq.gz  SRR25618944   \n",
       "1   HG00544.m64136_210520_180433.dc.q20.fastq.gz  SRR25618943   \n",
       "2   HG00544.m64043_210702_183246.dc.q20.fastq.gz  SRR25618932   \n",
       "3   HG00609.m64136_210622_163607.dc.q20.fastq.gz  SRR25618921   \n",
       "4  HG00642.m54329U_210524_174516.dc.q20.fastq.gz  SRR25618910   \n",
       "\n",
       "        library_ID total_reads     total_bp total_Gbp  min    max   mean  \\\n",
       "0  HG00423_lib1_dc     2169288  40034895477     40.03   43  49368  18455   \n",
       "1  HG00544_lib1_dc     1622000  31807341724     31.81   48  50293  19609   \n",
       "2  HG00544_lib2_dc     2258284  42595473911     42.60  473  48786  18861   \n",
       "3  HG00609_lib1_dc     1917582  39047683785     39.05   47  50778  20362   \n",
       "4  HG00642.HFSS_dc     2098346  41901198683     41.90    3  50177  19968   \n",
       "\n",
       "  quartile_25  ...    N75 sample_ID  instrument_model  shear_method  \\\n",
       "0       16373  ...  20614   HG00423  PacBio Sequel II  Megaruptor 1   \n",
       "1       16831  ...  22870   HG00544  PacBio Sequel II  Megaruptor 1   \n",
       "2       17346  ...  20289   HG00544  PacBio Sequel II  Megaruptor 1   \n",
       "3       17437  ...  23774   HG00609  PacBio Sequel II  Megaruptor 1   \n",
       "4       16616  ...  23857   HG00642  PacBio Sequel II  Megaruptor 3   \n",
       "\n",
       "  size_selection DeepConsensus_version polymerase_version  \\\n",
       "0        SageELF                   1.2                 P2   \n",
       "1        SageELF                   1.2                 P2   \n",
       "2        SageELF                   1.2                 P2   \n",
       "3        SageELF                   1.2                 P2   \n",
       "4     BluePippin                   1.2                 P2   \n",
       "\n",
       "  seq_plate_chemistry_version        generator_facility  generator_contact  \n",
       "0                          C2     Washington University  tgraves@wustl.edu  \n",
       "1                          C2     Washington University  tgraves@wustl.edu  \n",
       "2                          C2     Washington University  tgraves@wustl.edu  \n",
       "3                          C2     Washington University  tgraves@wustl.edu  \n",
       "4                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(\n",
    "    sra_df,\n",
    "    readstats_df,\n",
    "    on='filename')\n",
    "merged_df = pd.merge(\n",
    "    merged_df,\n",
    "    metadata_df,\n",
    "    on='filename')\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write output files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Create table\n",
    "#dataframe_to_table(\"post_sra_metadata\", merged_df, WORKSPACE, PROJECT)\n",
    "outf = os.path.join(bucket, submission_id + '_post_sra_metadata.tsv')\n",
    "merged_df.to_csv(outf, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(132, 24)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
