{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "# Reset the options to default\n",
    "# Set the maximum number of columns to display\n",
    "pd.set_option('display.max_columns', 500)\n",
    "\n",
    "# Set the maximum number of rows to display\n",
    "pd.set_option('display.max_rows', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "hprc_biosample = pd.read_csv('/private/groups/hprc/human-pangenomics/documentation-metadata/HPRC_metadata/data/production/hprc-production-biosample-table.tsv', sep='\\t')\n",
    "hprc_biosample.rename(columns={'Sample':'sample_ID'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra = pd.read_csv('HPRC_WUSTL_Y1_PacBio_HiFi_TopUp_Metadata_Submission_v0.1.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra['sample_id'] = wustl_y1_topup_sra['sample_ID'] + '-' + wustl_y1_topup_sra['filename']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing columns: ['biosample_accession', 'title', 'assembly']\n"
     ]
    }
   ],
   "source": [
    "# List of column names from your metadata sheet\n",
    "metadata_columns = [\n",
    "    'filename', 'library_ID', 'library_strategy', 'library_source', \n",
    "    'library_selection', 'library_layout', 'platform', 'instrument_model', \n",
    "    'design_description', 'data_type', 'shear_method', 'size_selection', \n",
    "    'ccs_algorithm', 'polymerase_version', 'seq_plate_chemistry_version', \n",
    "    'generator_facility', 'generator_contact', 'biosample_accession', \n",
    "    'title', 'filetype', 'assembly'\n",
    "]\n",
    "\n",
    "# Assume you have another DataFrame named 'other_df'\n",
    "# Example: other_df = pd.read_csv('path_to_other_file.csv')\n",
    "\n",
    "# Check if the metadata columns are present in the other DataFrame\n",
    "missing_columns = [col for col in metadata_columns if col not in wustl_y1_topup_sra.columns]\n",
    "\n",
    "# Output the missing columns\n",
    "print(\"Missing columns:\", missing_columns)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_output_df = pd.read_csv('/private/groups/hprc/human-pangenomics/HPRC_metadata/submissions/WUSTL_HPRC_HiFi_Year1_TopUp/hifi_qc/hifi_qc_samples_output_table.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra = wustl_y1_topup_sra[wustl_y1_topup_sra['sample_id'].isin(wustl_y1_topup_sra['sample_id'].tolist())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert wustl_y1_topup_output_df.shape[0] == wustl_y1_topup_sra.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra['assembly'] = 'unaligned'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra = pd.merge(hprc_biosample, wustl_y1_topup_sra, on='sample_ID', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra['title'] = 'PacBio HiFi sequencing of ' + wustl_y1_topup_sra['sample_ID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra.rename(columns={'Accession':'biosample_accession'},inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "wustl_y1_topup_sra = wustl_y1_topup_sra[metadata_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/tmp/ipykernel_1830256/113464685.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wustl_y1_topup_sra['instrument_model'] = 'Revio'\n"
     ]
    }
   ],
   "source": [
    "wustl_y1_topup_sra['instrument_model'] = 'Revio'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/tmp/ipykernel_1830256/1043829791.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wustl_y1_topup_sra['library_ID'] = wustl_y1_topup_sra['library_ID'] + '_' + wustl_y1_topup_sra['filename']\n"
     ]
    }
   ],
   "source": [
    "wustl_y1_topup_sra['library_ID'] = wustl_y1_topup_sra['library_ID'] + '_' + wustl_y1_topup_sra['filename']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/tmp/ipykernel_1830256/2457177396.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  wustl_y1_topup_sra['library_layout'] = 'single'\n"
     ]
    }
   ],
   "source": [
    "wustl_y1_topup_sra['library_layout'] = 'single'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>ccs_algorithm</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>title</th>\n",
       "      <th>filetype</th>\n",
       "      <th>assembly</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2028.bam</td>\n",
       "      <td>HG00673_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861654</td>\n",
       "      <td>PacBio HiFi sequencing of HG00673</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2028.bam</td>\n",
       "      <td>HG00673_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861654</td>\n",
       "      <td>PacBio HiFi sequencing of HG00673</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2028.bam</td>\n",
       "      <td>HG00673_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861654</td>\n",
       "      <td>PacBio HiFi sequencing of HG00673</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2028.bam</td>\n",
       "      <td>HG00673_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861654</td>\n",
       "      <td>PacBio HiFi sequencing of HG00673</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2026.bam</td>\n",
       "      <td>HG01071_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861657</td>\n",
       "      <td>PacBio HiFi sequencing of HG01071</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2026.bam</td>\n",
       "      <td>HG01071_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861657</td>\n",
       "      <td>PacBio HiFi sequencing of HG01071</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2026.bam</td>\n",
       "      <td>HG01071_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861657</td>\n",
       "      <td>PacBio HiFi sequencing of HG01071</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2026.bam</td>\n",
       "      <td>HG01071_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861657</td>\n",
       "      <td>PacBio HiFi sequencing of HG01071</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2024.bam</td>\n",
       "      <td>HG00741_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861656</td>\n",
       "      <td>PacBio HiFi sequencing of HG00741</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2024.bam</td>\n",
       "      <td>HG00741_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861656</td>\n",
       "      <td>PacBio HiFi sequencing of HG00741</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2024.bam</td>\n",
       "      <td>HG00741_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861656</td>\n",
       "      <td>PacBio HiFi sequencing of HG00741</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2024.bam</td>\n",
       "      <td>HG00741_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861656</td>\n",
       "      <td>PacBio HiFi sequencing of HG00741</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2025.bam</td>\n",
       "      <td>HG01175_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861659</td>\n",
       "      <td>PacBio HiFi sequencing of HG01175</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2025.bam</td>\n",
       "      <td>HG01175_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861659</td>\n",
       "      <td>PacBio HiFi sequencing of HG01175</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2025.bam</td>\n",
       "      <td>HG01175_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861659</td>\n",
       "      <td>PacBio HiFi sequencing of HG01175</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2025.bam</td>\n",
       "      <td>HG01175_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861659</td>\n",
       "      <td>PacBio HiFi sequencing of HG01175</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2029.bam</td>\n",
       "      <td>HG01952_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861661</td>\n",
       "      <td>PacBio HiFi sequencing of HG01952</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2029.bam</td>\n",
       "      <td>HG01952_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861661</td>\n",
       "      <td>PacBio HiFi sequencing of HG01952</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2029.bam</td>\n",
       "      <td>HG01952_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861661</td>\n",
       "      <td>PacBio HiFi sequencing of HG01952</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2029.bam</td>\n",
       "      <td>HG01952_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861661</td>\n",
       "      <td>PacBio HiFi sequencing of HG01952</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2023.bam</td>\n",
       "      <td>HG02148_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861663</td>\n",
       "      <td>PacBio HiFi sequencing of HG02148</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2023.bam</td>\n",
       "      <td>HG02148_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861663</td>\n",
       "      <td>PacBio HiFi sequencing of HG02148</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2023.bam</td>\n",
       "      <td>HG02148_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861663</td>\n",
       "      <td>PacBio HiFi sequencing of HG02148</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2023.bam</td>\n",
       "      <td>HG02148_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861663</td>\n",
       "      <td>PacBio HiFi sequencing of HG02148</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2022.bam</td>\n",
       "      <td>HG01928_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861660</td>\n",
       "      <td>PacBio HiFi sequencing of HG01928</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2022.bam</td>\n",
       "      <td>HG01928_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861660</td>\n",
       "      <td>PacBio HiFi sequencing of HG01928</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2022.bam</td>\n",
       "      <td>HG01928_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861660</td>\n",
       "      <td>PacBio HiFi sequencing of HG01928</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2022.bam</td>\n",
       "      <td>HG01928_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861660</td>\n",
       "      <td>PacBio HiFi sequencing of HG01928</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2021.bam</td>\n",
       "      <td>HG01978_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861662</td>\n",
       "      <td>PacBio HiFi sequencing of HG01978</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2021.bam</td>\n",
       "      <td>HG01978_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861662</td>\n",
       "      <td>PacBio HiFi sequencing of HG01978</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2021.bam</td>\n",
       "      <td>HG01978_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861662</td>\n",
       "      <td>PacBio HiFi sequencing of HG01978</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2021.bam</td>\n",
       "      <td>HG01978_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861662</td>\n",
       "      <td>PacBio HiFi sequencing of HG01978</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>m84081_231222_081401_s1.hifi_reads.bc2027.bam</td>\n",
       "      <td>HG00621_lib1_m84081_231222_081401_s1.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861653</td>\n",
       "      <td>PacBio HiFi sequencing of HG00621</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>m84081_231222_101332_s2.hifi_reads.bc2027.bam</td>\n",
       "      <td>HG00621_lib1_m84081_231222_101332_s2.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861653</td>\n",
       "      <td>PacBio HiFi sequencing of HG00621</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>m84081_231222_121301_s3.hifi_reads.bc2027.bam</td>\n",
       "      <td>HG00621_lib1_m84081_231222_121301_s3.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861653</td>\n",
       "      <td>PacBio HiFi sequencing of HG00621</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>m84081_231222_141231_s4.hifi_reads.bc2027.bam</td>\n",
       "      <td>HG00621_lib1_m84081_231222_141231_s4.hifi_read...</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>size fractionation</td>\n",
       "      <td>single</td>\n",
       "      <td>PACBIO_SMRT</td>\n",
       "      <td>Revio</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 1</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>Revio P1</td>\n",
       "      <td>Revio C1</td>\n",
       "      <td>Washington University</td>\n",
       "      <td>tgraves@wustl.edu</td>\n",
       "      <td>SAMN17861653</td>\n",
       "      <td>PacBio HiFi sequencing of HG00621</td>\n",
       "      <td>bam</td>\n",
       "      <td>unaligned</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         filename  \\\n",
       "0   m84081_231222_081401_s1.hifi_reads.bc2028.bam   \n",
       "1   m84081_231222_101332_s2.hifi_reads.bc2028.bam   \n",
       "2   m84081_231222_121301_s3.hifi_reads.bc2028.bam   \n",
       "3   m84081_231222_141231_s4.hifi_reads.bc2028.bam   \n",
       "4   m84081_231222_081401_s1.hifi_reads.bc2026.bam   \n",
       "5   m84081_231222_101332_s2.hifi_reads.bc2026.bam   \n",
       "6   m84081_231222_121301_s3.hifi_reads.bc2026.bam   \n",
       "7   m84081_231222_141231_s4.hifi_reads.bc2026.bam   \n",
       "8   m84081_231222_081401_s1.hifi_reads.bc2024.bam   \n",
       "9   m84081_231222_101332_s2.hifi_reads.bc2024.bam   \n",
       "10  m84081_231222_121301_s3.hifi_reads.bc2024.bam   \n",
       "11  m84081_231222_141231_s4.hifi_reads.bc2024.bam   \n",
       "12  m84081_231222_081401_s1.hifi_reads.bc2025.bam   \n",
       "13  m84081_231222_101332_s2.hifi_reads.bc2025.bam   \n",
       "14  m84081_231222_121301_s3.hifi_reads.bc2025.bam   \n",
       "15  m84081_231222_141231_s4.hifi_reads.bc2025.bam   \n",
       "16  m84081_231222_081401_s1.hifi_reads.bc2029.bam   \n",
       "17  m84081_231222_101332_s2.hifi_reads.bc2029.bam   \n",
       "18  m84081_231222_121301_s3.hifi_reads.bc2029.bam   \n",
       "19  m84081_231222_141231_s4.hifi_reads.bc2029.bam   \n",
       "20  m84081_231222_081401_s1.hifi_reads.bc2023.bam   \n",
       "21  m84081_231222_101332_s2.hifi_reads.bc2023.bam   \n",
       "22  m84081_231222_121301_s3.hifi_reads.bc2023.bam   \n",
       "23  m84081_231222_141231_s4.hifi_reads.bc2023.bam   \n",
       "24  m84081_231222_081401_s1.hifi_reads.bc2022.bam   \n",
       "25  m84081_231222_101332_s2.hifi_reads.bc2022.bam   \n",
       "26  m84081_231222_121301_s3.hifi_reads.bc2022.bam   \n",
       "27  m84081_231222_141231_s4.hifi_reads.bc2022.bam   \n",
       "28  m84081_231222_081401_s1.hifi_reads.bc2021.bam   \n",
       "29  m84081_231222_101332_s2.hifi_reads.bc2021.bam   \n",
       "30  m84081_231222_121301_s3.hifi_reads.bc2021.bam   \n",
       "31  m84081_231222_141231_s4.hifi_reads.bc2021.bam   \n",
       "32  m84081_231222_081401_s1.hifi_reads.bc2027.bam   \n",
       "33  m84081_231222_101332_s2.hifi_reads.bc2027.bam   \n",
       "34  m84081_231222_121301_s3.hifi_reads.bc2027.bam   \n",
       "35  m84081_231222_141231_s4.hifi_reads.bc2027.bam   \n",
       "\n",
       "                                           library_ID library_strategy  \\\n",
       "0   HG00673_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "1   HG00673_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "2   HG00673_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "3   HG00673_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "4   HG01071_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "5   HG01071_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "6   HG01071_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "7   HG01071_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "8   HG00741_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "9   HG00741_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "10  HG00741_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "11  HG00741_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "12  HG01175_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "13  HG01175_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "14  HG01175_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "15  HG01175_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "16  HG01952_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "17  HG01952_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "18  HG01952_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "19  HG01952_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "20  HG02148_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "21  HG02148_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "22  HG02148_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "23  HG02148_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "24  HG01928_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "25  HG01928_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "26  HG01928_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "27  HG01928_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "28  HG01978_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "29  HG01978_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "30  HG01978_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "31  HG01978_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "32  HG00621_lib1_m84081_231222_081401_s1.hifi_read...              WGS   \n",
       "33  HG00621_lib1_m84081_231222_101332_s2.hifi_read...              WGS   \n",
       "34  HG00621_lib1_m84081_231222_121301_s3.hifi_read...              WGS   \n",
       "35  HG00621_lib1_m84081_231222_141231_s4.hifi_read...              WGS   \n",
       "\n",
       "   library_source   library_selection library_layout     platform  \\\n",
       "0         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "1         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "2         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "3         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "4         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "5         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "6         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "7         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "8         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "9         GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "10        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "11        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "12        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "13        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "14        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "15        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "16        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "17        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "18        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "19        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "20        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "21        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "22        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "23        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "24        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "25        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "26        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "27        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "28        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "29        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "30        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "31        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "32        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "33        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "34        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "35        GENOMIC  size fractionation         single  PACBIO_SMRT   \n",
       "\n",
       "   instrument_model                         design_description  \\\n",
       "0             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "1             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "2             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "3             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "4             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "5             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "6             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "7             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "8             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "9             Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "10            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "11            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "12            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "13            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "14            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "15            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "16            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "17            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "18            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "19            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "20            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "21            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "22            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "23            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "24            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "25            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "26            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "27            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "28            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "29            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "30            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "31            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "32            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "33            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "34            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "35            Revio  HiFi sequencing of 20kb fractionated gDNA   \n",
       "\n",
       "          data_type  shear_method size_selection ccs_algorithm  \\\n",
       "0   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "1   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "2   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "3   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "4   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "5   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "6   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "7   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "8   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "9   unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "10  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "11  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "12  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "13  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "14  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "15  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "16  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "17  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "18  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "19  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "20  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "21  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "22  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "23  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "24  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "25  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "26  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "27  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "28  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "29  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "30  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "31  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "32  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "33  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "34  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "35  unaligned reads  Megaruptor 1        SageELF         8.0.0   \n",
       "\n",
       "   polymerase_version seq_plate_chemistry_version     generator_facility  \\\n",
       "0            Revio P1                    Revio C1  Washington University   \n",
       "1            Revio P1                    Revio C1  Washington University   \n",
       "2            Revio P1                    Revio C1  Washington University   \n",
       "3            Revio P1                    Revio C1  Washington University   \n",
       "4            Revio P1                    Revio C1  Washington University   \n",
       "5            Revio P1                    Revio C1  Washington University   \n",
       "6            Revio P1                    Revio C1  Washington University   \n",
       "7            Revio P1                    Revio C1  Washington University   \n",
       "8            Revio P1                    Revio C1  Washington University   \n",
       "9            Revio P1                    Revio C1  Washington University   \n",
       "10           Revio P1                    Revio C1  Washington University   \n",
       "11           Revio P1                    Revio C1  Washington University   \n",
       "12           Revio P1                    Revio C1  Washington University   \n",
       "13           Revio P1                    Revio C1  Washington University   \n",
       "14           Revio P1                    Revio C1  Washington University   \n",
       "15           Revio P1                    Revio C1  Washington University   \n",
       "16           Revio P1                    Revio C1  Washington University   \n",
       "17           Revio P1                    Revio C1  Washington University   \n",
       "18           Revio P1                    Revio C1  Washington University   \n",
       "19           Revio P1                    Revio C1  Washington University   \n",
       "20           Revio P1                    Revio C1  Washington University   \n",
       "21           Revio P1                    Revio C1  Washington University   \n",
       "22           Revio P1                    Revio C1  Washington University   \n",
       "23           Revio P1                    Revio C1  Washington University   \n",
       "24           Revio P1                    Revio C1  Washington University   \n",
       "25           Revio P1                    Revio C1  Washington University   \n",
       "26           Revio P1                    Revio C1  Washington University   \n",
       "27           Revio P1                    Revio C1  Washington University   \n",
       "28           Revio P1                    Revio C1  Washington University   \n",
       "29           Revio P1                    Revio C1  Washington University   \n",
       "30           Revio P1                    Revio C1  Washington University   \n",
       "31           Revio P1                    Revio C1  Washington University   \n",
       "32           Revio P1                    Revio C1  Washington University   \n",
       "33           Revio P1                    Revio C1  Washington University   \n",
       "34           Revio P1                    Revio C1  Washington University   \n",
       "35           Revio P1                    Revio C1  Washington University   \n",
       "\n",
       "    generator_contact biosample_accession                              title  \\\n",
       "0   tgraves@wustl.edu        SAMN17861654  PacBio HiFi sequencing of HG00673   \n",
       "1   tgraves@wustl.edu        SAMN17861654  PacBio HiFi sequencing of HG00673   \n",
       "2   tgraves@wustl.edu        SAMN17861654  PacBio HiFi sequencing of HG00673   \n",
       "3   tgraves@wustl.edu        SAMN17861654  PacBio HiFi sequencing of HG00673   \n",
       "4   tgraves@wustl.edu        SAMN17861657  PacBio HiFi sequencing of HG01071   \n",
       "5   tgraves@wustl.edu        SAMN17861657  PacBio HiFi sequencing of HG01071   \n",
       "6   tgraves@wustl.edu        SAMN17861657  PacBio HiFi sequencing of HG01071   \n",
       "7   tgraves@wustl.edu        SAMN17861657  PacBio HiFi sequencing of HG01071   \n",
       "8   tgraves@wustl.edu        SAMN17861656  PacBio HiFi sequencing of HG00741   \n",
       "9   tgraves@wustl.edu        SAMN17861656  PacBio HiFi sequencing of HG00741   \n",
       "10  tgraves@wustl.edu        SAMN17861656  PacBio HiFi sequencing of HG00741   \n",
       "11  tgraves@wustl.edu        SAMN17861656  PacBio HiFi sequencing of HG00741   \n",
       "12  tgraves@wustl.edu        SAMN17861659  PacBio HiFi sequencing of HG01175   \n",
       "13  tgraves@wustl.edu        SAMN17861659  PacBio HiFi sequencing of HG01175   \n",
       "14  tgraves@wustl.edu        SAMN17861659  PacBio HiFi sequencing of HG01175   \n",
       "15  tgraves@wustl.edu        SAMN17861659  PacBio HiFi sequencing of HG01175   \n",
       "16  tgraves@wustl.edu        SAMN17861661  PacBio HiFi sequencing of HG01952   \n",
       "17  tgraves@wustl.edu        SAMN17861661  PacBio HiFi sequencing of HG01952   \n",
       "18  tgraves@wustl.edu        SAMN17861661  PacBio HiFi sequencing of HG01952   \n",
       "19  tgraves@wustl.edu        SAMN17861661  PacBio HiFi sequencing of HG01952   \n",
       "20  tgraves@wustl.edu        SAMN17861663  PacBio HiFi sequencing of HG02148   \n",
       "21  tgraves@wustl.edu        SAMN17861663  PacBio HiFi sequencing of HG02148   \n",
       "22  tgraves@wustl.edu        SAMN17861663  PacBio HiFi sequencing of HG02148   \n",
       "23  tgraves@wustl.edu        SAMN17861663  PacBio HiFi sequencing of HG02148   \n",
       "24  tgraves@wustl.edu        SAMN17861660  PacBio HiFi sequencing of HG01928   \n",
       "25  tgraves@wustl.edu        SAMN17861660  PacBio HiFi sequencing of HG01928   \n",
       "26  tgraves@wustl.edu        SAMN17861660  PacBio HiFi sequencing of HG01928   \n",
       "27  tgraves@wustl.edu        SAMN17861660  PacBio HiFi sequencing of HG01928   \n",
       "28  tgraves@wustl.edu        SAMN17861662  PacBio HiFi sequencing of HG01978   \n",
       "29  tgraves@wustl.edu        SAMN17861662  PacBio HiFi sequencing of HG01978   \n",
       "30  tgraves@wustl.edu        SAMN17861662  PacBio HiFi sequencing of HG01978   \n",
       "31  tgraves@wustl.edu        SAMN17861662  PacBio HiFi sequencing of HG01978   \n",
       "32  tgraves@wustl.edu        SAMN17861653  PacBio HiFi sequencing of HG00621   \n",
       "33  tgraves@wustl.edu        SAMN17861653  PacBio HiFi sequencing of HG00621   \n",
       "34  tgraves@wustl.edu        SAMN17861653  PacBio HiFi sequencing of HG00621   \n",
       "35  tgraves@wustl.edu        SAMN17861653  PacBio HiFi sequencing of HG00621   \n",
       "\n",
       "   filetype   assembly  \n",
       "0       bam  unaligned  \n",
       "1       bam  unaligned  \n",
       "2       bam  unaligned  \n",
       "3       bam  unaligned  \n",
       "4       bam  unaligned  \n",
       "5       bam  unaligned  \n",
       "6       bam  unaligned  \n",
       "7       bam  unaligned  \n",
       "8       bam  unaligned  \n",
       "9       bam  unaligned  \n",
       "10      bam  unaligned  \n",
       "11      bam  unaligned  \n",
       "12      bam  unaligned  \n",
       "13      bam  unaligned  \n",
       "14      bam  unaligned  \n",
       "15      bam  unaligned  \n",
       "16      bam  unaligned  \n",
       "17      bam  unaligned  \n",
       "18      bam  unaligned  \n",
       "19      bam  unaligned  \n",
       "20      bam  unaligned  \n",
       "21      bam  unaligned  \n",
       "22      bam  unaligned  \n",
       "23      bam  unaligned  \n",
       "24      bam  unaligned  \n",
       "25      bam  unaligned  \n",
       "26      bam  unaligned  \n",
       "27      bam  unaligned  \n",
       "28      bam  unaligned  \n",
       "29      bam  unaligned  \n",
       "30      bam  unaligned  \n",
       "31      bam  unaligned  \n",
       "32      bam  unaligned  \n",
       "33      bam  unaligned  \n",
       "34      bam  unaligned  \n",
       "35      bam  unaligned  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wustl_y1_topup_sra.to_csv('sra_biosample_WUSTL_Y1_TopUp.tsv',index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
