{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "hprc_biosample = pd.read_csv('/private/groups/hprc/human-pangenomics/HPRC_metadata/data/production/hprc-production-biosample-table.tsv',sep='\\t')\n",
    "hprc_biosample.rename(columns={'Sample':'sample_id'},inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_df = pd.read_csv('/private/groups/hprc/human-pangenomics/HPRC_metadata/submissions/HIC_Y3_fillin_and_Y4_part1-redo/HIC_Y3_fillin_and_Y4_part1-redo-sample_file_hifi_qc_aggregate.csv')\n",
    "\n",
    "hic_df['sample_id'] = [sample_id.split('-')[0] for sample_id in hic_df['file_name'].tolist()]\n",
    "\n",
    "hic_sra_df = pd.read_csv('/private/groups/hprc/human-pangenomics/HPRC_metadata/submissions/HIC_Y3_fillin_and_Y4_part1-redo/sra_biosample_HIC_Y3_fillin_and_Y4_part1.csv')\n",
    "\n",
    "release = pd.read_csv('/private/groups/hprc/human-pangenomics/documentation-metadata/HPRC_metadata/submissions/HIC_Y3_fillin_and_Y4_part1-redo/HIC_Y3_fillin_and_Y4_part1-redo.transfer.csv',header=None)\n",
    "\n",
    "release['file_name'] = [filename.split('/')[-1] for filename in release[1].tolist()]\n",
    "release.rename(columns={1:'path'},inplace=True)\n",
    "release.drop(columns=[0],inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_df = pd.merge(hic_df, release, on='file_name', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_name</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>coverage</th>\n",
       "      <th>ntsm_score</th>\n",
       "      <th>ntsm_result</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG01975-2_S4_L002_R2_001.fastq.gz</td>\n",
       "      <td>45449986488</td>\n",
       "      <td>14.66</td>\n",
       "      <td>0.304342</td>\n",
       "      <td>Similar</td>\n",
       "      <td>HG01975</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01975/ra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG03834-2_S8_L003_R2_001.fastq.gz</td>\n",
       "      <td>69883486748</td>\n",
       "      <td>22.54</td>\n",
       "      <td>0.309635</td>\n",
       "      <td>Similar</td>\n",
       "      <td>HG03834</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG03834/ra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG02129-1_S2_L003_R2_001.fastq.gz</td>\n",
       "      <td>124595417314</td>\n",
       "      <td>40.19</td>\n",
       "      <td>0.309858</td>\n",
       "      <td>Similar</td>\n",
       "      <td>HG02129</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG02129/ra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00408-1_S4_L002_R2_001.fastq.gz</td>\n",
       "      <td>75586775321</td>\n",
       "      <td>24.38</td>\n",
       "      <td>0.311182</td>\n",
       "      <td>Similar</td>\n",
       "      <td>HG00408</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00408/ra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG02514-1_S5_L004_R2_001.fastq.gz</td>\n",
       "      <td>103145633415</td>\n",
       "      <td>33.27</td>\n",
       "      <td>0.320622</td>\n",
       "      <td>Similar</td>\n",
       "      <td>HG02514</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG02514/ra...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           file_name      total_bp  coverage  ntsm_score  \\\n",
       "0  HG01975-2_S4_L002_R2_001.fastq.gz   45449986488     14.66    0.304342   \n",
       "1  HG03834-2_S8_L003_R2_001.fastq.gz   69883486748     22.54    0.309635   \n",
       "2  HG02129-1_S2_L003_R2_001.fastq.gz  124595417314     40.19    0.309858   \n",
       "3  HG00408-1_S4_L002_R2_001.fastq.gz   75586775321     24.38    0.311182   \n",
       "4  HG02514-1_S5_L004_R2_001.fastq.gz  103145633415     33.27    0.320622   \n",
       "\n",
       "  ntsm_result sample_id                                               path  \n",
       "0     Similar   HG01975  s3://human-pangenomics/working/HPRC/HG01975/ra...  \n",
       "1     Similar   HG03834  s3://human-pangenomics/working/HPRC/HG03834/ra...  \n",
       "2     Similar   HG02129  s3://human-pangenomics/working/HPRC/HG02129/ra...  \n",
       "3     Similar   HG00408  s3://human-pangenomics/working/HPRC/HG00408/ra...  \n",
       "4     Similar   HG02514  s3://human-pangenomics/working/HPRC/HG02514/ra...  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(344, 7)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# number of qc samples\n",
    "hic_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hic_df['sample_id'].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# inner merge on hprc biosamples\n",
    "hic_df = pd.merge(hic_df, hprc_biosample[hprc_biosample['sample_id'].isin(hic_df['sample_id'])], on='sample_id', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "# poor qc samples\n",
    "# hic_df[~hic_df['file_name'].isin(hic_sra_df['filename'].tolist())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/tmp/ipykernel_1511726/651745410.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  hic_sra_df['common_id'] = hic_sra_df['filename'].str.extract(r'(^.+_L\\d{3})')\n"
     ]
    }
   ],
   "source": [
    "hic_sra_df = hic_sra_df[hic_sra_df['filename'].isin(hic_df['file_name'])]\n",
    "hic_sra_df['common_id'] = hic_sra_df['filename'].str.extract(r'(^.+_L\\d{3})')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_R1 = hic_sra_df[hic_sra_df['filename'].str.contains('_R1_')].copy()\n",
    "df_R2 = hic_sra_df[hic_sra_df['filename'].str.contains('_R2_')].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_sra_merge_df = pd.merge(df_R1, df_R2, on='common_id', how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(172, 35)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hic_sra_merge_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_sra_merge_df = hic_sra_merge_df.loc[:, ~hic_sra_merge_df.columns.str.endswith('_y')]\n",
    "hic_sra_merge_df.columns = hic_sra_merge_df.columns.str.replace('_x$', '', regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_sra_merge_df['library_ID'] = hic_sra_merge_df['library_ID'].str.extract(r'(.+_L\\d{3})')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_sra_merge_df.drop(columns=['common_id'],inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>filetype</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>assembly</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>title</th>\n",
       "      <th>data_type</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>shear_method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00126-1_FC1_S3_L004_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG00126-1_FC1_S3_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN41021637</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00126-1_FC2_S3_L004_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG00126-1_FC2_S3_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN41021637</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00126-2_FC1_S4_L004_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG00126-2_FC1_S4_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN41021637</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00126-2_FC2_S4_L004_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG00126-2_FC2_S4_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN41021637</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00128-1_FC1_S9_L004_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG00128-1_FC1_S9_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN41021641</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>HG01975-2_S4_L002_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG01975-2_S4_L002</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN33621947</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>HG02841-1_S5_L001_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG02841-1_S5_L001</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN37797106</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>HG02841-2_S5_L002_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG02841-2_S5_L002</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN37797106</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>HG02984-1_S6_L001_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG02984-1_S6_L001</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN37797107</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>HG02984-2_S6_L002_R1_001.fastq.gz</td>\n",
       "      <td>fastq</td>\n",
       "      <td>HG02984-2_S6_L002</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>SAMN37797107</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>172 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  filename filetype             library_ID  \\\n",
       "0    HG00126-1_FC1_S3_L004_R1_001.fastq.gz   fastq   HG00126-1_FC1_S3_L004   \n",
       "1    HG00126-1_FC2_S3_L004_R1_001.fastq.gz   fastq   HG00126-1_FC2_S3_L004   \n",
       "2    HG00126-2_FC1_S4_L004_R1_001.fastq.gz   fastq   HG00126-2_FC1_S4_L004   \n",
       "3    HG00126-2_FC2_S4_L004_R1_001.fastq.gz   fastq   HG00126-2_FC2_S4_L004   \n",
       "4    HG00128-1_FC1_S9_L004_R1_001.fastq.gz   fastq   HG00128-1_FC1_S9_L004   \n",
       "..                                     ...      ...                    ...   \n",
       "167      HG01975-2_S4_L002_R1_001.fastq.gz   fastq       HG01975-2_S4_L002   \n",
       "168      HG02841-1_S5_L001_R1_001.fastq.gz   fastq       HG02841-1_S5_L001   \n",
       "169      HG02841-2_S5_L002_R1_001.fastq.gz   fastq       HG02841-2_S5_L002   \n",
       "170      HG02984-1_S6_L001_R1_001.fastq.gz   fastq       HG02984-1_S6_L001   \n",
       "171      HG02984-2_S6_L002_R1_001.fastq.gz   fastq       HG02984-2_S6_L002   \n",
       "\n",
       "    library_strategy library_source  platform       instrument_model  \\\n",
       "0               Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "1               Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "2               Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "3               Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "4               Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "..               ...            ...       ...                    ...   \n",
       "167             Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "168             Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "169             Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "170             Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "171             Hi-C        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "\n",
       "                          design_description  \\\n",
       "0    Illumina Sequencing of Omni-C Libraries   \n",
       "1    Illumina Sequencing of Omni-C Libraries   \n",
       "2    Illumina Sequencing of Omni-C Libraries   \n",
       "3    Illumina Sequencing of Omni-C Libraries   \n",
       "4    Illumina Sequencing of Omni-C Libraries   \n",
       "..                                       ...   \n",
       "167  Illumina Sequencing of Omni-C Libraries   \n",
       "168  Illumina Sequencing of Omni-C Libraries   \n",
       "169  Illumina Sequencing of Omni-C Libraries   \n",
       "170  Illumina Sequencing of Omni-C Libraries   \n",
       "171  Illumina Sequencing of Omni-C Libraries   \n",
       "\n",
       "                      generator_facility  generator_contact   assembly  \\\n",
       "0    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "1    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "2    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "3    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "4    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "..                                   ...                ...        ...   \n",
       "167  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "168  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "169  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "170  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "171  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "\n",
       "    biosample_accession                                              title  \\\n",
       "0          SAMN41021637  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "1          SAMN41021637  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "2          SAMN41021637  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "3          SAMN41021637  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "4          SAMN41021641  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "..                  ...                                                ...   \n",
       "167        SAMN33621947  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "168        SAMN37797106  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "169        SAMN37797106  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "170        SAMN37797107  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "171        SAMN37797107  Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "\n",
       "           data_type library_layout library_selection shear_method  \n",
       "0    unaligned reads         paired            RANDOM     no shear  \n",
       "1    unaligned reads         paired            RANDOM     no shear  \n",
       "2    unaligned reads         paired            RANDOM     no shear  \n",
       "3    unaligned reads         paired            RANDOM     no shear  \n",
       "4    unaligned reads         paired            RANDOM     no shear  \n",
       "..               ...            ...               ...          ...  \n",
       "167  unaligned reads         paired            RANDOM     no shear  \n",
       "168  unaligned reads         paired            RANDOM     no shear  \n",
       "169  unaligned reads         paired            RANDOM     no shear  \n",
       "170  unaligned reads         paired            RANDOM     no shear  \n",
       "171  unaligned reads         paired            RANDOM     no shear  \n",
       "\n",
       "[172 rows x 17 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hic_sra_merge_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_sra_merge_df.to_csv('sra_biosample_HIC_Y3_fillin_and_Y4_part1-redo.tsv',index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
