{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_y3_metadata = pd.read_csv('HPRC_Illumina_HiC_Metadata_HIC_Y3.csv')\n",
    "hic_y3_transfer = pd.read_csv('HiC-Y3.transfer.csv',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "hprc_production_biosample = pd.read_csv('/private/groups/hprc/human-pangenomics/HPRC_metadata/data/production/hprc-production-biosample-table.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "# title, data_type, library_layout, library_selection, shear_method\n",
    "hic_y3_metadata = hic_y3_metadata[['filename', 'library_ID','library_strategy', 'library_source','platform', 'instrument_model','design_description',\\\n",
    "\t\t\t\t 'generator_facility','generator_contact','assembly','data_type','filetype']]\n",
    "# library_source, data type\n",
    "hic_y3_metadata['library_source'] ='GENOMIC'\n",
    "hic_y3_metadata['data_type'] = 'unaligned reads'\n",
    "hic_y3_metadata['library_layout'] = 'paired'\n",
    "hic_y3_metadata['library_strategy'] = 'Hi-C'\n",
    "hic_y3_metadata['library_selection'] = 'RANDOM'\n",
    "hic_y3_metadata['shear_method'] = 'no shear'\n",
    "title_list = []\n",
    "# hic_y3_metadata['title']\n",
    "for data in hic_y3_metadata['library_ID'].tolist():\n",
    "\ttitle_list.append('Illumina Sequencing of Omni-C Libraries of ' + data.split('-')[0])\n",
    "hic_y3_metadata['title'] = title_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_y3_metadata = hic_y3_metadata[hic_y3_metadata['filename'].isin([file.split('/')[-1] for file in hic_y3_transfer[1].tolist()])]\n",
    "hic_y3_metadata = hic_y3_metadata.sort_values(by='filename')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_y3_metadata['common_id'] = hic_y3_metadata['filename'].str.extract(r'(^.+_L\\d{3})')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_R1 = hic_y3_metadata[hic_y3_metadata['filename'].str.contains('_R1_')].copy()\n",
    "df_R2 = hic_y3_metadata[hic_y3_metadata['filename'].str.contains('_R2_')].copy()\n",
    "# Rename the filename column in the R2 DataFrame to filename2\n",
    "df_R2 = df_R2.rename(columns={'filename': 'filename2'})\n",
    "df_merged = pd.merge(df_R1, df_R2, on=[col for col in hic_y3_metadata.columns if col not in ['filename', 'filename2', 'merge_key']], suffixes=('_R1', '_R2'))\n",
    "df_merged['library_ID'] = df_merged['common_id']\n",
    "df_merged.drop(columns=['common_id'],inplace=True)\n",
    "\n",
    "df_merged['Sample'] = [sample.split('-')[0] for sample in df_merged['filename'].tolist()]\n",
    "hic_y3_sra = pd.merge(df_merged, hprc_production_biosample[['Sample','Accession']], on='Sample',how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "752"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(hic_y3_sra['filename'].tolist() + hic_y3_sra['filename2'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_y3_sra.drop(columns=['Sample'], inplace=True)\n",
    "hic_y3_sra.rename(columns={'Accession':'biosample_accession'},inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>platform</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>design_description</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>assembly</th>\n",
       "      <th>data_type</th>\n",
       "      <th>filetype</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>title</th>\n",
       "      <th>filename2</th>\n",
       "      <th>biosample_accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099-1_S20_L001_R1_001.fastq.gz</td>\n",
       "      <td>HG00099-1_S20_L001</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>HG00099-1_S20_L001_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00099-1_S20_L002_R1_001.fastq.gz</td>\n",
       "      <td>HG00099-1_S20_L002</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>HG00099-1_S20_L002_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099-1_S20_L003_R1_001.fastq.gz</td>\n",
       "      <td>HG00099-1_S20_L003</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>HG00099-1_S20_L003_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00099-1_S20_L004_R1_001.fastq.gz</td>\n",
       "      <td>HG00099-1_S20_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>HG00099-1_S20_L004_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00099-2_S28_L001_R1_001.fastq.gz</td>\n",
       "      <td>HG00099-2_S28_L001</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of HG0...</td>\n",
       "      <td>HG00099-2_S28_L001_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>NA20905-1_S22_L004_R1_001.fastq.gz</td>\n",
       "      <td>NA20905-1_S22_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of NA2...</td>\n",
       "      <td>NA20905-1_S22_L004_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>NA20905-2_S30_L001_R1_001.fastq.gz</td>\n",
       "      <td>NA20905-2_S30_L001</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of NA2...</td>\n",
       "      <td>NA20905-2_S30_L001_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>NA20905-2_S30_L002_R1_001.fastq.gz</td>\n",
       "      <td>NA20905-2_S30_L002</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of NA2...</td>\n",
       "      <td>NA20905-2_S30_L002_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>374</th>\n",
       "      <td>NA20905-2_S30_L003_R1_001.fastq.gz</td>\n",
       "      <td>NA20905-2_S30_L003</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of NA2...</td>\n",
       "      <td>NA20905-2_S30_L003_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>NA20905-2_S30_L004_R1_001.fastq.gz</td>\n",
       "      <td>NA20905-2_S30_L004</td>\n",
       "      <td>Hi-C</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries</td>\n",
       "      <td>University of California Santa Cruz</td>\n",
       "      <td>wseligma@ucsc.edu</td>\n",
       "      <td>unaligned</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>fastq</td>\n",
       "      <td>paired</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>no shear</td>\n",
       "      <td>Illumina Sequencing of Omni-C Libraries of NA2...</td>\n",
       "      <td>NA20905-2_S30_L004_R2_001.fastq.gz</td>\n",
       "      <td>SAMN33758801</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>376 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               filename          library_ID library_strategy  \\\n",
       "0    HG00099-1_S20_L001_R1_001.fastq.gz  HG00099-1_S20_L001             Hi-C   \n",
       "1    HG00099-1_S20_L002_R1_001.fastq.gz  HG00099-1_S20_L002             Hi-C   \n",
       "2    HG00099-1_S20_L003_R1_001.fastq.gz  HG00099-1_S20_L003             Hi-C   \n",
       "3    HG00099-1_S20_L004_R1_001.fastq.gz  HG00099-1_S20_L004             Hi-C   \n",
       "4    HG00099-2_S28_L001_R1_001.fastq.gz  HG00099-2_S28_L001             Hi-C   \n",
       "..                                  ...                 ...              ...   \n",
       "371  NA20905-1_S22_L004_R1_001.fastq.gz  NA20905-1_S22_L004             Hi-C   \n",
       "372  NA20905-2_S30_L001_R1_001.fastq.gz  NA20905-2_S30_L001             Hi-C   \n",
       "373  NA20905-2_S30_L002_R1_001.fastq.gz  NA20905-2_S30_L002             Hi-C   \n",
       "374  NA20905-2_S30_L003_R1_001.fastq.gz  NA20905-2_S30_L003             Hi-C   \n",
       "375  NA20905-2_S30_L004_R1_001.fastq.gz  NA20905-2_S30_L004             Hi-C   \n",
       "\n",
       "    library_source  platform       instrument_model  \\\n",
       "0          GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "1          GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "2          GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "3          GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "4          GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "..             ...       ...                    ...   \n",
       "371        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "372        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "373        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "374        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "375        GENOMIC  ILLUMINA  Illumina NovaSeq 6000   \n",
       "\n",
       "                          design_description  \\\n",
       "0    Illumina Sequencing of Omni-C Libraries   \n",
       "1    Illumina Sequencing of Omni-C Libraries   \n",
       "2    Illumina Sequencing of Omni-C Libraries   \n",
       "3    Illumina Sequencing of Omni-C Libraries   \n",
       "4    Illumina Sequencing of Omni-C Libraries   \n",
       "..                                       ...   \n",
       "371  Illumina Sequencing of Omni-C Libraries   \n",
       "372  Illumina Sequencing of Omni-C Libraries   \n",
       "373  Illumina Sequencing of Omni-C Libraries   \n",
       "374  Illumina Sequencing of Omni-C Libraries   \n",
       "375  Illumina Sequencing of Omni-C Libraries   \n",
       "\n",
       "                      generator_facility  generator_contact   assembly  \\\n",
       "0    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "1    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "2    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "3    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "4    University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "..                                   ...                ...        ...   \n",
       "371  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "372  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "373  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "374  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "375  University of California Santa Cruz  wseligma@ucsc.edu  unaligned   \n",
       "\n",
       "           data_type filetype library_layout library_selection shear_method  \\\n",
       "0    unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "1    unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "2    unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "3    unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "4    unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "..               ...      ...            ...               ...          ...   \n",
       "371  unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "372  unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "373  unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "374  unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "375  unaligned reads    fastq         paired            RANDOM     no shear   \n",
       "\n",
       "                                                 title  \\\n",
       "0    Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "1    Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "2    Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "3    Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "4    Illumina Sequencing of Omni-C Libraries of HG0...   \n",
       "..                                                 ...   \n",
       "371  Illumina Sequencing of Omni-C Libraries of NA2...   \n",
       "372  Illumina Sequencing of Omni-C Libraries of NA2...   \n",
       "373  Illumina Sequencing of Omni-C Libraries of NA2...   \n",
       "374  Illumina Sequencing of Omni-C Libraries of NA2...   \n",
       "375  Illumina Sequencing of Omni-C Libraries of NA2...   \n",
       "\n",
       "                              filename2 biosample_accession  \n",
       "0    HG00099-1_S20_L001_R2_001.fastq.gz        SAMN33758778  \n",
       "1    HG00099-1_S20_L002_R2_001.fastq.gz        SAMN33758778  \n",
       "2    HG00099-1_S20_L003_R2_001.fastq.gz        SAMN33758778  \n",
       "3    HG00099-1_S20_L004_R2_001.fastq.gz        SAMN33758778  \n",
       "4    HG00099-2_S28_L001_R2_001.fastq.gz        SAMN33758778  \n",
       "..                                  ...                 ...  \n",
       "371  NA20905-1_S22_L004_R2_001.fastq.gz        SAMN33758801  \n",
       "372  NA20905-2_S30_L001_R2_001.fastq.gz        SAMN33758801  \n",
       "373  NA20905-2_S30_L002_R2_001.fastq.gz        SAMN33758801  \n",
       "374  NA20905-2_S30_L003_R2_001.fastq.gz        SAMN33758801  \n",
       "375  NA20905-2_S30_L004_R2_001.fastq.gz        SAMN33758801  \n",
       "\n",
       "[376 rows x 18 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hic_y3_sra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_y3_sra.to_csv('sra_biosample_HIC_Y3.tsv',index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
