{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create a table with SRA IDs, readstats QC data, and submitter metadata\n",
    "\n",
    "**Below are the steps taken in this notebook:**\n",
    "1. Installs, import Statements & Global Variable Definitions\n",
    "2. Read in tables as dataframes\n",
    "3. Merge dataframes\n",
    "4. Write merged dataframe to data table and file\n",
    "\n",
    "**Important Note**  \n",
    "This may not be the best way to aggregate the data, because SRA submissions are done after QC, which means the workspace needs to be kept and run with newly uploaded info. It might be easier to just output the Readstats table and merge outside Terra/Anvil\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "%pip install gcsfs\n",
    "## capture CANNOT have comments above it\n",
    "## For reading CSVs stored in Google Cloud (without downloading them first)\n",
    "## May need to restart kernel after install "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "%pip install --upgrade --no-cache-dir --force-reinstall terra-pandas\n",
    "%pip install --upgrade --no-cache-dir  --force-reinstall git+https://github.com/DataBiosphere/terra-notebook-utils\n",
    "## For reading/writing data tables into pandas data frames\n",
    "## May need to restart kernel after install "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Statements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "from firecloud import fiss\n",
    "import pandas as pd      \n",
    "import os                 \n",
    "import subprocess       \n",
    "import re                 \n",
    "import io\n",
    "import gcsfs\n",
    "\n",
    "from typing import Any, Callable, List, Optional\n",
    "from terra_notebook_utils import table, WORKSPACE_NAME, WORKSPACE_GOOGLE_PROJECT\n",
    "from terra_pandas import dataframe_to_table, table_to_dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Global Variable Declarations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Billing project: human-pangenome-ucsc\n",
      "Workspace: HPRC_DEEPCONSENSUS_v1pt2_2023_08_q20\n",
      "Workspace storage bucket: gs://fc-684a1d20-49ec-4a0a-90ab-57b0f5c0aab5/\n"
     ]
    }
   ],
   "source": [
    "# AnVIL_HPRC WorkspaceBucket\n",
    "anvil_hprc_bucket       = \"gs://fc-285575a2-c2d5-4f17-a6a1-d1811d4afab4/\"\n",
    "\n",
    "# table filenames (expected in the workspace bucket, see below)\n",
    "submitter_metadata = 'HPRC_DEEPCONSENSUS_v1pt2_2023_08_q20_submitter_metadata.tsv'\n",
    "sra_metadata = 'HPRC_DEEPCONSENSUS_v1pt2_2023_08_q20-13931699-processed-ok.tsv'\n",
    "\n",
    "# submission id\n",
    "submission_id = 'HPRC_DEEPCONSENSUS_v1pt2_2023_08_q20'\n",
    "\n",
    "# Get the Google billing project name and workspace name for current workspace\n",
    "PROJECT = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE =os.path.basename(os.path.dirname(os.getcwd()))\n",
    "bucket = os.environ['WORKSPACE_BUCKET'] + \"/\"\n",
    "\n",
    "# Verify that we've captured the environment variables\n",
    "print(\"Billing project: \" + PROJECT)\n",
    "print(\"Workspace: \" + WORKSPACE)\n",
    "print(\"Workspace storage bucket: \" + bucket)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read In tables as dataframes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Readstats table\n",
    "Find the individual readstats output files in the table and concatenate their contents.\n",
    "Add the hifi filename and sample name."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>1</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>mean</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>quartile_50</th>\n",
       "      <th>quartile_75</th>\n",
       "      <th>N25</th>\n",
       "      <th>N50</th>\n",
       "      <th>N75</th>\n",
       "      <th>filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2687625</td>\n",
       "      <td>56102007798</td>\n",
       "      <td>56.10</td>\n",
       "      <td>107</td>\n",
       "      <td>64413</td>\n",
       "      <td>20874</td>\n",
       "      <td>17762</td>\n",
       "      <td>20158</td>\n",
       "      <td>23398</td>\n",
       "      <td>18348</td>\n",
       "      <td>21041</td>\n",
       "      <td>24496</td>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2643186</td>\n",
       "      <td>52918724515</td>\n",
       "      <td>52.92</td>\n",
       "      <td>345</td>\n",
       "      <td>62341</td>\n",
       "      <td>20020</td>\n",
       "      <td>17277</td>\n",
       "      <td>19348</td>\n",
       "      <td>22242</td>\n",
       "      <td>17731</td>\n",
       "      <td>20061</td>\n",
       "      <td>23169</td>\n",
       "      <td>HG00099.m54329U_220827_143814.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>352327</td>\n",
       "      <td>7524316582</td>\n",
       "      <td>7.52</td>\n",
       "      <td>1028</td>\n",
       "      <td>57489</td>\n",
       "      <td>21356</td>\n",
       "      <td>19009</td>\n",
       "      <td>20965</td>\n",
       "      <td>23289</td>\n",
       "      <td>19422</td>\n",
       "      <td>21456</td>\n",
       "      <td>23918</td>\n",
       "      <td>HG00323.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>811437</td>\n",
       "      <td>17181739490</td>\n",
       "      <td>17.18</td>\n",
       "      <td>398</td>\n",
       "      <td>75864</td>\n",
       "      <td>21174</td>\n",
       "      <td>16931</td>\n",
       "      <td>20457</td>\n",
       "      <td>24597</td>\n",
       "      <td>18227</td>\n",
       "      <td>22011</td>\n",
       "      <td>26361</td>\n",
       "      <td>HG03017.m64076_221001_041132.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2319689</td>\n",
       "      <td>44900731287</td>\n",
       "      <td>44.90</td>\n",
       "      <td>351</td>\n",
       "      <td>70885</td>\n",
       "      <td>19356</td>\n",
       "      <td>15860</td>\n",
       "      <td>18511</td>\n",
       "      <td>22051</td>\n",
       "      <td>16646</td>\n",
       "      <td>19670</td>\n",
       "      <td>23521</td>\n",
       "      <td>HG03041.m54329U_211214_012740.dc.q20.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "1 total_reads     total_bp total_Gbp   min    max   mean quartile_25  \\\n",
       "2     2687625  56102007798     56.10   107  64413  20874       17762   \n",
       "2     2643186  52918724515     52.92   345  62341  20020       17277   \n",
       "2      352327   7524316582      7.52  1028  57489  21356       19009   \n",
       "2      811437  17181739490     17.18   398  75864  21174       16931   \n",
       "2     2319689  44900731287     44.90   351  70885  19356       15860   \n",
       "\n",
       "1 quartile_50 quartile_75    N25    N50    N75  \\\n",
       "2       20158       23398  18348  21041  24496   \n",
       "2       19348       22242  17731  20061  23169   \n",
       "2       20965       23289  19422  21456  23918   \n",
       "2       20457       24597  18227  22011  26361   \n",
       "2       18511       22051  16646  19670  23521   \n",
       "\n",
       "1                                       filename  \n",
       "2  HG00099.m54329U_220825_174247.dc.q20.fastq.gz  \n",
       "2  HG00099.m54329U_220827_143814.dc.q20.fastq.gz  \n",
       "2   HG00323.m64043_220728_173215.dc.q20.fastq.gz  \n",
       "2   HG03017.m64076_221001_041132.dc.q20.fastq.gz  \n",
       "2  HG03041.m54329U_211214_012740.dc.q20.fastq.gz  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# readstats\n",
    "readstats_df = table_to_dataframe(\"readstats\", workspace=WORKSPACE, workspace_namespace=PROJECT)\n",
    "qc_list = []\n",
    "\n",
    "for index, row in readstats_df.iterrows():\n",
    "\n",
    "        sample_readstats_fp = row['ReadStatsReport']\n",
    "        sample_readstats_fn = os.path.basename(sample_readstats_fp)\n",
    "        df = pd.read_csv(sample_readstats_fp, header=None, sep='\\t')\n",
    "        df = df[df[0]=='sample.fastq']\n",
    "        # add sample and file name\n",
    "        #df = df.append({0:'undef', 1:'sample', 2:row['sample']}, ignore_index=True)\n",
    "        #df = df.append({0:'undef', 1:'filename', 2:os.path.basename(row['hifi'])}, ignore_index=True)\n",
    "        new_row = pd.DataFrame({0:'undef', 1:'filename', 2:os.path.basename(row['hifi'])}, index=[0])\n",
    "        df = pd.concat([df, new_row], ignore_index=True)\n",
    "        # remove all but the variables (total_bp, quartile_25 etc)\n",
    "        df2 = df.drop(df.columns[[0, 1]], axis=1)\n",
    "        # make rownames ('total_bp', 'quartile_25' etc)\n",
    "        df2.index = df.iloc[:,1]\n",
    "        # remove the now meaningless 'sample.fastq' filename\n",
    "        df2 = df2.drop(index=['file'])\n",
    "        qc_list.append(df2)\n",
    "        \n",
    "\n",
    "# merge\n",
    "readstats_df = pd.concat(qc_list, axis=1).transpose()\n",
    "readstats_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Submitter metadata table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00099.m54329U_220827_143814.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099.m54329U_220829_095708.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00140.m64136_220715_182717.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename sample_ID  instrument_model  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz   HG00099  PacBio Sequel II   \n",
       "1  HG00099.m54329U_220827_143814.dc.q20.fastq.gz   HG00099  PacBio Sequel II   \n",
       "2  HG00099.m54329U_220829_095708.dc.q20.fastq.gz   HG00099  PacBio Sequel II   \n",
       "3   HG00140.m64043_220728_173215.dc.q20.fastq.gz   HG00140  PacBio Sequel II   \n",
       "4   HG00140.m64136_220715_182717.dc.q20.fastq.gz   HG00140  PacBio Sequel II   \n",
       "\n",
       "   shear_method size_selection  DeepConsensus_version polymerase_version  \\\n",
       "0  Megaruptor 3       PippinHT                    1.2                 P2   \n",
       "1  Megaruptor 3       PippinHT                    1.2                 P2   \n",
       "2  Megaruptor 3       PippinHT                    1.2                 P2   \n",
       "3  Megaruptor 3       PippinHT                    1.2                 P2   \n",
       "4  Megaruptor 3       PippinHT                    1.2                 P2   \n",
       "\n",
       "  seq_plate_chemistry_version        generator_facility generator_contact  \n",
       "0                          C2  University of Washington   kmiyamot@uw.edu  \n",
       "1                          C2  University of Washington   kmiyamot@uw.edu  \n",
       "2                          C2  University of Washington   kmiyamot@uw.edu  \n",
       "3                          C2  University of Washington   kmiyamot@uw.edu  \n",
       "4                          C2  University of Washington   kmiyamot@uw.edu  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_fp = os.path.join(bucket, submitter_metadata)\n",
    "metadata_df = pd.read_csv(meta_fp, sep='\\t')\n",
    "metadata_df = metadata_df[['filename', 'sample_ID', 'instrument_model', 'shear_method', 'size_selection', 'DeepConsensus_version', \n",
    "                           'polymerase_version', 'seq_plate_chemistry_version', 'generator_facility', 'generator_contact']]\n",
    "metadata_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SRA table\n",
    "Must download after finishing submission.\n",
    "The downloaded table needs to be split in one row per file (instead of filename, filename2 <...> filenameN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>accession</th>\n",
       "      <th>library_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>PG00099_1.HFSS_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>HG00140_lib1_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00280.m54329U_220901_221341.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545335</td>\n",
       "      <td>PG00280.HFSS_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00323.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545324</td>\n",
       "      <td>HG00323_lib1_dc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00408.m64136_211111_194404.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545313</td>\n",
       "      <td>HG00408_lib1_dc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename    accession  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz  SRR26545347   \n",
       "1   HG00140.m64043_220728_173215.dc.q20.fastq.gz  SRR26545346   \n",
       "2  HG00280.m54329U_220901_221341.dc.q20.fastq.gz  SRR26545335   \n",
       "3   HG00323.m64043_220728_173215.dc.q20.fastq.gz  SRR26545324   \n",
       "4   HG00408.m64136_211111_194404.dc.q20.fastq.gz  SRR26545313   \n",
       "\n",
       "          library_ID  \n",
       "0  PG00099_1.HFSS_dc  \n",
       "1    HG00140_lib1_dc  \n",
       "2    PG00280.HFSS_dc  \n",
       "3    HG00323_lib1_dc  \n",
       "4    HG00408_lib1_dc  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Read in SRA file and split by filename\n",
    "\n",
    "sra_file = os.path.join(bucket + sra_metadata)\n",
    "sample_df = pd.read_csv(sra_file, sep='\\t')\n",
    "\n",
    "fnames = [x for x in sample_df.columns if x.startswith('filename')]\n",
    "sra_df = pd.melt(sample_df, id_vars =['library_ID','accession'], value_vars = fnames, value_name='file')\n",
    "sra_df= sra_df.dropna()\n",
    "sra_df = sra_df[['file', 'accession', 'library_ID']]\n",
    "sra_df.columns = ['filename', 'accession', 'library_ID']\n",
    "sra_df.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge all dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(168, 3) (168, 13) (168, 10)\n"
     ]
    }
   ],
   "source": [
    "print(sra_df.shape, readstats_df.shape, metadata_df.shape)\n",
    "# two samples of HG02572 were not uploaded to SRA with the rest of this batch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(readstats_df['filename']) - set(sra_df['filename'])\n",
    "#m54329U_201103_231616.ccs.bam (HG002 sample) is not present in the submitter metadata or on SRA, but it is in the submissions "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>accession</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>mean</th>\n",
       "      <th>quartile_25</th>\n",
       "      <th>...</th>\n",
       "      <th>N75</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>PG00099_1.HFSS_dc</td>\n",
       "      <td>2687625</td>\n",
       "      <td>56102007798</td>\n",
       "      <td>56.10</td>\n",
       "      <td>107</td>\n",
       "      <td>64413</td>\n",
       "      <td>20874</td>\n",
       "      <td>17762</td>\n",
       "      <td>...</td>\n",
       "      <td>24496</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>HG00140_lib1_dc</td>\n",
       "      <td>572695</td>\n",
       "      <td>11345594738</td>\n",
       "      <td>11.35</td>\n",
       "      <td>117</td>\n",
       "      <td>55287</td>\n",
       "      <td>19810</td>\n",
       "      <td>17863</td>\n",
       "      <td>...</td>\n",
       "      <td>21783</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00280.m54329U_220901_221341.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545335</td>\n",
       "      <td>PG00280.HFSS_dc</td>\n",
       "      <td>2766007</td>\n",
       "      <td>55104620054</td>\n",
       "      <td>55.10</td>\n",
       "      <td>91</td>\n",
       "      <td>66666</td>\n",
       "      <td>19922</td>\n",
       "      <td>16987</td>\n",
       "      <td>...</td>\n",
       "      <td>23253</td>\n",
       "      <td>HG00280</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00323.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545324</td>\n",
       "      <td>HG00323_lib1_dc</td>\n",
       "      <td>352327</td>\n",
       "      <td>7524316582</td>\n",
       "      <td>7.52</td>\n",
       "      <td>1028</td>\n",
       "      <td>57489</td>\n",
       "      <td>21356</td>\n",
       "      <td>19009</td>\n",
       "      <td>...</td>\n",
       "      <td>23918</td>\n",
       "      <td>HG00323</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00408.m64136_211111_194404.dc.q20.fastq.gz</td>\n",
       "      <td>SRR26545313</td>\n",
       "      <td>HG00408_lib1_dc</td>\n",
       "      <td>2333870</td>\n",
       "      <td>44812028504</td>\n",
       "      <td>44.81</td>\n",
       "      <td>83</td>\n",
       "      <td>61297</td>\n",
       "      <td>19200</td>\n",
       "      <td>16485</td>\n",
       "      <td>...</td>\n",
       "      <td>22353</td>\n",
       "      <td>HG00408</td>\n",
       "      <td>PacBio Sequel II</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename    accession  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz  SRR26545347   \n",
       "1   HG00140.m64043_220728_173215.dc.q20.fastq.gz  SRR26545346   \n",
       "2  HG00280.m54329U_220901_221341.dc.q20.fastq.gz  SRR26545335   \n",
       "3   HG00323.m64043_220728_173215.dc.q20.fastq.gz  SRR26545324   \n",
       "4   HG00408.m64136_211111_194404.dc.q20.fastq.gz  SRR26545313   \n",
       "\n",
       "          library_ID total_reads     total_bp total_Gbp   min    max   mean  \\\n",
       "0  PG00099_1.HFSS_dc     2687625  56102007798     56.10   107  64413  20874   \n",
       "1    HG00140_lib1_dc      572695  11345594738     11.35   117  55287  19810   \n",
       "2    PG00280.HFSS_dc     2766007  55104620054     55.10    91  66666  19922   \n",
       "3    HG00323_lib1_dc      352327   7524316582      7.52  1028  57489  21356   \n",
       "4    HG00408_lib1_dc     2333870  44812028504     44.81    83  61297  19200   \n",
       "\n",
       "  quartile_25  ...    N75 sample_ID  instrument_model  shear_method  \\\n",
       "0       17762  ...  24496   HG00099  PacBio Sequel II  Megaruptor 3   \n",
       "1       17863  ...  21783   HG00140  PacBio Sequel II  Megaruptor 3   \n",
       "2       16987  ...  23253   HG00280  PacBio Sequel II  Megaruptor 3   \n",
       "3       19009  ...  23918   HG00323  PacBio Sequel II  Megaruptor 3   \n",
       "4       16485  ...  22353   HG00408  PacBio Sequel II  Megaruptor 3   \n",
       "\n",
       "  size_selection DeepConsensus_version polymerase_version  \\\n",
       "0       PippinHT                   1.2                 P2   \n",
       "1       PippinHT                   1.2                 P2   \n",
       "2       PippinHT                   1.2                 P2   \n",
       "3       PippinHT                   1.2                 P2   \n",
       "4       PippinHT                   1.2                 P2   \n",
       "\n",
       "  seq_plate_chemistry_version        generator_facility  generator_contact  \n",
       "0                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "1                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "2                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "3                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "4                          C2  University of Washington    kmiyamot@uw.edu  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(\n",
    "    sra_df,\n",
    "    readstats_df,\n",
    "    on='filename')\n",
    "merged_df = pd.merge(\n",
    "    merged_df,\n",
    "    metadata_df,\n",
    "    on='filename')\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write output files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Create table\n",
    "#dataframe_to_table(\"post_sra_metadata\", merged_df, WORKSPACE, PROJECT)\n",
    "outf = os.path.join(bucket, submission_id + '_post_sra_metadata.tsv')\n",
    "merged_df.to_csv(outf, sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(132, 24)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
