{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "74c04205-1d11-4b06-8f15-29c4e6fdf0f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import collections"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4330ab11-d06e-4aa0-b617-0e722486d572",
   "metadata": {},
   "source": [
    "# Wrangling Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dcabf311-295e-47dd-b8a6-5de0f97109ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_missing_1_submitter(sample_df, required_1_submitter_columns):\n",
    "    \n",
    "    require_1_submitter_dict = {sample: {key: None for key in required_1_submitter_columns} for sample in sample_df['sample_ID'].tolist()}\n",
    "    \n",
    "    for sample in list(set(sample_df['sample_ID'].tolist())):\n",
    "        sample_array = sample_df[sample_df['sample_ID'].isin([sample])][required_1_submitter_columns].values\n",
    "\n",
    "        assert len(required_1_submitter_columns) == sample_array.shape[1]\n",
    "\n",
    "        for column_index in range(sample_array.shape[1]):\n",
    "\n",
    "            if np.all(sample_array[:, column_index] == sample_array[0, column_index]):\n",
    "                require_1_submitter_dict[sample][required_1_submitter_columns[column_index]] = sample_array[0, column_index]\n",
    "            else:\n",
    "                require_1_submitter_dict[sample][required_1_submitter_columns[column_index]] = sample_array[:, column_index]\n",
    "    \n",
    "    samples_missing_1_submitter = _samples_missing_1_submitter(require_1_submitter_dict)\n",
    "    \n",
    "    submitter_df = pd.DataFrame.from_dict(require_1_submitter_dict, orient='index')\n",
    "    \n",
    "    return submitter_df, list(set(samples_missing_1_submitter))\n",
    "\n",
    "\n",
    "def _samples_missing_1_submitter(require_1_submitter_dict):\n",
    "    \n",
    "    samples_missing_1_submitter = []\n",
    "    for sample_id, meta_dict in require_1_submitter_dict.items():\n",
    "        for key, value in meta_dict.items():\n",
    "            if isinstance(value, np.ndarray):\n",
    "                if any(item == 'nan' or pd.isna(item) for item in value):\n",
    "                    if key == 'notes':\n",
    "                        pass\n",
    "                    else:\n",
    "                        samples_missing_1_submitter.append(sample_id)\n",
    "    return samples_missing_1_submitter\n",
    "\n",
    "\n",
    "def sample_aggregate_5_readstats(sample_df, required_readstat_columns, readstat_dict):\n",
    "\n",
    "    sample_5_readstats_df = sample_df[required_readstat_columns]\n",
    "    sample_5_readstats_df = sample_5_readstats_df.groupby('sample_ID').agg(readstat_dict).reset_index()\n",
    "    \n",
    "    return sample_5_readstats_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51ae2473-354f-4a51-b767-84d600536ece",
   "metadata": {},
   "source": [
    "# ONT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "da2337b0-a4f9-4dd2-857e-0075b3576f29",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_df = pd.read_csv('sample-files/hprc_metadata_sample_files_ONT.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "840d6350-7ebd-41d4-82d8-0bf18a77f8e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if ont_df['notes'].isnull().all() == True:\n",
    "    ont_df = ont_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "b737011d-c7db-4a50-8a72-0798a64f612c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: Integrate with HPRC_metadata/merge_metadata.py combine_readstats.keep_columns\n",
    "required_ONT_1_submitter_columns = ['filename',\n",
    "                                    'sample_ID',\n",
    "                                    'path',\n",
    "                                    'filetype',\n",
    "                                    'library_ID',\n",
    "                                    'library_strategy',\n",
    "                                    'library_source',\n",
    "                                    'library_selection',\n",
    "                                    'library_layout',\n",
    "                                    'platform',\n",
    "                                    'instrument_model',\n",
    "                                    'design_description',\n",
    "                                    'data_type',\n",
    "                                    'shear_method',\n",
    "                                    'size_selection',\n",
    "                                    'seq_kit',\n",
    "                                    'basecaller',\n",
    "                                    'basecaller_version',\n",
    "                                    'basecaller_model',\n",
    "                                    'generator_facility',\n",
    "                                    'generator_contact']\n",
    "\n",
    "required_ONT_1_submitter_columns_df = pd.DataFrame(required_ONT_1_submitter_columns, columns=['1_submitter_columns'])\n",
    "required_ONT_1_submitter_columns_df.to_csv('aggregate-sample-inputs/hprc_1_submitter_columns_ONT.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "40ad1629-15f2-4b1c-b800-8612f293394d",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_ONT_5_readstats_columns = ['sample_ID', \n",
    "                                    'filename',\n",
    "                                    'read_N50',\n",
    "                                    'Gb',\n",
    "                                    'coverage',\n",
    "                                    '100kb+',\n",
    "                                    '200kb+',\n",
    "                                    '300kb+',\n",
    "                                    '400kb+',\n",
    "                                    '500kb+',\n",
    "                                    '1Mb+',\n",
    "                                    'whales']\n",
    "required_ONT_5_readstats_columns  = pd.DataFrame(required_ONT_5_readstats_columns , columns=['5_readstats_columns'])\n",
    "required_ONT_5_readstats_columns.to_csv('aggregate-sample-inputs/hprc_5_readstats_columns_ONT.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "82045141-7fbd-4496-970e-9c2cc5888519",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_ONT_5_readstat_dict = {'read_N50': 'mean',\n",
    "                                'coverage':'sum',\n",
    "                                '100kb+': 'sum',\n",
    "                                '200kb+': 'sum',\n",
    "                                '300kb+': 'sum',\n",
    "                                '400kb+': 'sum',\n",
    "                                '500kb+': 'sum',\n",
    "                                '1Mb+': 'sum',\n",
    "                                'whales': 'sum'}\n",
    "\n",
    "pd.DataFrame.from_dict(required_ONT_5_readstat_dict,orient='index').to_csv('aggregate-sample-inputs/hprc_5_readstat_sample_aggregate_rules_ONT.tsv',sep='\\t')\n",
    "# pd.read_csv('hprc_5_readstat_sample_aggregate_rules.tsv',sep='\\t', index_col=[0]).to_dict()['0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "0d577821-a538-4168-85ec-18914b1742bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_1_submitter_df, samples_missing_1_submmiter_ont = check_missing_1_submitter(ont_df, required_ONT_1_submitter_columns)\n",
    "if len(samples_missing_1_submmiter_ont) > 0:\n",
    "    ont_1_submitter_df[ont_1_submitter_df['sample_ID'].isin(samples_missing_1_submmiter_ont)].to_csv('aggregate-sample-inputs/hprc_metadata_sample_files_missing_1_submitter_ONT.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "d2c503a1-d8d1-411e-9591-0a7cdd924dfb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample_df, required_readstat_columns\n",
    "ont_5_readstats_df = sample_aggregate_5_readstats(ont_df, required_ONT_5_readstats_columns, required_ONT_5_readstat_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "bd2ded1f-b9f6-416e-8215-d9d6282faa9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df = pd.merge(ont_1_submitter_df, ont_5_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "66273dce-b7a0-4b0d-b61c-08185ce4944f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(124, 30)"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_sample_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "1b6ef918-e548-43ab-80b9-0e2895678530",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "124"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(ont_sample_df['sample_ID'].tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "9219c04a-4e68-4105-b459-04697d8f0fbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_low_coverage_sample_df = ont_sample_df[ont_sample_df['100kb+'] <= 30.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "c784ea43-f7f0-4529-9600-e3deac7272a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>filetype</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>...</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>read_N50</th>\n",
       "      <th>coverage</th>\n",
       "      <th>100kb+</th>\n",
       "      <th>200kb+</th>\n",
       "      <th>300kb+</th>\n",
       "      <th>400kb+</th>\n",
       "      <th>500kb+</th>\n",
       "      <th>1Mb+</th>\n",
       "      <th>whales</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03834</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03834/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[02_08_22_R941_HG03834_1, 02_08_22_R941_HG0383...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>107647.333333</td>\n",
       "      <td>48.80</td>\n",
       "      <td>26.33</td>\n",
       "      <td>8.97</td>\n",
       "      <td>3.23</td>\n",
       "      <td>1.35</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.03</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[02_15_22_R941_HG03041_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03041</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03041/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[02_15_22_R941_HG03041_1, 02_15_22_R941_HG0304...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>92468.000000</td>\n",
       "      <td>59.41</td>\n",
       "      <td>27.28</td>\n",
       "      <td>7.89</td>\n",
       "      <td>2.09</td>\n",
       "      <td>0.65</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.01</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>[02_15_22_R941_HG03050_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03050</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03050/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[02_15_22_R941_HG03050_1, 02_15_22_R941_HG0305...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>92270.666667</td>\n",
       "      <td>60.55</td>\n",
       "      <td>27.88</td>\n",
       "      <td>7.14</td>\n",
       "      <td>1.54</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.00</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[02_15_22_R941_HG03239_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03239</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03239/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[02_15_22_R941_HG03239_1, 02_15_22_R941_HG0323...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>82147.000000</td>\n",
       "      <td>64.02</td>\n",
       "      <td>25.48</td>\n",
       "      <td>5.39</td>\n",
       "      <td>1.03</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>[04_12_22_R941_HG03195_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03195</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03195/r...</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>[04_12_22_R941_HG03195_1, 04_12_22_R941_HG0319...</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan, nan]</td>\n",
       "      <td>79603.500000</td>\n",
       "      <td>47.55</td>\n",
       "      <td>18.08</td>\n",
       "      <td>3.57</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.02</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>[05_17_22_R941_HG03130_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03130</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03130/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[05_17_22_R941_HG03130_1, 05_17_22_R941_HG0313...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>77859.000000</td>\n",
       "      <td>58.86</td>\n",
       "      <td>21.29</td>\n",
       "      <td>3.12</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>[05_17_22_R941_HG03139_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03139</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03139/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[05_17_22_R941_HG03139_1, 05_17_22_R941_HG0313...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>87795.333333</td>\n",
       "      <td>57.65</td>\n",
       "      <td>24.80</td>\n",
       "      <td>5.31</td>\n",
       "      <td>1.03</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.00</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>[05_17_22_R941_HG03209_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03209</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03209/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[05_17_22_R941_HG03209_1, 05_17_22_R941_HG0320...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>84171.333333</td>\n",
       "      <td>70.94</td>\n",
       "      <td>29.79</td>\n",
       "      <td>6.57</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>[05_24_22_R941_HG03458_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03458</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03458/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[05_24_22_R941_HG03458_1, 05_24_22_R941_HG0345...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>82253.000000</td>\n",
       "      <td>68.27</td>\n",
       "      <td>27.82</td>\n",
       "      <td>7.42</td>\n",
       "      <td>1.58</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>[08_10_22_R941_HG03792_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>HG03792</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03792/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[08_10_22_R941_HG03792_1, 08_10_22_R941_HG0379...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>57058.000000</td>\n",
       "      <td>73.05</td>\n",
       "      <td>17.48</td>\n",
       "      <td>2.69</td>\n",
       "      <td>0.38</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>[09_27_2022_R941_HG03742_1_Guppy_6.5.7_450bps_...</td>\n",
       "      <td>HG03742</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG03742/r...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[09_27_2022_R941_HG03742_1, 09_27_2022_R941_HG...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>...</td>\n",
       "      <td>[nan, nan, nan]</td>\n",
       "      <td>76131.333333</td>\n",
       "      <td>59.89</td>\n",
       "      <td>22.38</td>\n",
       "      <td>5.74</td>\n",
       "      <td>1.22</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              filename sample_ID  \\\n",
       "3    [02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mo...   HG03834   \n",
       "6    [02_15_22_R941_HG03041_1_Guppy_6.5.7_450bps_mo...   HG03041   \n",
       "7    [02_15_22_R941_HG03050_1_Guppy_6.5.7_450bps_mo...   HG03050   \n",
       "8    [02_15_22_R941_HG03239_1_Guppy_6.5.7_450bps_mo...   HG03239   \n",
       "19   [04_12_22_R941_HG03195_1_Guppy_6.5.7_450bps_mo...   HG03195   \n",
       "22   [05_17_22_R941_HG03130_1_Guppy_6.5.7_450bps_mo...   HG03130   \n",
       "23   [05_17_22_R941_HG03139_1_Guppy_6.5.7_450bps_mo...   HG03139   \n",
       "24   [05_17_22_R941_HG03209_1_Guppy_6.5.7_450bps_mo...   HG03209   \n",
       "28   [05_24_22_R941_HG03458_1_Guppy_6.5.7_450bps_mo...   HG03458   \n",
       "89   [08_10_22_R941_HG03792_1_Guppy_6.5.7_450bps_mo...   HG03792   \n",
       "116  [09_27_2022_R941_HG03742_1_Guppy_6.5.7_450bps_...   HG03742   \n",
       "\n",
       "                                                  path              filetype  \\\n",
       "3    [s3://human-pangenomics/working/HPRC/HG03834/r...       [nan, nan, nan]   \n",
       "6    [s3://human-pangenomics/working/HPRC/HG03041/r...       [nan, nan, nan]   \n",
       "7    [s3://human-pangenomics/working/HPRC/HG03050/r...       [nan, nan, nan]   \n",
       "8    [s3://human-pangenomics/working/HPRC/HG03239/r...       [nan, nan, nan]   \n",
       "19   [s3://human-pangenomics/working/HPRC/HG03195/r...  [nan, nan, nan, nan]   \n",
       "22   [s3://human-pangenomics/working/HPRC/HG03130/r...       [nan, nan, nan]   \n",
       "23   [s3://human-pangenomics/working/HPRC/HG03139/r...       [nan, nan, nan]   \n",
       "24   [s3://human-pangenomics/working/HPRC/HG03209/r...       [nan, nan, nan]   \n",
       "28   [s3://human-pangenomics/working/HPRC/HG03458/r...       [nan, nan, nan]   \n",
       "89   [s3://human-pangenomics/working/HPRC/HG03792/r...       [nan, nan, nan]   \n",
       "116  [s3://human-pangenomics/working/HPRC/HG03742/r...       [nan, nan, nan]   \n",
       "\n",
       "                                            library_ID      library_strategy  \\\n",
       "3    [02_08_22_R941_HG03834_1, 02_08_22_R941_HG0383...       [nan, nan, nan]   \n",
       "6    [02_15_22_R941_HG03041_1, 02_15_22_R941_HG0304...       [nan, nan, nan]   \n",
       "7    [02_15_22_R941_HG03050_1, 02_15_22_R941_HG0305...       [nan, nan, nan]   \n",
       "8    [02_15_22_R941_HG03239_1, 02_15_22_R941_HG0323...       [nan, nan, nan]   \n",
       "19   [04_12_22_R941_HG03195_1, 04_12_22_R941_HG0319...  [nan, nan, nan, nan]   \n",
       "22   [05_17_22_R941_HG03130_1, 05_17_22_R941_HG0313...       [nan, nan, nan]   \n",
       "23   [05_17_22_R941_HG03139_1, 05_17_22_R941_HG0313...       [nan, nan, nan]   \n",
       "24   [05_17_22_R941_HG03209_1, 05_17_22_R941_HG0320...       [nan, nan, nan]   \n",
       "28   [05_24_22_R941_HG03458_1, 05_24_22_R941_HG0345...       [nan, nan, nan]   \n",
       "89   [08_10_22_R941_HG03792_1, 08_10_22_R941_HG0379...       [nan, nan, nan]   \n",
       "116  [09_27_2022_R941_HG03742_1, 09_27_2022_R941_HG...       [nan, nan, nan]   \n",
       "\n",
       "           library_source     library_selection        library_layout  \\\n",
       "3         [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "6         [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "7         [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "8         [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "19   [nan, nan, nan, nan]  [nan, nan, nan, nan]  [nan, nan, nan, nan]   \n",
       "22        [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "23        [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "24        [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "28        [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "89        [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "116       [nan, nan, nan]       [nan, nan, nan]       [nan, nan, nan]   \n",
       "\n",
       "                 platform  ...     generator_contact       read_N50 coverage  \\\n",
       "3         [nan, nan, nan]  ...       [nan, nan, nan]  107647.333333    48.80   \n",
       "6         [nan, nan, nan]  ...       [nan, nan, nan]   92468.000000    59.41   \n",
       "7         [nan, nan, nan]  ...       [nan, nan, nan]   92270.666667    60.55   \n",
       "8         [nan, nan, nan]  ...       [nan, nan, nan]   82147.000000    64.02   \n",
       "19   [nan, nan, nan, nan]  ...  [nan, nan, nan, nan]   79603.500000    47.55   \n",
       "22        [nan, nan, nan]  ...       [nan, nan, nan]   77859.000000    58.86   \n",
       "23        [nan, nan, nan]  ...       [nan, nan, nan]   87795.333333    57.65   \n",
       "24        [nan, nan, nan]  ...       [nan, nan, nan]   84171.333333    70.94   \n",
       "28        [nan, nan, nan]  ...       [nan, nan, nan]   82253.000000    68.27   \n",
       "89        [nan, nan, nan]  ...       [nan, nan, nan]   57058.000000    73.05   \n",
       "116       [nan, nan, nan]  ...       [nan, nan, nan]   76131.333333    59.89   \n",
       "\n",
       "    100kb+ 200kb+ 300kb+ 400kb+ 500kb+  1Mb+ whales  \n",
       "3    26.33   8.97   3.23   1.35   0.64  0.03     99  \n",
       "6    27.28   7.89   2.09   0.65   0.25  0.01     32  \n",
       "7    27.88   7.14   1.54   0.40   0.13  0.00     15  \n",
       "8    25.48   5.39   1.03   0.29   0.12  0.00     16  \n",
       "19   18.08   3.57   0.78   0.27   0.14  0.02     46  \n",
       "22   21.29   3.12   0.30   0.05   0.02  0.00      2  \n",
       "23   24.80   5.31   1.03   0.29   0.12  0.00     25  \n",
       "24   29.79   6.57   1.05   0.14   0.02  0.00      1  \n",
       "28   27.82   7.42   1.58   0.30   0.05  0.00      0  \n",
       "89   17.48   2.69   0.38   0.06   0.02  0.00      1  \n",
       "116  22.38   5.74   1.22   0.28   0.08  0.00     18  \n",
       "\n",
       "[11 rows x 30 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_low_coverage_sample_df[ont_low_coverage_sample_df['sample_ID'].isin(list(set(samples_missing_1_submmiter_ont)))]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2dcf851-8ec2-4b4b-82e3-d1ff10548ed3",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "### Remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "60c2ac04-0e46-4d3b-90c4-5f02a5c2f517",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14\n"
     ]
    }
   ],
   "source": [
    "ont_sample_missing_1_submitter_df = ont_df[ont_df.isnull().any(axis=1)]\n",
    "\n",
    "# Number of ONT samples that have at least one missing submitter file missing\n",
    "print(len(set(ont_df[ont_df['sample_ID'].isin(ont_sample_missing_1_submitter_df['sample_ID'].tolist())]['sample_ID'].tolist())))\n",
    "\n",
    "# Export ONT sample files that have at least one missing submitter file missing\n",
    "ont_df[ont_df['sample_ID'].isin(ont_sample_missing_1_submitter_df['sample_ID'].tolist())].reset_index(drop=True).to_csv('hprc_metadata_sample_files_missing_1_submitter_ONT.tsv', sep='\\t')\n",
    "\n",
    "for nan_column in ont_df_nan_columns:\n",
    "    ont_df[nan_column] = ont_df[nan_column].ffill().tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "39efbb45-d23e-4205-8248-c8cd35a1e589",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_meta_df = ont_df[ont_meta_columns].drop_duplicates(subset=['sample_ID'])\n",
    "ont_sample_readstats_df = ont_df[ont_readstat_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "647365b5-1cb9-49ef-ae19-9fc744499f14",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check no missing sample_ID for the sample meta and readstat subset\n",
    "assert ont_df[~ont_df['sample_ID'].isin(ont_sample_meta_df['sample_ID'].tolist())].shape[0] == ont_df[~ont_df['sample_ID'].isin(ont_sample_readstats_df['sample_ID'].tolist())].shape[0] == 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7d632452-8890-4f6d-8db9-4a0b7a5a3eb5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_readstats_df = ont_sample_readstats_df.groupby('sample_ID').agg({'read_N50': 'mean',\n",
    "                                                                          'coverage':'sum',\n",
    "                                                                          '100kb+': 'sum',\n",
    "                                                                          '200kb+': 'sum',\n",
    "                                                                          '300kb+': 'sum',\n",
    "                                                                          '400kb+': 'sum',\n",
    "                                                                          '500kb+': 'sum',\n",
    "                                                                          '1Mb+': 'sum',\n",
    "                                                                          'whales': 'sum'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d4a230a1-5044-483e-beff-036478edd6a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df = pd.merge(ont_sample_meta_df, ont_sample_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0328faca-30c7-4fcb-b488-672aae9ed04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "assert len(set(ont_sample_df['sample_ID'].tolist())) == ont_sample_df.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ef949b29-e2dd-484c-acf2-20a7d8497b3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df.to_csv('hprc_metadata_sample_aggregate_ONT.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c195e0ba-6b3b-49ae-816d-80c1a6c5dd1b",
   "metadata": {},
   "source": [
    "# HiFi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "212775d8-d606-4918-8e71-eb7e294fbbf8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>accession</th>\n",
       "      <th>study</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>...</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>ccs_algorithm</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>m54329U_200124_193652.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1596347</td>\n",
       "      <td>27122049640</td>\n",
       "      <td>27.12</td>\n",
       "      <td>47</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>m54329U_200127_180554.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1755465</td>\n",
       "      <td>29883779569</td>\n",
       "      <td>29.88</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>m54329U_200129_001928.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1779732</td>\n",
       "      <td>30200500419</td>\n",
       "      <td>30.20</td>\n",
       "      <td>45</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>m54329U_200130_064539.ccs.bam</td>\n",
       "      <td>HG01891</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01891/ra...</td>\n",
       "      <td>SRR13684280</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861236</td>\n",
       "      <td>1636421</td>\n",
       "      <td>27919782706</td>\n",
       "      <td>27.92</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 18kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>m54329U_200201_051510.ccs.bam</td>\n",
       "      <td>HG01123</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01123/ra...</td>\n",
       "      <td>SRR13684290</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN17861232</td>\n",
       "      <td>2072143</td>\n",
       "      <td>34554070679</td>\n",
       "      <td>34.55</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 17kb fractionated gDNA</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>g-TUBE</td>\n",
       "      <td>SageELF</td>\n",
       "      <td>ccs 4.0.0 (commit SL-release-8.0.0)</td>\n",
       "      <td>P2.0</td>\n",
       "      <td>C2.0</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        filename sample_ID  \\\n",
       "0  m54329U_200124_193652.ccs.bam   HG01891   \n",
       "1  m54329U_200127_180554.ccs.bam   HG01891   \n",
       "2  m54329U_200129_001928.ccs.bam   HG01891   \n",
       "3  m54329U_200130_064539.ccs.bam   HG01891   \n",
       "4  m54329U_200201_051510.ccs.bam   HG01123   \n",
       "\n",
       "                                                path    accession      study  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "1  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "2  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "3  s3://human-pangenomics/working/HPRC/HG01891/ra...  SRR13684280  SRP305758   \n",
       "4  s3://human-pangenomics/working/HPRC/HG01123/ra...  SRR13684290  SRP305758   \n",
       "\n",
       "  biosample_accession  total_reads     total_bp  total_Gbp  min  ...  \\\n",
       "0        SAMN17861236      1596347  27122049640      27.12   47  ...   \n",
       "1        SAMN17861236      1755465  29883779569      29.88   46  ...   \n",
       "2        SAMN17861236      1779732  30200500419      30.20   45  ...   \n",
       "3        SAMN17861236      1636421  27919782706      27.92   46  ...   \n",
       "4        SAMN17861232      2072143  34554070679      34.55   46  ...   \n",
       "\n",
       "                          design_description        data_type  shear_method  \\\n",
       "0  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "1  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "2  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "3  HiFi sequencing of 18kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "4  HiFi sequencing of 17kb fractionated gDNA  unaligned reads        g-TUBE   \n",
       "\n",
       "   size_selection                        ccs_algorithm  polymerase_version  \\\n",
       "0         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "1         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "2         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "3         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "4         SageELF  ccs 4.0.0 (commit SL-release-8.0.0)                P2.0   \n",
       "\n",
       "   seq_plate_chemistry_version        generator_facility generator_contact  \\\n",
       "0                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "1                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "2                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "3                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "4                         C2.0  University of Washington   kmiyamot@uw.edu   \n",
       "\n",
       "  notes  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_df = pd.read_csv('sample-files/hprc_metadata_sample_files_HiFi.tsv', sep='\\t')\n",
    "hifi_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "23991c33-88eb-48cc-8375-3c4b4a3e9fc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if hifi_df['notes'].isnull().all() == True:\n",
    "    hifi_df = hifi_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1b35b613-a4aa-4042-8db2-7345dcdab51f",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_hifi_1_submitter_columns = ['sample_ID', 'filename', 'library_ID', 'library_strategy', 'library_source', 'library_selection', 'library_layout', 'platform', 'instrument_model', \n",
    "                    'design_description', 'data_type', 'shear_method', 'size_selection', 'ccs_algorithm', 'polymerase_version', 'seq_plate_chemistry_version', \n",
    "                    'generator_facility', 'generator_contact', 'notes']\n",
    "\n",
    "required_hifi_5_readstats_columns = ['sample_ID','filename', 'total_reads', 'total_bp', 'total_Gbp', 'min', 'max', 'mean', 'quartile_25', 'quartile_50', 'quartile_75', 'N25', 'N50', 'N75']\n",
    "\n",
    "required_hifi_5_readstat_dict = {'total_reads': 'sum',\n",
    "                 'total_bp': 'sum',\n",
    "                 'total_Gbp': 'sum',\n",
    "                 'min': 'mean',\n",
    "                 'max': 'mean',\n",
    "                 'quartile_25': 'mean',\n",
    "                 'quartile_50': 'mean',\n",
    "                 'quartile_75': 'mean',\n",
    "                 'N25': 'mean',\n",
    "                 'N50': 'mean', # assembly term shortest contig of the total assembly length (different for ONT)\n",
    "                 'N75': 'mean'}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3c3706c9-9ae5-4bdb-bd89-cf9a6a6ab660",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_1_submitter_df, samples_missing_1_submmiter_hifi = check_missing_1_submitter(hifi_df, required_hifi_1_submitter_columns)\n",
    "hifi_1_submitter_df[hifi_1_submitter_df['sample_ID'].isin(samples_missing_1_submmiter_hifi)].to_csv('hprc_metadata_sample_files_missing_1_submitter_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7aab1fd3-58db-4c6b-8259-daa83a074683",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample_df, required_readstat_columns\n",
    "hifi_5_readstats_df = sample_aggregate_5_readstats(hifi_df, required_hifi_5_readstats_columns, readstat_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b17aa307-2878-4a80-ae90-47e874e2f712",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df = pd.merge(hifi_1_submitter_df, hifi_5_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "448de880-8599-4ba3-875d-613c1fab9940",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check that sample's with missing 1_submitter metadata are present\n",
    "assert hifi_sample_df[hifi_sample_df['sample_ID'].isin(samples_missing_1_submmiter_hifi)].shape[0] == len(set(samples_missing_1_submmiter_hifi))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9e478ebb-15fb-4db2-9911-3181ebacebc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df.to_csv('hprc_metadata_sample_aggregate_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa45e42e-d5ba-42e4-887c-34e74d45e97b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "### Remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "8f54dd29-ba81-4462-b686-93c60ac63e1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "require_hifi_1_submitter_dict = {sample: {key: None for key in required_hifi_1_submitter_columns} for sample in hifi_df['sample_ID'].tolist()} \n",
    "for sample in list(set(hifi_df['sample_ID'].tolist())):\n",
    "    # print(sample, required_hifi_1_submitter_columns)\n",
    "    \n",
    "    sample_array = hifi_df[hifi_df['sample_ID'].isin([sample])][required_hifi_1_submitter_columns].values\n",
    "    \n",
    "    # print(sample)\n",
    "    # print(sample_array)\n",
    "    # print(len(required_hifi_1_submitter_columns))\n",
    "    \n",
    "    assert len(required_hifi_1_submitter_columns) == sample_array.shape[1]\n",
    "    \n",
    "    for column_index in range(sample_array.shape[1]):\n",
    "        # print(sample_array[0, column_index], sample_array[:, column_index])\n",
    "        if np.all(sample_array[:, column_index] == sample_array[0, column_index]):\n",
    "            # print(sample_array[0, column_index], \n",
    "            #       required_hifi_1_submitter_columns[column_index], \n",
    "            #       sample_array[:, column_index].tolist()[0])\n",
    "            # print('\\n')\n",
    "            require_hifi_1_submitter_dict[sample][required_hifi_1_submitter_columns[column_index]] = sample_array[0, column_index]\n",
    "        else:\n",
    "            # print(sample_array[0, column_index])\n",
    "            # print(required_hifi_1_submitter_columns,column_index)\n",
    "            # print(required_hifi_1_submitter_columns[column_index])\n",
    "            require_hifi_1_submitter_dict[sample][required_hifi_1_submitter_columns[column_index]] = sample_array[:, column_index]\n",
    "            # print(required_hifi_1_submitter_columns[column_index],\n",
    "                  # sample_array[:, column_index])\n",
    "            # print('\\n')\n",
    "            # pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "900a19f4-4844-42fc-82a7-6e53e537836c",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples_missing_submmiter_1_hifi = []\n",
    "for sample_id, meta_dict in require_hifi_1_submitter_dict.items():\n",
    "    # print(sample_id, meta_dict)\n",
    "    for key, value in meta_dict.items():\n",
    "        if isinstance(value, np.ndarray):\n",
    "            if any(item == 'nan' or pd.isna(item) for item in value):\n",
    "                if key == 'notes':\n",
    "                    pass\n",
    "                else:\n",
    "                    samples_missing_submmiter_1_hifi.append(sample_id)\n",
    "hifi_1_submitter_df = pd.DataFrame.from_dict(require_hifi_1_submitter_dict, orient='index')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23fc00dd-f44b-48c2-9276-31fc584068cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_1_submitter_df[hifi_1_submitter_df['sample_ID'].isin(samples_missing_submmiter_1_hifi)].to_csv('hprc_metadata_sample_files_missing_1_submitter_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "051a5a1f-6913-468c-bdfb-8376cdd36231",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Retired\n",
    "# hifi_sample_meta_df = hifi_df[required_hifi_1_submitter_columns].drop_duplicates(subset=['sample_ID']).reset_index(drop=True)\n",
    "# hifi_sample_missing_1_submitter_list = []\n",
    "# for sample in hifi_sample_meta_df['sample_ID'].tolist():\n",
    "#     if sum(hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin([sample])].isna().any()) == 0:\n",
    "#         pass\n",
    "#     else:\n",
    "#         hifi_sample_missing_1_submitter_df = hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin([sample])]\n",
    "#         if hifi_sample_missing_1_submitter_df.columns[hifi_sample_missing_1_submitter_df.isna().any()].tolist() == ['notes']:\n",
    "#             pass\n",
    "#         else:\n",
    "#             # print(sample, hifi_sample_missing_1_submitter_df.columns[hifi_sample_missing_1_submitter_df.isna().any()].tolist())\n",
    "#             hifi_sample_missing_1_submitter_list.append(sample)\n",
    "# print(len(hifi_sample_missing_1_submitter_list))\n",
    "\n",
    "# hifi_sample_meta_df[hifi_sample_meta_df['sample_ID'].isin(hifi_sample_missing_1_submitter_list)].reset_index().to_csv('hprc_metadata_sample_files_missing_1_submitter_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "7c2f06ce-35fd-4189-bfd2-0de55acf4de9",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_readstats_df = hifi_df[required_hifi_5_readstats_columns]\n",
    "hifi_sample_readstats_df = hifi_sample_readstats_df.groupby('sample_ID').agg({'total_reads': 'sum',\n",
    "                                                   'total_bp': 'sum',\n",
    "                                                   'total_Gbp': 'sum',\n",
    "                                                   'min': 'mean',\n",
    "                                                   'max': 'mean',\n",
    "                                                   'quartile_25': 'mean',\n",
    "                                                   'quartile_50': 'mean',\n",
    "                                                   'quartile_75': 'mean',\n",
    "                                                   'N25': 'mean',\n",
    "                                                   'N50': 'mean', # assembly term shortest contig of the total assembly length (different for ONT)\n",
    "                                                   'N75': 'mean'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "fa772927-2134-43ee-81fb-6301dec93e2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df = pd.merge(hifi_1_submitter_df, hifi_sample_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "672deb01-56a7-4a4f-9af8-938224e3ff67",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check that sample's with missing 1_submitter metadata are present\n",
    "assert hifi_sample_df[hifi_sample_df['sample_ID'].isin(samples_missing_submmiter_1_hifi)].shape[0] == len(set(samples_missing_submmiter_1_hifi))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "59a94d0f-69f3-4bd2-9a1a-5386dc0957fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "hifi_sample_df.to_csv('hprc_metadata_sample_aggregate_HiFi.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "0e135ef1-15dd-4a71-a466-d13c7ef1828d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(122, 30)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hifi_sample_df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5a4b645-6bc4-4094-881d-d83112a51c5b",
   "metadata": {},
   "source": [
    "# Deep Consensus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3ea8e4ee-9519-4415-a349-4d8586ccea2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>accession</th>\n",
       "      <th>study</th>\n",
       "      <th>biosample_accession</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>total_bp</th>\n",
       "      <th>total_Gbp</th>\n",
       "      <th>min</th>\n",
       "      <th>...</th>\n",
       "      <th>design_description</th>\n",
       "      <th>data_type</th>\n",
       "      <th>shear_method</th>\n",
       "      <th>size_selection</th>\n",
       "      <th>DeepConsensus_version</th>\n",
       "      <th>polymerase_version</th>\n",
       "      <th>seq_plate_chemistry_version</th>\n",
       "      <th>generator_facility</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00099.m54329U_220825_174247.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2687625</td>\n",
       "      <td>56102007798</td>\n",
       "      <td>56.10</td>\n",
       "      <td>107</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00099.m54329U_220827_143814.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2643186</td>\n",
       "      <td>52918724515</td>\n",
       "      <td>52.92</td>\n",
       "      <td>345</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099.m54329U_220829_095708.dc.q20.fastq.gz</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00099/ra...</td>\n",
       "      <td>SRR26545347</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33758778</td>\n",
       "      <td>2866917</td>\n",
       "      <td>57539149577</td>\n",
       "      <td>57.54</td>\n",
       "      <td>138</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00140.m64043_220728_173215.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00140/ra...</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33621941</td>\n",
       "      <td>572695</td>\n",
       "      <td>11345594738</td>\n",
       "      <td>11.35</td>\n",
       "      <td>117</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00140.m64136_220715_182717.dc.q20.fastq.gz</td>\n",
       "      <td>HG00140</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00140/ra...</td>\n",
       "      <td>SRR26545346</td>\n",
       "      <td>SRP305758</td>\n",
       "      <td>SAMN33621941</td>\n",
       "      <td>1674999</td>\n",
       "      <td>36918329655</td>\n",
       "      <td>36.92</td>\n",
       "      <td>400</td>\n",
       "      <td>...</td>\n",
       "      <td>HiFi sequencing of 20kb fractionated gDNA reba...</td>\n",
       "      <td>unaligned reads</td>\n",
       "      <td>Megaruptor 3</td>\n",
       "      <td>PippinHT</td>\n",
       "      <td>1.2</td>\n",
       "      <td>P2</td>\n",
       "      <td>C2</td>\n",
       "      <td>University of Washington</td>\n",
       "      <td>kmiyamot@uw.edu</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        filename sample_ID  \\\n",
       "0  HG00099.m54329U_220825_174247.dc.q20.fastq.gz   HG00099   \n",
       "1  HG00099.m54329U_220827_143814.dc.q20.fastq.gz   HG00099   \n",
       "2  HG00099.m54329U_220829_095708.dc.q20.fastq.gz   HG00099   \n",
       "3   HG00140.m64043_220728_173215.dc.q20.fastq.gz   HG00140   \n",
       "4   HG00140.m64136_220715_182717.dc.q20.fastq.gz   HG00140   \n",
       "\n",
       "                                                path    accession      study  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "1  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "2  s3://human-pangenomics/working/HPRC/HG00099/ra...  SRR26545347  SRP305758   \n",
       "3  s3://human-pangenomics/working/HPRC/HG00140/ra...  SRR26545346  SRP305758   \n",
       "4  s3://human-pangenomics/working/HPRC/HG00140/ra...  SRR26545346  SRP305758   \n",
       "\n",
       "  biosample_accession  total_reads     total_bp  total_Gbp  min  ...  \\\n",
       "0        SAMN33758778      2687625  56102007798      56.10  107  ...   \n",
       "1        SAMN33758778      2643186  52918724515      52.92  345  ...   \n",
       "2        SAMN33758778      2866917  57539149577      57.54  138  ...   \n",
       "3        SAMN33621941       572695  11345594738      11.35  117  ...   \n",
       "4        SAMN33621941      1674999  36918329655      36.92  400  ...   \n",
       "\n",
       "                                  design_description        data_type  \\\n",
       "0  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "1  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "2  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "3  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "4  HiFi sequencing of 20kb fractionated gDNA reba...  unaligned reads   \n",
       "\n",
       "   shear_method  size_selection  DeepConsensus_version  polymerase_version  \\\n",
       "0  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "1  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "2  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "3  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "4  Megaruptor 3        PippinHT                    1.2                  P2   \n",
       "\n",
       "   seq_plate_chemistry_version        generator_facility generator_contact  \\\n",
       "0                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "1                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "2                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "3                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "4                           C2  University of Washington   kmiyamot@uw.edu   \n",
       "\n",
       "  notes  \n",
       "0   NaN  \n",
       "1   NaN  \n",
       "2   NaN  \n",
       "3   NaN  \n",
       "4   NaN  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dc_df = pd.read_csv('hprc_metadata_sample_files_DEEPCONSENSUS.tsv', sep='\\t')\n",
    "dc_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "82d21b80-2594-4ee5-8309-6d17420847c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove notes if all nan\n",
    "if dc_df['notes'].isnull().all() == True:\n",
    "    dc_df = dc_df.drop('notes', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e506f5fd-e106-4919-a6bd-7b020099e79f",
   "metadata": {},
   "outputs": [],
   "source": [
    "required_deepconsensus_1_submitter_columns = ['filename', 'sample_ID', 'library_ID', 'library_strategy', 'library_source', \n",
    "                                            'library_selection', 'library_layout', 'platform', 'instrument_model', 'design_description', \n",
    "                                            'data_type', 'shear_method', 'size_selection', \n",
    "                                            'DeepConsensus_version', 'polymerase_version', \n",
    "                                            'seq_plate_chemistry_version', 'generator_facility', \n",
    "                                            'generator_contact']\n",
    "\n",
    "required_deepconsensus_5_readstats_columns = ['sample_ID', 'filename', 'total_reads', 'total_bp', 'total_Gbp', 'min', 'max', 'mean', 'quartile_25', 'quartile_50', 'quartile_75', 'N25', 'N50', 'N75']\n",
    "\n",
    "required_deepconsensus_5_readstat_dict = {'total_reads': 'sum',\n",
    "                                               'total_bp': 'sum',\n",
    "                                               'total_Gbp': 'sum',\n",
    "                                               'min': 'mean',\n",
    "                                               'max': 'mean',\n",
    "                                               'quartile_25': 'mean',\n",
    "                                               'quartile_50': 'mean',\n",
    "                                               'quartile_75': 'mean',\n",
    "                                               'N25': 'mean',\n",
    "                                               'N50': 'mean',\n",
    "                                               'N75': 'mean'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bcf29ccc-bfe4-4767-a404-b2604ed3a20e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_1_submitter_df, samples_missing_1_submmiter_hifi = check_missing_1_submitter(dc_df, \n",
    "                                                                                required_deepconsensus_1_submitter_columns)\n",
    "if len(samples_missing_1_submmiter_hifi) > 0:\n",
    "    dc_1_submitter_df[hifi_1_submitter_df['sample_ID'].isin(samples_missing_1_submmiter_hifi)].to_csv('hprc_metadata_sample_files_missing_1_submitter_DeepConsensus.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "bd8a82a3-bd9a-46f4-9171-ad4b9443b703",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample_df, required_readstat_columns\n",
    "dc_5_readstats_df = sample_aggregate_5_readstats(dc_df, \n",
    "                                                 required_deepconsensus_5_readstats_columns, \n",
    "                                                 required_deepconsensus_5_readstat_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e5f86393-3631-4e91-b7be-f1f462d26693",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_df = pd.merge(dc_1_submitter_df, dc_5_readstats_df, on='sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "591d4c7f-5ec3-4409-bb9b-439a03091d6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dc_sample_df.to_csv('hprc_metadata_sample_aggregate_DEEPCONSENSUS.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26491e20-a204-4a0f-bf90-40231f6187e3",
   "metadata": {},
   "source": [
    "# Sample Aggregate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "50d26c92-2ec6-470c-b60a-742ac8dc65d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "merge_samples = set(dc_sample_df['sample_ID'].tolist()).union(hifi_sample_df['sample_ID'].tolist(), ont_sample_df['sample_ID'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "c5a6fc28-704e-4f08-8a7f-9847fc0370ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename</th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>path</th>\n",
       "      <th>filetype</th>\n",
       "      <th>library_ID</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>platform</th>\n",
       "      <th>...</th>\n",
       "      <th>generator_contact</th>\n",
       "      <th>read_N50</th>\n",
       "      <th>coverage</th>\n",
       "      <th>100kb+</th>\n",
       "      <th>200kb+</th>\n",
       "      <th>300kb+</th>\n",
       "      <th>400kb+</th>\n",
       "      <th>500kb+</th>\n",
       "      <th>1Mb+</th>\n",
       "      <th>whales</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>02_08_22_R941_HG00558_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG00558</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00558/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG00558_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>98073.000000</td>\n",
       "      <td>54.43</td>\n",
       "      <td>26.69</td>\n",
       "      <td>7.30</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.45</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>02_08_22_R941_HG01252_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG01252</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG01252/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG01252_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>79697.666667</td>\n",
       "      <td>47.53</td>\n",
       "      <td>17.45</td>\n",
       "      <td>3.02</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>02_08_22_R941_HG02258_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG02258</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG02258/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG02258_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>98838.500000</td>\n",
       "      <td>49.73</td>\n",
       "      <td>25.01</td>\n",
       "      <td>6.62</td>\n",
       "      <td>1.26</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG03834</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG03834/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_08_22_R941_HG03834_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>107647.333333</td>\n",
       "      <td>48.80</td>\n",
       "      <td>26.33</td>\n",
       "      <td>8.97</td>\n",
       "      <td>3.23</td>\n",
       "      <td>1.35</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.03</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>02_15_22_R941_HG00658_1_Guppy_6.5.7_450bps_mod...</td>\n",
       "      <td>HG00658</td>\n",
       "      <td>s3://human-pangenomics/working/HPRC/HG00658/ra...</td>\n",
       "      <td>bam</td>\n",
       "      <td>02_15_22_R941_HG00658_1</td>\n",
       "      <td>WGS</td>\n",
       "      <td>GENOMIC</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>single</td>\n",
       "      <td>OXFORD_NANOPORE</td>\n",
       "      <td>...</td>\n",
       "      <td>iviolich@ucsc.edu</td>\n",
       "      <td>90313.333333</td>\n",
       "      <td>68.25</td>\n",
       "      <td>30.80</td>\n",
       "      <td>8.37</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            filename sample_ID  \\\n",
       "0  02_08_22_R941_HG00558_1_Guppy_6.5.7_450bps_mod...   HG00558   \n",
       "1  02_08_22_R941_HG01252_1_Guppy_6.5.7_450bps_mod...   HG01252   \n",
       "2  02_08_22_R941_HG02258_1_Guppy_6.5.7_450bps_mod...   HG02258   \n",
       "3  02_08_22_R941_HG03834_1_Guppy_6.5.7_450bps_mod...   HG03834   \n",
       "4  02_15_22_R941_HG00658_1_Guppy_6.5.7_450bps_mod...   HG00658   \n",
       "\n",
       "                                                path filetype  \\\n",
       "0  s3://human-pangenomics/working/HPRC/HG00558/ra...      bam   \n",
       "1  s3://human-pangenomics/working/HPRC/HG01252/ra...      bam   \n",
       "2  s3://human-pangenomics/working/HPRC/HG02258/ra...      bam   \n",
       "3  s3://human-pangenomics/working/HPRC/HG03834/ra...      bam   \n",
       "4  s3://human-pangenomics/working/HPRC/HG00658/ra...      bam   \n",
       "\n",
       "                library_ID library_strategy library_source library_selection  \\\n",
       "0  02_08_22_R941_HG00558_1              WGS        GENOMIC            RANDOM   \n",
       "1  02_08_22_R941_HG01252_1              WGS        GENOMIC            RANDOM   \n",
       "2  02_08_22_R941_HG02258_1              WGS        GENOMIC            RANDOM   \n",
       "3  02_08_22_R941_HG03834_1              WGS        GENOMIC            RANDOM   \n",
       "4  02_15_22_R941_HG00658_1              WGS        GENOMIC            RANDOM   \n",
       "\n",
       "  library_layout         platform  ...  generator_contact       read_N50  \\\n",
       "0         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   98073.000000   \n",
       "1         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   79697.666667   \n",
       "2         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   98838.500000   \n",
       "3         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu  107647.333333   \n",
       "4         single  OXFORD_NANOPORE  ...  iviolich@ucsc.edu   90313.333333   \n",
       "\n",
       "  coverage 100kb+ 200kb+ 300kb+ 400kb+ 500kb+  1Mb+ whales  \n",
       "0    54.43  26.69   7.30   1.60   0.45   0.18  0.00     35  \n",
       "1    47.53  17.45   3.02   0.66   0.18   0.07  0.00      7  \n",
       "2    49.73  25.01   6.62   1.26   0.26   0.08  0.00      5  \n",
       "3    48.80  26.33   8.97   3.23   1.35   0.64  0.03     99  \n",
       "4    68.25  30.80   8.37   1.85   0.49   0.18  0.00     16  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84027db9-8584-496d-ac1d-ab23844da48d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ont_sample_df.groupby(['Age', 'Gender']).agg(lambda x: x.tolist()).reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "addd0b0a-0220-421c-b709-41a0aac5aa12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_ID</th>\n",
       "      <th>filename</th>\n",
       "      <th>path</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GM18522</td>\n",
       "      <td>[05_17_22_R941_GM18522_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18522/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GM18570</td>\n",
       "      <td>[08_10_22_R941_GM18570_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18570/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GM18612</td>\n",
       "      <td>[06_28_22_R941_GM18612_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18612/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GM18747</td>\n",
       "      <td>[06_28_22_R941_GM18747_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18747/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GM18971</td>\n",
       "      <td>[06_28_22_R941_GM18971_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/NA18971/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>HG04184</td>\n",
       "      <td>[08_25_21_R941_HG04184_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04184/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>HG04187</td>\n",
       "      <td>[08_10_21_R941_HG04187_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04187/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>HG04199</td>\n",
       "      <td>[08_10_21_R941_HG04199_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04199/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>HG04204</td>\n",
       "      <td>[08_10_21_R941_HG04204_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04204/r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>HG04228</td>\n",
       "      <td>[08_10_21_R941_HG04228_1_Guppy_6.5.7_450bps_mo...</td>\n",
       "      <td>[s3://human-pangenomics/working/HPRC/HG04228/r...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>124 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_ID                                           filename  \\\n",
       "0     GM18522  [05_17_22_R941_GM18522_1_Guppy_6.5.7_450bps_mo...   \n",
       "1     GM18570  [08_10_22_R941_GM18570_1_Guppy_6.5.7_450bps_mo...   \n",
       "2     GM18612  [06_28_22_R941_GM18612_1_Guppy_6.5.7_450bps_mo...   \n",
       "3     GM18747  [06_28_22_R941_GM18747_1_Guppy_6.5.7_450bps_mo...   \n",
       "4     GM18971  [06_28_22_R941_GM18971_1_Guppy_6.5.7_450bps_mo...   \n",
       "..        ...                                                ...   \n",
       "119   HG04184  [08_25_21_R941_HG04184_1_Guppy_6.5.7_450bps_mo...   \n",
       "120   HG04187  [08_10_21_R941_HG04187_1_Guppy_6.5.7_450bps_mo...   \n",
       "121   HG04199  [08_10_21_R941_HG04199_1_Guppy_6.5.7_450bps_mo...   \n",
       "122   HG04204  [08_10_21_R941_HG04204_1_Guppy_6.5.7_450bps_mo...   \n",
       "123   HG04228  [08_10_21_R941_HG04228_1_Guppy_6.5.7_450bps_mo...   \n",
       "\n",
       "                                                  path  \n",
       "0    [s3://human-pangenomics/working/HPRC/NA18522/r...  \n",
       "1    [s3://human-pangenomics/working/HPRC/NA18570/r...  \n",
       "2    [s3://human-pangenomics/working/HPRC/NA18612/r...  \n",
       "3    [s3://human-pangenomics/working/HPRC/NA18747/r...  \n",
       "4    [s3://human-pangenomics/working/HPRC/NA18971/r...  \n",
       "..                                                 ...  \n",
       "119  [s3://human-pangenomics/working/HPRC/HG04184/r...  \n",
       "120  [s3://human-pangenomics/working/HPRC/HG04187/r...  \n",
       "121  [s3://human-pangenomics/working/HPRC/HG04199/r...  \n",
       "122  [s3://human-pangenomics/working/HPRC/HG04204/r...  \n",
       "123  [s3://human-pangenomics/working/HPRC/HG04228/r...  \n",
       "\n",
       "[124 rows x 3 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ont_sample_df[['sample_ID','filename','path']].groupby(['sample_ID']).agg(lambda x: x.tolist()).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f615cdf-fae2-4907-84ba-7345f2715996",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
